---
title: "GWA WDH Barley"
author: "Karl Kunze"
date: "1/7/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---
Workflow:
Winter_DH_data_consolidation.Rmd ->analysis_script.R->WDH_GenotypeData_Processing.Rmd->GWA WDH Barley.Rmd
Add_morex_v2_postions.. file is for aligning correct genome
read in phenotype files
```{r}
library(here)
dir1<-"C:/Users/Karl/git/BarleyGermination-KK"
setwd(dir1)
getwd()
load(here("WMB DH 2020 all data for analysis.RData"))
#load(here("myGM20_LDprune.Rdata"))
#load(here("myGD20_LDprune.Rdata"))

load(here("myGD20.Rdata"))
load(here("myGM20.Rdata"))
load(here("AlaT_markers.Rdata"))
colnames(AlaT_markers)[4]<-"Entry"
table(All_pheno_TPall$Entry)
str(myGM20$SNP)

All_pheno_TPall
myGD20$taxa
dim(myGD20)
str(All_pheno_TPall$Entry)
All_pheno_TPall<-All_pheno_TPall[!is.na(All_pheno_TPall$Entry),]
All_pheno_TPall[All_pheno_TPall$Entry=="Check 1-Flavia",]$Entry<-"Flavia"
All_pheno_TPall[All_pheno_TPall$Entry=="Check 2-Scala",]$Entry<-"Scala"
All_pheno_TPall[All_pheno_TPall$Entry=="Check 3-DH130910",]$Entry<-"DH130910"
All_pheno_TPall[All_pheno_TPall$Entry=="Check 4-SY Tepee",]$Entry<-"SY Tepee"
All_pheno_TPall[All_pheno_TPall$Entry=="Check 5-Wintmalt",]$Entry<-"Wintmalt"
WDH20_pheno<-All_pheno_TPall

str(WDH20_pheno)
#make separate data object for field phenotypes
colnames(WDH20_pheno)
WDH20_phenoF<-WDH20_pheno[,c(1:27,30)]
WDH20_phenoF<-WDH20_phenoF[!duplicated(WDH20_phenoF$PLOT), ]#remove duplication
str(WDH20_phenoF)
table(WDH20_phenoF$Scald)
```
Read in packages and functions
```{r}
#GAPIT functions

library(rrBLUP)
library(lme4)
library(readxl)
library(readr)
library(gplots)
library(LDheatmap)
library(genetics)
library(ape)
library(EMMREML)
library("scatterplot3d")
library(compiler)

#source code for GAPIT functions since GAPIT is not on CRAN
source("http://zzlab.net/GAPIT/gapit_functions.txt")

```
Germination Time point models
```{r}
#separate data based on timepoints
str(WDH20_pheno)
TP1WDH<-droplevels(WDH20_pheno[WDH20_pheno$Timepoint=="1",])
TP2WDH<-droplevels(WDH20_pheno[WDH20_pheno$Timepoint=="2",])
TP3WDH<-droplevels(WDH20_pheno[WDH20_pheno$Timepoint=="3",])
TP4WDH<-droplevels(WDH20_pheno[WDH20_pheno$Timepoint=="4",])
str(TP4WDH)



TP1WDH<-TP1WDH[!is.na(TP1WDH$PLOT),]
TP2WDH<-TP2WDH[!is.na(TP2WDH$PLOT),]
TP3WDH<-TP3WDH[!is.na(TP3WDH$PLOT),]
TP4WDH<-TP4WDH[!is.na(TP4WDH$PLOT),]
TP4WDH[TP4WDH$PLOT==11023,]#checking to see if the a{1}1023 lines were retained


```

```{r}
WDH20phs.lm=lm(phs~Entry+PM, data=WDH20_pheno)

anova(WDH20phs.lm) #all factors signif
#TP1
wdhge1.lm=lm(GE~Entry + Location, data=TP1WDH)

anova(wdhge1.lm) #location and PM sg
str(TP1WDH)
wdhgi1.lm=lm(GIscale~ Entry+Location, data=TP1WDH)

anova(wdhgi1.lm) #location slightly s, PM

wdhge1_5D.lm=lm(GE_5D~ Entry +Location:Scald, data=TP1WDH)
anova(wdhge1_5D.lm) #location and PM sg
wdhgi1_5D.lm=lm(GIscale_5D~ Entry+Scald, data=TP1WDH)
anova(wdhgi1_5D.lm) #location slightly s, PM

wdhge1.lm
wdhgi1.lm
wdhge1_5D.lm
wdhgi1_5D.lm



#TP2
wdhge2.lm=lm(GE~ Entry +Location, data=TP2WDH)
anova(wdhge2.lm) # all signif
wdhgi2.lm=lm(GIscale~ Entry +Location, data=TP2WDH)
anova(wdhgi2.lm) #all signif

wdhge2_5D.lm=lm(GE_5D~ Entry +Location, data=TP2WDH)
anova(wdhge2_5D.lm) # all signif
wdhgi2_5D.lm=lm(GIscale_5D~ Entry +Location, data=TP2WDH)
anova(wdhgi2_5D.lm) #all signif
#TP3
wdhge3.lm=lm(GE~ Entry +Location+ replication, data=TP3WDH)
anova(wdhge3.lm) # all signif
wdhgi3.lm=lm(GIscale~ Entry +Location, data=TP3WDH)
anova(wdhgi3.lm) #all signif

wdhge3_5D.lm=lm(GE_5D~ Entry, data=TP3WDH)
anova(wdhge3_5D.lm) # only Entry is significant
wdhgi3_5D.lm=lm(GIscale_5D~ Entry +Location + replication, data=TP3WDH)
anova(wdhgi3_5D.lm) #all signif
#TP4
wdhge4.lm=lm(GE~ Entry +Location+ replication, data=TP4WDH)
anova(wdhge4.lm) # rep slightly sign
wdhgi4.lm=lm(GIscale~ Entry +replication, data=TP4WDH)
#should consider not using PM as fixed effect due to highly unbalanced set
anova(wdhgi4.lm) #all signif

wdhge4_5D.lm=lm(GE_5D~ Entry +Location+ replication, data=TP4WDH)
anova(wdhge4_5D.lm) # only Entry is significant
wdhgi4_5D.lm=lm(GIscale_5D~ Entry+ replication, data=TP4WDH)
anova(wdhgi4_5D.lm) #all signif

```
Analysis of phenotypes
```{r}
hist(WDH20_phenoF$Ht)

```

```{r}
#Field Phenotypes fixed effects
str(WDH20_phenoF)
colnames(WDH20_phenoF)

wdh_scald.lm=lm(Scald~Entry+Pedigree,data=WDH20_phenoF)

anova(wdh_scald.lm)

wdh_wh.lm=lm(wh_percent~Entry+Location,data=WDH20_phenoF)
anova(wdh_wh.lm)
#cant include check because there isn't balance
wdh_HD.lm=lm(HD~Entry+Location,data=WDH20_phenoF)
anova(wdh_HD.lm)
wdh_Ht.lm=lm(Ht~Entry+Location,data=WDH20_phenoF)
anova(wdh_Ht.lm)
wdh_PM.lm=lm(PM~Entry+Location+wh_percent,data=WDH20_phenoF)
anova(wdh_PM.lm)



##BLUPS
#base mixed model for each trait for genotypic BLUPs
wdhphs.lmer=lmer(phs~1+(1|Entry), data=WDH20_pheno)
summary(wdhphs.lmer)

wdhGE_TP1.lmer=lmer(GE~ Location + (1|Entry), data=TP1WDH)
AIC(wdhGE_TP1.lmer)
wdhGI_TP1.lmer=lmer(GIscale~ Location +(1|PM)+ (1|Entry), data=TP1WDH)
AIC(wdhGI_TP1.lmer)
wdhGE_TP1_5D.lmer=lmer(GE_5D~ Location +(1|PM)+ (1|Entry), data=TP1WDH)
AIC(wdhGE_TP1_5D.lmer)
wdhGI_TP1_5D.lmer=lmer(GIscale_5D~ Location + (1|PM )+ (1|Entry), data=TP1WDH)
AIC(wdhGI_TP1_5D.lmer)







wdhGE_TP2.lmer=lmer(GE~  Location + PM+ (1|Entry) , data=TP2WDH)
AIC(wdhGE_TP2.lmer)
wdhGI_TP2.lmer=lmer(GIscale~Location +(1|PM)+ (1|Entry), data=TP2WDH)
AIC(wdhGI_TP2.lmer)
wdhGE_TP2_5D.lmer=lmer(GE_5D~  Location + PM+ (1|Entry) , data=TP2WDH)
AIC(wdhGE_TP2_5D.lmer)
wdhGI_TP2_5D.lmer=lmer(GIscale_5D~  Location +PM + (1|Entry), data=TP2WDH)
AIC(wdhGI_TP2_5D.lmer)
wdhGE_TP3.lmer=lmer(GE~  Location + PM+PM:replication +(1|Entry) , data=TP3WDH)
AIC(wdhGE_TP3.lmer)
wdhGI_TP3.lmer=lmer(GIscale~  Location +PM+PM:replication + (1|Entry), data=TP3WDH)
AIC(wdhGI_TP3.lmer)


wdhGE_TP3_5D.lmer=lmer(GE_5D~  Location +(1|PM)+replication + (1|Entry) , data=TP3WDH)
AIC(wdhGE_TP3_5D.lmer)
wdhGI_TP3_5D.lmer=lmer(GIscale_5D~  Location+ (1|PM)+ (1|Entry), data=TP3WDH)
AIC(wdhGI_TP3_5D.lmer)

wdhGE_TP4.lmer=lmer(GE~  Location + replication +(1|Entry) , data=TP4WDH)
AIC(wdhGE_TP4.lmer)

wdhGI_TP4.lmer=lmer(GIscale~  Location +replication + (1|Entry), data=TP4WDH)
AIC(wdhGI_TP4.lmer)
wdhGE_TP4_5D.lmer=lmer(GE_5D~  Location + replication+ (1|PM) + (1|Entry) , data=TP4WDH)
AIC(wdhGE_TP4_5D.lmer)

wdhGI_TP4_5D.lmer=lmer(GIscale_5D~  Location +PM+PM:replication  + (1|Entry), data=TP4WDH)
AIC(wdhGI_TP4_5D.lmer)


```
Field Phenotypes BLUPS
```{r}
#scald
wdh_scald.lmer=lmer(Scald~(1|Entry),data=WDH20_phenoF)
summary(wdh_scald.lmer)
AIC(wdh_scald.lmer)
anova(wdh_scald.lmer)
#winter hardiness percentage
wdh_wh.lmer=lmer(wh_percent~(1|Entry)+HD,data=WDH20_phenoF)
summary(wdh_wh.lmer)
AIC(wdh_wh.lmer)
anova(wdh_wh.lmer)
Cullis_H2(wdh_wh.lmer)
#heading data
wdh_HD.lmer=lmer(HD~(1|Entry)+Location,data=WDH20_phenoF)
summary(wdh_HD.lmer)
AIC(wdh_HD.lmer)
anova(wdh_HD.lmer)
#height
wdh_Ht.lmer=lmer(Ht~(1|Entry)+Location+HD,data=WDH20_phenoF)
summary(wdh_Ht.lmer)
AIC(wdh_Ht.lmer)
anova(wdh_Ht.lmer)
#Physiological maturity
wdh_PM.lmer=lmer(PM~(1|Entry)+Location+Check,data=WDH20_phenoF)
summary(wdh_PM.lmer)
AIC(wdh_PM.lmer)
anova(wdh_PM.lmer)

```
Cullis Heritability for unbalanced designs
```{r}
Cullis_H2=function(model){
  library(arm)
  ses<- se.ranef(model)$'Entry' #where 'm' is your model object from 'lmer' (replace 'genotypes' with whatever you call your individuals in the data)
  v_BLUP<- ses^2
  sigma2_g=VarCorr(model, comp="Variance")$'Entry'[1]
  Reliability<- 1- v_BLUP/ (2*sigma2_g)  #where sigma2_g is the genetic variance estimated with the model saved in 'm'
  H2<- round(mean(Reliability),3) #This is equivalent to broad-sense heritability on the line-mean (or family-mean, if your individuals are non-inbred families) basis
  H2
}



Cullis_H2(wdhphs.lmer)#0.999
Cullis_H2(wdh_scald.lmer)#0.728
Cullis_H2(wdh_wh.lmer)#0.706
Cullis_H2(wdh_HD.lmer)#0.915
Cullis_H2(wdh_Ht.lmer)#0.695 ,0.779<-there's a data issue here
Cullis_H2(wdh_PM.lmer)#0.777
#TP1
Cullis_H2(wdhGE_TP1.lmer)#0.982
Cullis_H2(wdhGI_TP1.lmer)#0.973
Cullis_H2(wdhGE_TP1_5D.lmer)#0.97
Cullis_H2(wdhGI_TP1_5D.lmer)#0.97
#TP2
Cullis_H2(wdhGE_TP2.lmer)#0.984
Cullis_H2(wdhGI_TP2.lmer)#0.989
Cullis_H2(wdhGE_TP2_5D.lmer)#0.958
Cullis_H2(wdhGI_TP2_5D.lmer)#0.988
#TP3
Cullis_H2(wdhGE_TP3.lmer)#0.923
Cullis_H2(wdhGI_TP3.lmer)#0.976
Cullis_H2(wdhGE_TP3_5D.lmer)#0.762
Cullis_H2(wdhGI_TP3_5D.lmer)#0.974
#TP4
Cullis_H2(wdhGE_TP4.lmer)#0.855
Cullis_H2(wdhGI_TP4.lmer)#0.939
Cullis_H2(wdhGE_TP4_5D.lmer)#0.7
Cullis_H2(wdhGI_TP4_5D.lmer)#0.934



```

```{r}
get_entry_fixed_Effects = function(model, trait){
  x = data.frame(coef(model))
  x$names = rownames(x)
  x$names2 = substr(x$names,1,5)
  x = subset(x, x$names2 %in% c('(Inte', 'Entry'))
  x$taxa = substring(x$names, 6)
  x$taxa[1] = model$xlevels$Entry[1]
  intercept = x$coef.model.[1]
  x$coef.model.[1] = 0
  x = data.frame(taxa = x$taxa, x$coef.model.)
  colnames(x)[2] = trait
  return(x)
}



wdhge1.lm
wdhgi1.lm
wdhge1_5D.lm
wdhgi1_5D.lm
phs.blues = get_entry_fixed_Effects(WDH20phs.lm, 'phs')

GETP1.blues = get_entry_fixed_Effects(wdhge1.lm, 'GE_TP1')
GITP1.blues = get_entry_fixed_Effects(wdhgi1.lm, 'GI_TP1')

GETP1_5D.blues = get_entry_fixed_Effects(wdhge1_5D.lm, 'GE_5D_TP1')
GITP1_5D.blues = get_entry_fixed_Effects(wdhgi1_5D.lm, 'GI_5D_TP1')


GETP2.blues = get_entry_fixed_Effects(wdhge2.lm, 'GE_TP2')
GITP2.blues = get_entry_fixed_Effects(wdhgi2.lm, 'GI_TP2')

GETP2_5D.blues = get_entry_fixed_Effects(wdhge2_5D.lm, 'GE_5D_TP2')
GITP2_5D.blues = get_entry_fixed_Effects(wdhgi2_5D.lm, 'GI_5D_TP2')


GETP3.blues = get_entry_fixed_Effects(wdhge3.lm, 'GE_TP3')
GITP3.blues = get_entry_fixed_Effects(wdhgi3.lm, 'GI_TP3')

GETP3_5D.blues = get_entry_fixed_Effects(wdhge3_5D.lm, 'GE_5D_TP3')
GITP3_5D.blues = get_entry_fixed_Effects(wdhgi3_5D.lm, 'GI_5D_TP3')



GETP4.blues = get_entry_fixed_Effects(wdhge4.lm, 'GE_TP4')
GITP4.blues = get_entry_fixed_Effects(wdhgi4.lm, 'GI_TP4')

GETP4_5D.blues = get_entry_fixed_Effects(wdhge4_5D.lm, 'GE_5D_TP4')
GITP4_5D.blues = get_entry_fixed_Effects(wdhgi4_5D.lm, 'GI_5D_TP4')

wdh_wh.blues = get_entry_fixed_Effects(wdh_wh.lm, 'wh')
wdh_HD.blues = get_entry_fixed_Effects(wdh_HD.lm, 'HD')
wdh_scald.blues = get_entry_fixed_Effects(wdh_scald.lm, 'Scald')


wdh_Ht.blues = get_entry_fixed_Effects(wdh_Ht.lm, 'Ht')
wdh_PM.blues = get_entry_fixed_Effects(wdh_PM.lm, 'PM')

###
#library(lsmeans)
library(emmeans)
Entry = GITP3_5D.blues$taxa
Entry
wdhge1.lm
wdhgi1.lm
wdhge1_5D.lm
wdhgi1_5D.lm
GETP1 = data.frame(lsmeans(wdhge1.lm, 'Entry'))
GETP2 = data.frame(lsmeans(wdhge2.lm, 'Entry'))
GETP3 = data.frame(lsmeans(wdhge3.lm, 'Entry'))
GETP4 = data.frame(lsmeans(wdhge4.lm, 'Entry'))

GITP1 = data.frame(lsmeans(wdhgi1.lm, 'Entry'))
GITP2 = data.frame(lsmeans(wdhgi2.lm, 'Entry'))
GITP3 = data.frame(lsmeans(wdhgi3.lm, 'Entry'))
GITP4 = data.frame(lsmeans(wdhgi4.lm, 'Entry'))

GETP1_5D = data.frame(lsmeans(wdhge1_5D.lm, 'Entry'))
GETP2_5D = data.frame(lsmeans(wdhge2_5D.lm, 'Entry'))
GETP3_5D = data.frame(lsmeans(wdhge3_5D.lm, 'Entry'))
GETP4_5D = data.frame(lsmeans(wdhge4_5D.lm, 'Entry'))

GITP1_5D = data.frame(lsmeans(wdhgi1_5D.lm, 'Entry'))
GITP2_5D = data.frame(lsmeans(wdhgi2_5D.lm, 'Entry'))
GITP3_5D = data.frame(lsmeans(wdhgi3_5D.lm, 'Entry'))
GITP4_5D = data.frame(lsmeans(wdhgi4_5D.lm, 'Entry'))

PHSlsmeans = data.frame(lsmeans(WDH20phs.lm, 'Entry'))
scaldmeans = data.frame(lsmeans(wdh_scald.lm, 'Entry'))
wh_means = data.frame(lsmeans(wdh_wh.lm, 'Entry'))
HD_means = data.frame(lsmeans(wdh_HD.lm, 'Entry'))
PM_means = data.frame(lsmeans(wdh_PM.lm, 'Entry'))
Ht_means = data.frame(lsmeans(wdh_Ht.lm, 'Entry'))


WDH_BLUEs_2020 = Reduce(function(x, y) merge(x, y, all = TRUE,by = 'Entry'), 
                        list(PHSlsmeans[,1:2], GETP1[,1:2], GETP2[,1:2], GETP3[,1:2],GETP4[,1:2], GITP1[,1:2], GITP2[,1:2], GITP3[,1:2],GITP4[,1:2],GETP1_5D[,1:2],GETP2_5D[,1:2],GETP3_5D[,1:2],GETP4_5D[,1:2],GITP1_5D[,1:2],GITP2_5D[,1:2],GITP3_5D[,1:2],GITP4_5D[,1:2],scaldmeans[,1:2],wh_means[,1:2],HD_means[,1:2],PM_means[,1:2],Ht_means[,1:2]))
                             
colnames(WDH_BLUEs_2020) = c('Entry','PHS','GETP1','GETP2','GETP3','GETP4',
                             'GITP1','GITP2','GITP3','GITP4','GETP1_5D','GETP2_5D','GETP3_5D','GETP4_5D','GITP1_5D','GITP2_5D','GITP3_5D','GITP4_5D','scaldmeans','wh_means','HD_means','PM_means','Ht_means')
save(WDH_BLUEs_2020,file ="data/Winter_DH/WDH_BLUEs_2020.RData")




```

```{r}
gapit_blup=function(mod,trait){
  df=ranef(mod)$'Entry'+fixef(mod)[[1]]; 
  df=cbind(rownames(df), df); 
  colnames(df)=c("taxa", trait); 
  df[,1]=as.character(df[,1])
  df
}


wdhGE_TP1.lmer
wdhGI_TP1.lmer
wdhGE_TP1_5D.lmer
wdhGI_TP1_5D.lmer


PHS.blups=gapit_blup(wdhphs.lmer, "PHS")

GETP1.blups=gapit_blup(wdhGE_TP1.lmer, "GE_TP1")
GITP1.blups=gapit_blup(wdhGI_TP1.lmer, "GI_TP1")
GETP1_5D.blups=gapit_blup(wdhGE_TP1_5D.lmer, "GE_TP1")
GITP1_5D.blups=gapit_blup(wdhGI_TP1_5D.lmer, "GI_TP1")

GETP2.blups=gapit_blup(wdhGE_TP2.lmer, "GE_TP2")
GITP2.blups=gapit_blup(wdhGI_TP2.lmer, "GI_TP2")
GETP2_5D.blups=gapit_blup(wdhGE_TP2_5D.lmer, "GE_TP2")
GITP2_5D.blups=gapit_blup(wdhGI_TP2_5D.lmer, "GI_TP2")

GETP3.blups=gapit_blup(wdhGE_TP3.lmer, "GE_TP3")
GITP3.blups=gapit_blup(wdhGI_TP3.lmer, "GI_TP3")
GETP3_5D.blups=gapit_blup(wdhGE_TP3_5D.lmer, "GE_TP3")
GITP3_5D.blups=gapit_blup(wdhGI_TP3_5D.lmer, "GI_TP3")

GETP4.blups=gapit_blup(wdhGE_TP4.lmer, "GE_TP4")
GITP4.blups=gapit_blup(wdhGI_TP4.lmer, "GI_TP4")
GETP4_5D.blups=gapit_blup(wdhGE_TP4_5D.lmer, "GE_TP4")
GITP4_5D.blups=gapit_blup(wdhGI_TP4_5D.lmer, "GI_TP4")


wdh_scald.blups=gapit_blup(wdh_scald.lmer, "Scald")
wh_percent.blups<-gapit_blup(wdh_wh.lmer, "wh_percent")
HD.blups<-gapit_blup(wdh_HD.lmer, "HD")
Ht.blups<-gapit_blup(wdh_Ht.lmer, "Ht")
PM.blups<-gapit_blup(wdh_PM.lmer, "PM")
```
```{r}
table(myGD20$Qsd1)
hist(myGD20$`JHI-Hv50k-2016-38563`)

```

```{r}
# GETP1
getwd()
local1<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP1"
#when setting these directories, using MLMM
setwd(local1)
getwd()
#setwd(dir)<-default working directory
GWAS.GETP1.blues.mlmm  =GAPIT(Y= GETP1.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP1.blues.mlm  =GAPIT(Y= GETP1.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
 
GWAS.GETP1_5D.blues.mlmm  =GAPIT(Y= GETP1_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP1.blues.mlmm  =GAPIT(Y= GITP1.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
str(GWAS.GITP1.blues.mlmm$GWAS)
GWAS.GITP1_5D.blues.mlmm  =GAPIT(Y= GITP1_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T,Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#blups
GWAS.GETP1.blues.mlmm
GWAS.GETP1.blups.mlmm
GWAS.GETP1.blups.mlmm  =GAPIT(Y= GETP1.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GETP1_5D.blups.mlmm  =GAPIT(Y= GETP1_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GITP1.blups.mlmm  =GAPIT(Y= GITP1.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP1_5D.blups.mlmm  =GAPIT(Y= GITP1_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


####TP2###
getwd()
local2<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP2"
#when setting these directories, using MLMM
setwd(local2)
getwd()
#blues
GWAS.GETP2.blues.mlmm  =GAPIT(Y= GETP2.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP2.blues.mlmm =GAPIT(Y= GITP2.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


#GWAS.GITP2.blues_5D.mlmm =GAPIT(Y= GITP2_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP2_5D.blues.mlmm  =GAPIT(Y= GETP2_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP2_5D.blues.mlmm  =GAPIT(Y= GITP2_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
##BLUPS
GWAS.GETP2.blups.mlmm  =GAPIT(Y= GETP2.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP2_5D.blups.mlmm  =GAPIT(Y= GETP2_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)# significant marker on 5H and 1H

GWAS.GITP2.blups.mlmm =GAPIT(Y= GITP2.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)#5H and 1H
GWAS.GITP2_5D.blups.mlmm  =GAPIT(Y= GITP2_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F) 


#GWAS.GITP2.blups_5D.mlmm =GAPIT(Y= GITP2_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)






####TP 3 #####
getwd()
local3<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP3"
#when setting these directories, using MLMM

#BLUES
setwd(local3)
getwd()

GWAS.GETP3.blues.mlmm  =GAPIT(Y= GETP3.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=F, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP3.blues.mlmm  =GAPIT(Y= GITP3.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP3.blues_5D.mlmm  =GAPIT(Y= GETP3_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP3_5D.blues.mlmm  =GAPIT(Y= GITP3_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#BLUPS
GWAS.GETP3.blups.mlmm =GAPIT(Y= GETP3.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP3_5D.blups.mlmm =GAPIT(Y= GETP3_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP3_5D.blups.mlmm  =GAPIT(Y= GITP3_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GITP3_5D.blups.mlmm=GAPIT(Y= GITP3_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

####TP 4#####
getwd()
local4<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP4"
#when setting these directories, using MLMM
setwd(local4)
getwd()
#blues
GWAS.GETP4.blues.mlmm  =GAPIT(Y= GETP4.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4.blues.mlmm  =GAPIT(Y= GITP4.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP4.blues_5D.mlmm  =GAPIT(Y= GETP4_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4_5D.blues.mlmm  =GAPIT(Y= GITP4_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#blups
GWAS.GETP4.blups.mlmm  =GAPIT(Y= GETP4.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4.blups.mlmm  =GAPIT(Y= GITP4.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP4.blups_5D.mlmm  =GAPIT(Y= GETP4_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4_5D.blups.mlmm  =GAPIT(Y= GITP4_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#############Field phenoypes#################

getwd()
localF<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results_field"

#when setting these directories, using MLMM
setwd(localF)
getwd()
#setwd(dir)<-default working directory
#phs
PHS.blues.mlmm  =GAPIT(Y= phs.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#PHS.blups.mlm  =GAPIT(Y= PHS.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
PHS.blups.mlmm  =GAPIT(Y= PHS.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)



#scald
#GWAS.scald.blues.mlm  =GAPIT(Y=wdh_scald.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.scald.blues.mlmm  =GAPIT(Y=wdh_scald.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.scald.blups.mlm =GAPIT(Y=wdhscald.blups, GD=myGD20, GM=myGM20, PCA.total=0,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.scald.blups.mlmm  =GAPIT(Y= wdhscald.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#winter hardiness


#GWAS.wh.blues.mlm  =GAPIT(Y=wdh_wh.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.wh.blues.mlmm  =GAPIT(Y=wdh_wh.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.wh.blups.mlm =GAPIT(Y=wh_percent.blups , GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.wh.blups.mlmm  =GAPIT(Y=wh_percent.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#heading date

#GWAS.HD.blues.mlm  =GAPIT(Y=wdh_HD.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.HD.blues.mlmm  =GAPIT(Y=wdh_HD.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.HD.blups.mlm =GAPIT(Y=HD.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.HD.blups.mlmm  =GAPIT(Y= HD.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.HD.blups.farmcpu =GAPIT(Y= HD.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="FarmCPU", Major.allele.zero = F)




##height
#GWAS.Ht.blues.mlm  =GAPIT(Y=wdh_Ht.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.Ht.blues.mlmm  =GAPIT(Y=wdh_Ht.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.Ht.blups.mlm =GAPIT(Y=Ht.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.Ht.blups.mlmm  =GAPIT(Y= Ht.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#Physiological Maturity

#GWAS.PM.blues.mlm  =GAPIT(Y=wdh_PM.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.PM.blues.mlmm  =GAPIT(Y=wdh_PM.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.PM.blups.mlm =GAPIT(Y=PM.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.PM.blups.mlmm  =GAPIT(Y= PM.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)










###end blues

####BLUPS####





#Field Phenotypes
GWAS.Scald.blups.farmcpu =GAPIT(Y= wdhscald.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="FarmCPU", Major.allele.zero = F)
wh_percent.blups
GWAS.wh_percent.blups.farmcpu=GAPIT(Y= wh_percent.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="FarmCPU", Major.allele.zero = F)



```
Save Rdata of GWA results
```{r}
getwd()
setwd("C:/Users/Karl/git/BarleyGermination-KK/data/GWA_results")
getwd()

save(PHS.blues.mlmm,PHS.blups.mlmm,file="GWA_WDH_phs.Rdata")
#TP1
save(GWAS.GETP1.blues.mlmm,GWAS.GETP1_5D.blues.mlmm,GWAS.GITP1.blues.mlmm,GWAS.GITP1_5D.blues.mlmm,GWAS.GETP1.blups.mlmm,GWAS.GETP1_5D.blups.mlmm,GWAS.GITP1.blups.mlmm,GWAS.GITP1_5D.blups.mlmm, file = "GWA_WDH_TP1.RData")
#TP2
save(GWAS.GETP2.blues.mlmm,GWAS.GETP2_5D.blues.mlmm,GWAS.GITP2.blues.mlmm,GWAS.GITP2_5D.blues.mlmm,GWAS.GETP2.blups.mlmm,GWAS.GETP2_5D.blups.mlmm,GWAS.GITP2.blups.mlmm,GWAS.GITP2_5D.blups.mlmm, file = "GWA_WDH_TP2.RData")
#TP3
save(GWAS.GETP3.blues.mlmm,GWAS.GETP3.blues_5D.mlmm,GWAS.GITP3.blues.mlmm,GWAS.GITP3_5D.blues.mlmm,GWAS.GETP3.blups.mlmm,GWAS.GETP3_5D.blups.mlmm,GWAS.GITP3.blups.mlmm,GWAS.GITP3_5D.blups.mlmm, file = "GWA_WDH_TP3.RData")
#TP4
save(GWAS.GETP4.blues.mlmm,GWAS.GETP4.blues_5D.mlmm,GWAS.GITP4.blues.mlmm,GWAS.GITP4_5D.blues.mlmm,GWAS.GETP4.blups.mlmm,GWAS.GETP4.blups_5D.mlmm,GWAS.GITP4.blups.mlmm,GWAS.GITP4_5D.blups.mlmm, file = "GWA_WDH_TP4.RData")
#field phenotypes
save(PHS.blues.mlmm,PHS.blups.mlmm,GWAS.wh.blues.mlmm,GWAS.wh.blups.mlmm,GWAS.scald.blues.mlmm,GWAS.scald.blups.mlmm,GWAS.HD.blups.mlmm,GWAS.HD.blues.mlmm,GWAS.Ht.blues.mlmm,GWAS.Ht.blups.mlmm,GWAS.PM.blues.mlmm,GWAS.PM.blups.mlmm, file="GWA_WDH_field_phenotypes.Rdata")
 

getwd()




```


