---
title: "GWA WDH Barley"
author: "Karl Kunze"
date: "1/7/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---
Workflow:
Winter_DH_data_consolidation.Rmd ->analysis_script.R->WDH_GenotypeData_Processing.Rmd->WDH_PhenotypeData_Model_Claculation.Rmd->GWA_WDH_Barley.Rmd->Haplotype_analysis.Rmd->GWA_results_interpretation.Rmd
Add_morex_v2_postions.. file is for aligning correct genome
read in phenotype files

```{r}
table(myGD20$Qsd1)
hist(myGD20$`JHI-Hv50k-2016-38563`)

```
genotype
Read in packages and functions
```{r}
#GAPIT functions

library(rrBLUP)
library(lme4)
library(readxl)
library(readr)
library(gplots)
library(LDheatmap)
library(genetics)
library(ape)
library(EMMREML)
library("scatterplot3d")
library(compiler)

#source code for GAPIT functions since GAPIT is not on CRAN
source("http://zzlab.net/GAPIT/gapit_functions.txt")

load(here("myGD20.Rdata"))
load(here("myGM20.Rdata"))
load(here("AlaT_markers.Rdata"))
colnames(AlaT_markers)[4]<-"Entry"

```
NEED TO RUN WDH_PhenotypeData_Model_Calculation for model objects until I make separate .RData object for them

```{r}
# GETP1
getwd()
local1<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP1"
#when setting these directories, using MLMM
setwd(local1)
getwd()
#setwd(dir)<-default working directory
GWAS.GETP1.blues.mlmm  =GAPIT(Y= GETP1.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#GWAS.GETP1.blues.mlm  =GAPIT(Y= GETP1.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
 
GWAS.GETP1_5D.blues.mlmm  =GAPIT(Y= GETP1_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP1.blues.mlmm  =GAPIT(Y= GITP1.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP1_5D.blues.mlmm  =GAPIT(Y= GITP1_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T,Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#blups

GWAS.GETP1.blups.mlmm  =GAPIT(Y= GETP1.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GETP1_5D.blups.mlmm  =GAPIT(Y= GETP1_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GITP1.blups.mlmm  =GAPIT(Y= GITP1.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP1_5D.blups.mlmm  =GAPIT(Y= GITP1_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#ended here for GAPIT 2/7 16:36
####TP2###
getwd()
local2<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP2"
#when setting these directories, using MLMM
setwd(local2)
getwd()
#blues
GWAS.GETP2.blues.mlmm  =GAPIT(Y= GETP2.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP2.blues.mlmm =GAPIT(Y= GITP2.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


#GWAS.GITP2.blues_5D.mlmm =GAPIT(Y= GITP2_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP2_5D.blues.mlmm  =GAPIT(Y= GETP2_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP2_5D.blues.mlmm  =GAPIT(Y= GITP2_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
##BLUPS
GWAS.GETP2.blups.mlmm  =GAPIT(Y= GETP2.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP2_5D.blups.mlmm  =GAPIT(Y= GETP2_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)# significant marker on 5H and 1H

GWAS.GITP2.blups.mlmm =GAPIT(Y= GITP2.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)#5H and 1H
GWAS.GITP2_5D.blups.mlmm  =GAPIT(Y= GITP2_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F) 


#GWAS.GITP2.blups_5D.mlmm =GAPIT(Y= GITP2_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)






####TP 3 #####
getwd()
local3<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP3"
#when setting these directories, using MLMM

#BLUES
setwd(local3)
getwd()

GWAS.GETP3.blues.mlmm  =GAPIT(Y= GETP3.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=F, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP3.blues.mlmm  =GAPIT(Y= GITP3.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP3.blues_5D.mlmm  =GAPIT(Y= GETP3_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP3_5D.blues.mlmm  =GAPIT(Y= GITP3_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#BLUPS
getwd()
local3.P<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP3/BLUPS"
setwd(local3.P)
GWAS.GETP3.blups.mlmm =GAPIT(Y= GETP3.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GETP3_5D.blups.mlmm =GAPIT(Y= GETP3_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP3_5D.blups.mlmm  =GAPIT(Y= GITP3_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.GITP3_5D.blups.mlmm=GAPIT(Y= GITP3_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

####TP 4#####
getwd()
local4<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP4"
#when setting these directories, using MLMM
setwd(local4)
getwd()
#blues
GWAS.GETP4.blues.mlmm  =GAPIT(Y= GETP4.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4.blues.mlmm  =GAPIT(Y= GITP4.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP4.blues_5D.mlmm  =GAPIT(Y= GETP4_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4_5D.blues.mlmm  =GAPIT(Y= GITP4_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#blups
getwd()
local4.P<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP4/BLUPS"
setwd(local4.P)

GWAS.GETP4.blups.mlmm  =GAPIT(Y= GETP4.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4.blups.mlmm  =GAPIT(Y= GITP4.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP4.blups_5D.mlmm  =GAPIT(Y= GETP4_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP4_5D.blups.mlmm  =GAPIT(Y= GITP4_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)





####TP 5#####
getwd()
local5<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP5"
#when setting these directories, using MLMM
setwd(local5)
getwd()
#blues
GWAS.GETP5.blues.mlmm  =GAPIT(Y= GETP5.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP5.blues.mlmm  =GAPIT(Y= GITP5.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP5.blues_5D.mlmm  =GAPIT(Y= GETP5_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP5_5D.blues.mlmm  =GAPIT(Y= GITP5_5D.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#blups
local5.P<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results/TP5/BLUPS"
setwd(local5.P)

GWAS.GETP5.blups.mlmm  =GAPIT(Y= GETP5.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP5.blups.mlmm  =GAPIT(Y= GITP5.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)


GWAS.GETP5.blups_5D.mlmm  =GAPIT(Y= GETP5_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

GWAS.GITP5_5D.blups.mlmm  =GAPIT(Y= GITP5_5D.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)










#############Field phenoypes#################

getwd()
localF<-"C:/Users/Karl/Google Drive/Grad School_/Field 2020/Germination_assays/GAPIT_results_field"

#when setting these directories, using MLMM
setwd(localF)
getwd()
#setwd(dir)<-default working directory
#phs
PHS.blues.mlmm  =GAPIT(Y= phs.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#PHS.blups.mlm  =GAPIT(Y= PHS.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
PHS.blups.mlmm  =GAPIT(Y= PHS.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)



#scald
#GWAS.scald.blues.mlm  =GAPIT(Y=wdh_scald.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.scald.blues.mlmm  =GAPIT(Y=wdh_scald.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.scald.blups.mlm =GAPIT(Y=wdhscald.blups, GD=myGD20, GM=myGM20, PCA.total=0,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
GWAS.scald.blups.mlmm  =GAPIT(Y= wdhscald.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#winter hardiness


#GWAS.wh.blues.mlm  =GAPIT(Y=wdh_wh.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.wh.blues.mlmm  =GAPIT(Y=wdh_wh.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.wh.blups.mlm =GAPIT(Y=wh_percent.blups , GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.wh.blups.mlmm  =GAPIT(Y=wh_percent.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)
#heading date

#GWAS.HD.blues.mlm  =GAPIT(Y=wdh_HD.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.HD.blues.mlmm  =GAPIT(Y=wdh_HD.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.HD.blups.mlm =GAPIT(Y=HD.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.HD.blups.mlmm  =GAPIT(Y= HD.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.HD.blups.farmcpu =GAPIT(Y= HD.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="FarmCPU", Major.allele.zero = F)




##height
#GWAS.Ht.blues.mlm  =GAPIT(Y=wdh_Ht.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.Ht.blues.mlmm  =GAPIT(Y=wdh_Ht.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.Ht.blups.mlm =GAPIT(Y=Ht.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.Ht.blups.mlmm  =GAPIT(Y= Ht.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#Physiological Maturity

#GWAS.PM.blues.mlm  =GAPIT(Y=wdh_PM.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.PM.blues.mlmm  =GAPIT(Y=wdh_PM.blues, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)

#GWAS.PM.blups.mlm =GAPIT(Y=PM.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLM", Major.allele.zero = F)
GWAS.PM.blups.mlmm  =GAPIT(Y= PM.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="MLMM", Major.allele.zero = F)










###end blues

####BLUPS####





#Field Phenotypes
GWAS.Scald.blups.farmcpu =GAPIT(Y= wdhscald.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="FarmCPU", Major.allele.zero = F)
wh_percent.blups
GWAS.wh_percent.blups.farmcpu=GAPIT(Y= wh_percent.blups, GD=myGD20, GM=myGM20, PCA.total=2,file.output=T, Geno.View.output=T, model="FarmCPU", Major.allele.zero = F)



```
Save Rdata of GWA results
```{r}
getwd()
setwd("C:/Users/Karl/git/BarleyGermination-KK/data/GWA_results")
getwd()

save(PHS.blues.mlmm,PHS.blups.mlmm,file="GWA_WDH_phs.Rdata")
#TP1
save(GWAS.GETP1.blues.mlmm,GWAS.GETP1_5D.blues.mlmm,GWAS.GITP1.blues.mlmm,GWAS.GITP1_5D.blues.mlmm,GWAS.GETP1.blups.mlmm,GWAS.GETP1_5D.blups.mlmm,GWAS.GITP1.blups.mlmm,GWAS.GITP1_5D.blups.mlmm, file = "GWA_WDH_TP1.RData")
#TP2
save(GWAS.GETP2.blues.mlmm,GWAS.GETP2_5D.blues.mlmm,GWAS.GITP2.blues.mlmm,GWAS.GITP2_5D.blues.mlmm,GWAS.GETP2.blups.mlmm,GWAS.GETP2_5D.blups.mlmm,GWAS.GITP2.blups.mlmm,GWAS.GITP2_5D.blups.mlmm, file = "GWA_WDH_TP2.RData")
#TP3
save(GWAS.GETP3.blues.mlmm,GWAS.GETP3.blues_5D.mlmm,GWAS.GITP3.blues.mlmm,GWAS.GITP3_5D.blues.mlmm,GWAS.GETP3.blups.mlmm,GWAS.GETP3_5D.blups.mlmm,GWAS.GITP3.blups.mlmm,GWAS.GITP3_5D.blups.mlmm, file = "GWA_WDH_TP3.RData")
#TP4
save(GWAS.GETP4.blues.mlmm,GWAS.GETP4.blues_5D.mlmm,GWAS.GITP4.blues.mlmm,GWAS.GITP4_5D.blues.mlmm,GWAS.GETP4.blups.mlmm,GWAS.GETP4.blups_5D.mlmm,GWAS.GITP4.blups.mlmm,GWAS.GITP4_5D.blups.mlmm, file = "GWA_WDH_TP4.RData")

#TP5
save(GWAS.GETP5.blues.mlmm,GWAS.GETP5.blues_5D.mlmm,GWAS.GITP5.blues.mlmm,GWAS.GITP5_5D.blues.mlmm,GWAS.GETP5.blups.mlmm,GWAS.GETP5.blups_5D.mlmm,GWAS.GITP5.blups.mlmm,GWAS.GITP5_5D.blups.mlmm, file = "GWA_WDH_TP5.RData")




#field phenotypes
save(PHS.blues.mlmm,PHS.blups.mlmm,GWAS.wh.blues.mlmm,GWAS.wh.blups.mlmm,GWAS.scald.blues.mlmm,GWAS.scald.blups.mlmm,GWAS.HD.blups.mlmm,GWAS.HD.blues.mlmm,GWAS.Ht.blues.mlmm,GWAS.Ht.blups.mlmm,GWAS.PM.blues.mlmm,GWAS.PM.blups.mlmm, file="GWA_WDH_field_phenotypes.Rdata")
 

getwd()




```


