---
title: "Interpreting WDH GWA results"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

Workflow:
Winter_DH_data_consolidation.Rmd ->analysis_script.R->WDH_GenotypeData_Processing.Rmd->GWA WDH Barley.Rmd->GWA_results_interpretation.Rmd

read in GWA results data
```{r}
dir1<-"C:/Users/Karl/git/BarleyGermination-KK"
setwd(dir1)
#the set wd is only for this chunk which should be fine once data is read in
library(here)

load(here("WMB DH 2020 all data for analysis.RData"))
load(here("data/GWA_results/GWA_WDH_field_phenotypes.Rdata"))
load(here("data/GWA_results/WDH_TP1.RData"))
load(here("data/GWA_results/WDH_TP2.RData"))
load(here("data/GWA_results/WDH_TP3.RData"))
load(here("data/GWA_results/WDH_TP4.RData"))
#please see GWA WDH Barley for model names
library(ggplot2)
```

```{r}





library(ggpubr)
library(tidyverse)
GETP1.fe<-mahatoutput(GWAS.GETP1.blues.mlmm)
mahatoutput<-function(df){


df<-df$GWAS[with(df$GWAS, order(Chromosome, Position)),]
#plot(x=df[,3], y=-log10(df[,4]), main = "Manhattan Plot",xlab="Genotype")
#abline( h= -log10(0.05/nrow(df)) , col = "red")
nCHR <- length(unique(df$Chromosome))
df[is.na(df$Chromosome),]$Chromosome<-8
df$BPcum <- NA
s <- 0
nbp <- c()
for (i in unique(df$Chromosome)){
  nbp[i] <- max(df[df$Chromosome == i,]$Position)
  df[df$Chromosome == i,"BPcum"] <- df[df$Chromosome == i,"Position"] + s
  s <- s + nbp[i]
}
axis.set <- df%>% 
  group_by(Chromosome) %>% 
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
ylim <- abs(floor(log10(min(df$P.value)))) + 2 
#str(axis.set)
sig <- 5e-8/13143
manoutput<-ggplot(df, aes(x =BPcum, y = -log10(P.value), 
                                 color = as.factor(Chromosome), size = -log10(0.05))) +
  geom_point(alpha = 0.75) +
  geom_hline(yintercept = -log10(0.05/nrow(df)), color = "red", linetype = "solid") + 
  scale_x_continuous(label = axis.set$Chromosome, breaks = axis.set$center) +
  scale_y_continuous(expand = c(0,0), limits = c(0, ylim)) +
  scale_color_manual(values = rep(c("#276FBF", "#183059"),4)) +
  scale_size_continuous(range = c(0,2)) +
  labs(x = NULL, 
       y = NULL) + 
  theme_minimal() +
  theme( 
    legend.position = "none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5)
  )
manoutput
}
mahatoutput_all<-function(df){


df<-df$GWAS[with(df$GWAS, order(Chromosome, Position)),]
#plot(x=df[,3], y=-log10(df[,4]), main = "Manhattan Plot",xlab="Genotype")
#abline( h= -log10(0.05/nrow(df)) , col = "red")
nCHR <- length(unique(df$Chromosome))
#df[is.na(df$Chromosome),]$Chromosome<-8
df$BPcum <- NA
s <- 0
nbp <- c()
for (i in unique(df$Chromosome)){
  nbp[i] <- max(df[df$Chromosome == i,]$Position)
  df[df$Chromosome == i,"BPcum"] <- df[df$Chromosome == i,"Position"] + s
  s <- s + nbp[i]
}
axis.set <- df%>% 
  group_by(Chromosome) %>% 
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
ylim <- abs(floor(log10(min(df$P.value)))) + 2 
#str(axis.set)
sig <- 5e-8/13143
manoutput<-ggplot(df, aes(x =BPcum, y = -log10(P.value), 
                                 color = as.factor(Chromosome), size = -log10(0.05))) +
  geom_point(alpha = 0.75) +
  geom_hline(yintercept = -log10(0.05/nrow(df)), color = "red", linetype = "solid") + 
  scale_x_continuous(label = axis.set$Chromosome, breaks = axis.set$center) +
  scale_y_continuous(expand = c(0,0), limits = c(0, ylim)) +
  scale_color_manual(values = rep(c("#276FBF", "#183059"),4)) +
  scale_size_continuous(range = c(0,2)) +
  labs(x = NULL, 
       y = NULL) + 
  theme_minimal() +
  theme( 
    legend.position = "none",
    panel.border = element_blank(),
    panel.grid.major.x = element_blank(),
    panel.grid.minor.x = element_blank(),
    axis.text.x = element_text(angle = 90, size = 8, vjust = 0.5)
  )
manoutput
}



GETP1.fe<-mahatoutput(GWAS.GETP1.blues.mlmm)
GETP1.re<-mahatoutput(GWAS.GETP1.blups.mlmm)
GITP1.fe<-mahatoutput(GWAS.GITP1.blues.mlmm)
GITP1.re<-mahatoutput(GWAS.GITP1.blups.mlmm)

ggarrange(GETP1.fe, GETP1.re, GITP1.fe, GITP1.re, 
          labels = c("GE 1 Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)
####
GETP1_5D.fe<-mahatoutput(GWAS.GETP1_5D.blues.mlmm)
GETP1_5D.re<-mahatoutput(GWAS.GETP1_5D.blups.mlmm)
GITP1_5D.fe<-mahatoutput(GWAS.GITP1_5D.blues.mlmm)
GITP1_5D.re<-mahatoutput(GWAS.GITP1_5D.blups.mlmm)

ggarrange(GETP1_5D.fe, GETP1_5D.re, GITP1_5D.fe, GITP1_5D.re, 
          labels = c("GE 5 Day Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)
## Timepoint 2 ###
GETP2.fe<-mahatoutput(GWAS.GETP2.blues.mlmm)
GETP2.re<-mahatoutput(GWAS.GETP2.blups.mlmm)
GITP2.fe<-mahatoutput_all(GWAS.GITP2.blues.mlmm) #something is wrong with this one
GITP2.re<-mahatoutput(GWAS.GITP2.blups.mlmm)
#GITP2.fe had to remove it
ggarrange(GETP2.fe, GETP2.re, GITP2.re, 
          labels = c("GE 2 Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)


GETP2_5D.fe<-mahatoutput(GWAS.GETP1_5D.blues.mlmm)
GETP2_5D.re<-mahatoutput(GWAS.GETP1_5D.blups.mlmm)
GITP2_5D.fe<-mahatoutput(GWAS.GITP1_5D.blues.mlmm)
GITP2_5D.re<-mahatoutput(GWAS.GITP1_5D.blups.mlmm)

ggarrange(GETP2_5D.fe, GETP2_5D.re, GITP2_5D.fe, GITP2_5D.re, 
          labels = c("GE 2 5 Day Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)
##
## Timepoint 3 ###
GETP3.fe<-mahatoutput(GWAS.GETP3.blues.mlmm)
GETP3.re<-mahatoutput(GWAS.GETP3.blups.mlmm)
GITP3.fe<-mahatoutput(GWAS.GITP3.blues.mlmm)
GITP3.re<-mahatoutput(GWAS.GITP3.blups.mlmm)
#GITP3.fe had to remove it
ggarrange(GETP3.fe, GETP3.re,GITP3.fe, GITP3.re, 
          labels = c("GE 3 Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)

str(GWAS.GETP3.blues_5D.mlmm)
GETP3_5D.fe<-mahatoutput(GWAS.GETP3.blues_5D.mlmm)
GETP3_5D.re<-mahatoutput(GWAS.GETP3_5D.blups.mlmm)
GITP3_5D.fe<-mahatoutput(GWAS.GITP3_5D.blues.mlmm)
GITP3_5D.re<-mahatoutput(GWAS.GITP3_5D.blups.mlmm)

ggarrange(GETP3_5D.fe, GETP3_5D.re, GITP3_5D.fe, GITP3_5D.re, 
          labels = c("GE 3 5 Day Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)

###
## Timepoint 4 ###
GETP4.fe<-mahatoutput(GWAS.GETP4.blues.mlmm)
GETP4.re<-mahatoutput(GWAS.GETP4.blups.mlmm)
GITP4.fe<-mahatoutput(GWAS.GITP4.blues.mlmm)
GITP4.re<-mahatoutput(GWAS.GITP4.blups.mlmm)
GETP4.fe
#GITP3.fe had to remove it
ggarrange(GETP4.fe, GETP4.re,GITP4.fe, GITP4.re, 
          labels = c("GE 4 Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)

str(GWAS.GETP3.blues_5D.mlmm)
GETP4_5D.fe<-mahatoutput(GWAS.GETP4.blues_5D.mlmm)
GETP4_5D.re<-mahatoutput(GWAS.GETP4.blups_5D.mlmm)

GITP4_5D.fe<-mahatoutput(GWAS.GITP3_5D.blues.mlmm)
GITP4_5D.re<-mahatoutput(GWAS.GITP3_5D.blups.mlmm)

ggarrange(GETP4_5D.fe, GETP4_5D.re, GITP4_5D.fe, GITP4_5D.re, 
          labels = c("GE 4 5 Day Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)

#Field
GETP4_5D.fe<-mahatoutput(GWAS.GETP4.blues_5D.mlmm)
GETP4_5D.re<-mahatoutput(GWAS.GETP4.blups_5D.mlmm)

GITP4_5D.fe<-mahatoutput(GWAS.GITP3_5D.blues.mlmm)
GITP4_5D.re<-mahatoutput(GWAS.GITP3_5D.blups.mlmm)

ggarrange(GETP4_5D.fe, GETP4_5D.re, GITP4_5D.fe, GITP4_5D.re, 
          labels = c("GE 4 5 Day Fx", "GE Rd", "GI Fx","GI Rd"),
          ncol = 2, nrow = 2)


```
