---
title: "GWAS_analysis"
author: "Karl Kunze"
date: "2023-03-18"
output: html_document
editor_options: 
  chunk_output_type: console
---
#packages
```{r}
library(readxl)
library(ggplot2)

library(ASRgwas)
library(dplyr)
library(ggh4x)
```
#data upload
```{r}


load("data/Phenotype_Data/AllDHBluesPerYear_first_step.Rdata")
load("data/Phenotype_Data/Heritability_first_step.Rdata")

load("data/Genotype_data/wmb_file.Rdata")
load("data/Genotype_data/numeric_raw.Rdata")
WinterGM=wmbGAPIT$GM;WinterGD=as.data.frame(wmbGAPIT$GD)
load("data/WMB_DH_Germination_data2020_2021.RData")#DH2020_2021
DH2020_2021<-DH2020_2021%>%filter(!Location=="Spring")

DH2020_2021[DH2020_2021$taxa%in%c("Charles","Endeavor"),]$Check<-"2"

```



```{r}
# AllDHBluesPerYear=AllDHBluesPerYear%>%mutate(family = mapvalues(substr(taxa,1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
#                             to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
#       
#                                                                 'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")))
#         
# AllDHBluesPerYear$family

 
```
## H2 visualization
```{r}
library(dplyr)
DH_H2%>%
   ggplot(aes(x = TP, y = ah2, fill = trait)) +geom_bar(stat = 'identity', position = 'dodge')+
  facet_wrap(vars(Year), ncol = 1)+theme_bw()+labs(title= 'Narrow sense heritability\nover time and datasets')

#other ideas to expand here, perhaps calculate narrow sense heritability, since we have the genotype info
#adding genotype data

table(AllDHBluesPerYear$taxa)
#AllDHBluesPerYear= rbind(DH2020Estimates, DH2021Estimates,DHCombined) %>% filter(type =='BLUE') %>%mutate(PM_date=as.double(PM_date),year=as.character(year))%>%ungroup()
#load("/home/karl/git/TimeSeriesGermination/WinterBarley/Analysis/AllDHBluesPerYear.RData")
```

## Genotype data

```{r}


# geno<-wmb_num$GD
# 
# geno=as.data.frame(apply(geno[-1],1, function(x) { 
#   x[which(x == 0)] = -1
#   x[which(x == 1)] <- 0
#   x[which(x == 2)] = 1
#   x
# }))
# geno
# geno$SNP<-rownames(geno)
# geno=geno%>%relocate(SNP,.before=1)


#geno<-WinterGM%>%full_join(geno,by="SNP")%>%arrange(Chromosome,Position)%>%dplyr::rename(marker=SNP,Chrom=Chromosome,bp=Position)%>%as.data.frame()
#rownames(geno)<-geno$marker
genoD_raw<-wmbGAPIT$GD
genoM_raw<-wmbGAPIT$GM
dim(genoD_raw)

wmb_num$GM[wmb_num$GM$SNP=="Qsd1",]

#its good checks out for reference version 3
wmb_num$GD["LIGHTNING","Qsd1"]
```
#winterGD and WinterGM
```{r}
library(dplyr)
#load('data/Genotype_data/myGM20_LDprune.Rdata')
#load('data/Genotype_data/myGD20_LDprune.Rdata')

WinterGM=wmbGAPIT$GM;WinterGD=as.data.frame(wmbGAPIT$GD)
#rm(myGM20_prune);rm(myGD20_prune)
WinterGM = WinterGM %>% arrange(Chromosome, Position)
#my taxa IDs have "-" instead of an underscore which I prefer
#taxa = gsub(pattern = '-',replacement = '_',taxa)
rownames(WinterGD)
WinterGD$taxa<-rownames(WinterGD)

str(WinterGD)
WinterGD=WinterGD %>%relocate(taxa,.before = 1)%>%

  mutate(taxa = gsub(pattern = '\\ ', replacement = '_',taxa)) %>% tibble::remove_rownames()



WinterGD

WinterGD[WinterGD$taxa=="ENDEAVOR",]$Qsd1<-2
WinterGD[WinterGD$taxa=="LIGHTNING",]$Qsd1
#WinterGD[WinterGD$taxa=="CHARLES",]$Qsd1<-2

dim(WinterGM)
colnames(WinterGD)[!colnames(WinterGD)%in%WinterGM$SNP]
dim(WinterGD)
# Make sure things are in the right order
# Sum should = 9269
dim(WinterGD)

WinterRelationship=wmbGAPIT$A

save(WinterRelationship,file="ms/Rdata/WinterRelationship.Rdata")
WinterPCA = eigen(WinterRelationship)
substr(WinterGD$taxa,1,3)
WinterPVEPCA = WinterPCA$values/sum(WinterPCA$values)
data.frame(ordinal = 1:10, PVE = WinterPVEPCA[1:10]) %>%plot(., xlab = 'PC', col = 'red') 

winterlinePCAvalues = WinterPCA$vectors %>% data.frame()%>% 
  
  mutate(family = plyr::mapvalues(substr(WinterGD$taxa,1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
                            to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
      
                                                                'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")),
         taxa = WinterGD$taxa,
         shapes = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), taxa, 'Lines'),
         size = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), 3, 2))


winterlinePCAvalues %>% mutate(shapes=gsub("(^|[[:space:]])([[:alpha:]])", "\\1\\U\\2", 
     tolower(winterlinePCAvalues$shapes),
     perl = TRUE),size=ifelse(size==2,3,4))%>%filter(family !=c('Check')) %>%
  ggplot(aes(x = X1, y = X2, color = family, shape = shapes)) + geom_point(aes(size = size))+theme_bw() +guides(shape=guide_legend(title = "Variety/Experimental Line"),size="none",color=guide_legend(title="Family/Population"))+xlab(as.character(paste0("PC1: ",round(WinterPVEPCA[1]*100,2),"%")))+ylab(as.character(paste0("PC2: ",round(WinterPVEPCA[2]*100,2),"%")))+ggtitle(lab="Principal component analysis plot of the \n NY winter malting barley double haploid population")+theme(plot.title = element_text(hjust = 0.5))+scale_x_continuous(expand = c(0, 0), limits = c(-0.125, 0.125))+scale_y_continuous(expand = c(0, 0), limits = c(-0.125, 0.125))




setwd(rprojroot::find_rstudio_root_file())

?xlab()
#Travis' Time series perspective

AllDHBluesPerYear  %>%  filter(type == 'BLUE') %>% mutate(headerFacet = 'Data source') %>%
  ggplot(aes(x = PM_date, y = value, group = taxa))+
  geom_smooth()+facet_nested(trait~headerFacet+year, scales = 'free')+
  geom_vline(xintercept = c(12,33,68), color = 'red')+
  labs(title = 'GE and GI BLUEs over time with differeing datasets')+theme_bw()

AllDHBluesPerYear %>% filter(type == 'BLUE' & trait == 'GI') %>% ggplot(aes(x = PM_date, y = value, group = taxa))+geom_line()+
  geom_vline(xintercept = c(12,33,68), color = 'red')
#filter(Family %nin% c('DH130910'))%>%

#png('plots/Time_series/Sup_BluesByFamilyQsd1AllYears.png', 2400, 1500, res =120)



#png('WinterBarley/WinterDHGerminationPaper/picsPNGforQsd1Effects_paper/BluesByFamilyQsd12020_2021.png', 1400, 800, res =120)
#2020

substr(AllDHBluesPerYear$taxa,1,3)
library(stringr)
str_split_fixed(AllDHBluesPerYear$taxa)

AllDHBluesPerYear<-AllDHBluesPerYear%>% mutate(family = plyr::mapvalues(substr(AllDHBluesPerYear$taxa,1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
                            to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
      
                                                                'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")),
         taxa = AllDHBluesPerYear$taxa,
         shapes = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), taxa, 'Lines'),
         size = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), 3, 2))


    

"JHI_Hv50k_2016_2018956"
WinterGD
colnames(WinterGD[,c('taxa',"JHI_Hv50k_2016_8507")])

AllDHBluesPerYear%>%
 filter(type == 'BLUE') %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Nondormant','Dormant')) %>% plyr::join(WinterGD[,c('taxa','JHI_Hv50k_2016_8507')]) %>% filter(JHI_Hv50k_2016_8507!= 1) %>% mutate(JHI_Hv50k_2016_8507= ifelse(JHI_Hv50k_2016_8507==2,'minor','ref'))%>%
  filter(family %nin% c('Lightning')) %>% filter(year %in% c('2020'))  %>%
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x = TP, y = value, fill = Qsd1))+
 
  geom_boxplot()+facet_nested(trait~family, scales = 'free')+
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination Rates for 2020 Year"))+
theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))


library(Hmisc)
#2021
AllDHBluesPerYear
AllDHBluesPerYear %>%  filter(type == 'BLUE') %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(WinterGD[,c('taxa','JHI_Hv50k_2016_8507',"Qsd1")]) %>% filter(JHI_Hv50k_2016_8507!= 1,Qsd1!=1) %>% mutate(JHI_Hv50k_2016_8507= ifelse(JHI_Hv50k_2016_8507==2,'Alt','Ref'),Qsd1=ifelse(JHI_Hv50k_2016_8507==2,'Nondormant','Dormant')) %>%
  filter(family %nin% c('Lightning')) %>% filter(year %in% c('2021'))  %>%
  filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05))%>%
  ggplot(aes(x = TP, y = value, fill = c(JHI_Hv50k_2016_8507)))+  
  geom_boxplot()+facet_nested(trait~Qsd1+family, scales = 'free')+ 
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination Rates for 2021 Year"))+
  theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))
#2020/2021
AllDHBluesPerYear %>%  filter(type == 'BLUE') %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Nondormant','Dormant')) %>%
  filter(family %nin% c('Lightning')) %>% filter(year %in% c('2020/2021'))  %>%
  filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x = TP, y = value, fill = Qsd1))+  
  geom_boxplot()+facet_nested(trait~family, scales = 'free_y')+
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination Rates for Combined 2020/2021 Years"))+
  theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))

dev.off()
#####


#library(GAPIT3)
DF_OperationsV3 = function(df){
  df = df %>% arrange(P.value)  %>% mutate(logPercentileQQplot = -log10(c(1:length(df$SNP))/length(df$SNP)),
                                           rank = c(1:length(df$SNP))) %>% arrange(Chr, as.numeric(Pos)) %>%
    mutate(log10PVal = -log10(P.value),ordinal = c(1:length(df$SNP)))
  return(df)
}
#I am assuming this is to help make manhattan plots

GWA_MLMM_fortidyR = function(df, groupvars) {
  #fg<- DH2020_1 %>% rbind(DH2021_1, DHcomb_1) %>%  filter(type == 'BLUE') %>%
  # ungroup()%>%filter(year=="2020",TP=="TP1",trait=="GI")

  
  
  # %>% group_by(year, TP, trait)
  
  GWAS = GAPIT(Y = df%>% dplyr::select(taxa, value) %>% as.data.frame(),GD=WinterGD, GM=WinterGM,PCA.total = 2,
               Geno.View.output=F, model="MLMM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05)
    GWAS 
  Out = DF_OperationsV3(GWAS$GWAS) %>% arrange(P.value) %>% slice_head(n=1000)
  return(Out)
}
#GWA MLMM per timepoint,
# GWA_MLMM_fortidyR.perFamily = function(df, groupvars) {
#   WinterGD_sub = WinterGD %>% filter(taxa %in% df$taxa)
#   Out = tryCatch(
#     {GWAS = suppressWarnings(GAPIT(Y = df %>%ungroup() %>% select(taxa, value) %>% as.data.frame(),GD=WinterGD_sub, GM=WinterGM,PCA.total = 0,
#                                    Geno.View.output=F, model="MLMM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05))
#     DF_OperationsV3(GWAS$GWAS) %>%
#       arrange(P.value) %>% slice_head(n = 100)
#     }, error = function(e){data.frame() } )
#   return(Out)
# }
#per family, I'm not going to worry about this one for now
# GWA_MLM_fortidyR = function(df, groupvars) {
#   GWAS = GAPIT(Y = df %>% ungroup() %>% select(taxa, value) %>% as.data.frame(),GD=WinterGD, GM=WinterGM,PCA.total = 2,
#                Geno.View.output=F, model="MLM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05)
#   Out = (GWAS$GWAS) %>% arrange(P.value)  %>%
#     mutate(logPercentileQQplot = -log10(c(1:length(GWAS$GWAS$SNP))/length(GWAS$GWAS$SNP)),
#            rank = c(1:length(GWAS$GWAS$SNP))) %>% arrange(Chromosome, 'Position') %>%
#     mutate(log10PVal = -log10(P.value), ordinal = c(1:length(GWAS$GWAS$SNP)))%>% 
#     arrange(P.value) %>% slice_head(n = 1000)
#   return(Out)
# }
#not going to worry about this one either for germination traits at least
GWA_MLM_fortidyR.perFamily = function(df, groupvars) {
  WinterGD_sub = WinterGD %>% filter(taxa %in% df$taxa)
   Out = tryCatch(
    {GWAS = suppressWarnings(GAPIT(Y = df %>% ungroup() %>% select(taxa, value) %>% as.data.frame(),GD=WinterGD_sub, GM=WinterGM,PCA.total = 0,
                                    Geno.View.output=F, model="MLM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05))
    (GWAS$GWAS) %>% arrange(P.value)  %>%
      mutate(logPercentileQQplot = -log10(c(1:length(GWAS$GWAS$SNP))/length(GWAS$GWAS$SNP)),
             rank = c(1:length(GWAS$GWAS$SNP))) %>%  arrange(Chromosome, 'Position') %>%
       mutate(log10PVal = -log10(P.value), ordinal = c(1:length(GWAS$GWAS$SNP)))%>% 
       arrange(P.value) %>% filter(P.value < 0.01)
    }, error = function(e){data.frame() } )
   return(Out)
 }

# Add_DH130910_familyStructure = function(df3, groupvar){
#   DH130910_rows = df3 %>% filter(taxa == 'Lightning') %>% select(!family)
#   df3= df3 %>% filter(taxa != 'Lightning') %>%
#     rbind(DH130910_rows %>% mutate(Family='Flavia/DH130910'),
#           DH130910_rows %>% mutate(Family='SY_Tepee/DH130910'),
#           DH130910_rows %>% mutate(Family='Scala/DH130910'),
#           DH130910_rows %>% mutate(Family='Wintmalt/DH130910')
#     ) %>%
#     filter(Family %in% c('Flavia/DH130910','SY_Tepee/DH130910','Scala/DH130910','Wintmalt/DH130910'))
#   return(df3)
# }


#Per time point for MLMM
##takes some time
#SAVE THE RESULTS so that we dont have to run again
print("MLMM GWA model ahead")

library(GAPIT3)

 WinterPerTPGWAS = AllDHBluesPerYear %>%  filter(type == 'BLUE') %>%
  ungroup() %>% group_by(year, TP, trait)%>%group_modify(GWA_MLMM_fortidyR)
save(WinterPerTPGWAS ,file="data/GWA_results/WinterPerTPGWAS.Rdata")

```
```{r}
# 
# geno<-WinterGD
# rownames(geno)<-WinterGD$taxa
# geno=as.data.frame(t(geno[-1]))
# geno$SNP<-rownames(geno)
# geno=geno%>%relocate(SNP,.before=1)
# geno<-WinterGM%>%full_join(geno,by="SNP")%>%arrange(Chromosome,Position)%>%dplyr::rename(marker=SNP,chrom=Chromosome,bp=Position)
# geno

```



#asreml GWAS example
```{r}

library("ASRgenomics")

data(geno.pine655)
geno.pine655[1:5, 1:5]
M_filter <- qc.filtering(M = geno.pine655, base = FALSE, ref = NULL,
marker.callrate = 0.2, ind.callrate = 0.2, maf = 0.05, heterozygosity = 0.95,
Fis = 1, impute = FALSE, na.string = "-9", plots = TRUE)
M_filter$M.clean[1:5, 1:5]
dim(M_filter$M.clean)
G <- G.matrix(M = M_filter$M.clean, method = "VanRaden", na.string = NA)$G
#use Yang method for inbreds
data(ped.pine)
head(ped.pine)
tail(ped.pine)
A <- AGHmatrix::Amatrix(data = ped.pine)
A
A[601:605, 601:605]
check_G <- kinship.diagnostics(K = G, diagonal.thr.small = 0.8,
diagonal.thr.large = 1.2, duplicate.thr = 0.95)
check_G$list.diagonal
check_G$list.duplicate
check_G$plot.diag
check_G$plot.offdiag

####
G2A <- match.G2A(A = A, G = G, clean = TRUE, ord = TRUE, mism = TRUE,
RMdiff = TRUE)
G2A$Aclean
G2A$Aclean[343:347, 343:347]
G2A$Aclean[343:347, 343:347]
G2A$plotG2A
#clean up
G_align <- G.tuneup(G = G2A$Gclean, A = G2A$Aclean, align = TRUE)$Gb

Ga2A <- match.G2A(A = A, G = G_align, clean = TRUE, ord = TRUE,
mism = TRUE, RMdiff = TRUE)
dim(Ga2A$RM[Ga2A$RM$absdiff > 0.2, ])
check_G_align <- kinship.diagnostics(K = G_align)
check_G_align$plot.diag
check_G_align$plot.offdiag
#kinship
kinship.heatmap(K = G_align, dendrogram = TRUE, row.label = FALSE,
col.label = FALSE)

pca_pine <- kinship.pca(K = G_align, ncp = 15)
pca_pine$eigenvalues
pca_pine$plot.scree
pca_pine$plot.pca
##example two





library(ASRgenomics)
head(pheno.salmon)
head(geno.salmon)
pre.salmonA <- pre.gwas(pheno.data = pheno.salmon,
indiv = "ID", resp = "amoebic_load", geno.data = geno.salmon,
Q.method = "K", maf = 0.05, marker.callrate = 0.2,
ind.callrate = 0.2, method = "VanRaden", heterozygosity = 0.8,
Fis = 0.98, impute = TRUE)

pre.salmonD <- pre.gwas(pheno.data = pheno.salmon,
indiv = "ID", resp = "amoebic_load", geno.data = geno.salmon,
Q.method = "K", maf = 0.05, marker.callrate = 0.2,
ind.callrate = 0.2, method = "Su", heterozygosity = 0.8,
Fis = 0.98, impute = TRUE)

pheno.AD<-pre.salmonD$pheno.data
pheno.AD$IDA <- pheno.AD$ID
pheno.AD$IDD <- pheno.AD$ID

Ma <- round(pre.salmonA$geno.data, 0)
Md <- round(pre.salmonD$geno.data, 0)
colnames(Md) <- paste("D", colnames(Md), sep = "_")
Md[Md == 2] <- 0
dim(Ma)
dim(Md)
Mad <- cbind(Ma, Md)


gwasS <- gwas.asreml(pheno.data = pheno.AD, resp = "amoebic_load",
gen = c("IDA", "IDD"), Kinv = list(IDA = pre.salmonA$Kinv$ID,
IDD = pre.salmonD$Kinv$ID), Q = pre.salmonA$Q,
npc = 7, geno.data = Mad, map.data = NULL, pvalue.thr = 5e-04,
bonferroni = FALSE, P3D = TRUE)

summary(gwasS$mod)$varcomp
gwasS$heritability
gwasS$gwas.sel




```
#asreml GWAS setup
```{r}
library("ASRgwas")


genoD<-as.data.frame(genoD_raw)

genoD["TEPEE",]$Qsd1


genoD[genoD$Qsd1==1,]$Qsd1<-2
genoD=genoD%>%mutate(Qsd1=ifelse(Qsd1==0,2,0))
genoD["ENDEAVOR",]$Qsd1<-2
genoD["FLAVIA",]$Qsd1<-2
genoM_raw
genoM<-genoM_raw%>%dplyr::rename(marker=SNP,chrom="Chromosome","pos"="Position")%>%filter(marker%in%colnames(genoD))
genoM
rownames(dim)%in%c("ENDEAVOR")
library(dplyr)


#marker effects
marker_ef<-genoD%>%as.data.frame()%>%dplyr::select("Qsd1","JHI_Hv50k_2016_308762")%>%tibble::rownames_to_column()%>%rename(GID=rowname)%>%mutate(Qsd1=as.factor(Qsd1),JHI_Hv50k_2016_308762=as.factor(JHI_Hv50k_2016_308762))
marker_ef




df_begin<-DH2020_2021%>%mutate(GID=toupper(taxa))%>%filter(!Location=="Spring")%>%left_join(marker_ef,by="GID")%>%mutate(type="base_data_single_step")
table(df_begin$GID)
###
df_begin<-df_begin%>%mutate(replication=as.factor(replication),Location=as.factor(Location),Check=as.factor(Check),GI_0=GI/GE)

#Single Timepoint ASRgwas
FDR_thresholds<-function(df_ps){
df_ps<-df_ps$gwas.all$p.value
p<-df_ps%>%as.data.frame()%>%rename(p.val=1)%>%mutate(pdj=p.adjust(p.val,"fdr"))
if(length( which( p$pdj<=0.05 )>0)){
fdr_threshold<-max(p[p$pdj<=0.05,]$pdj,na.rm=TRUE)
p_thershFDR<-max(p[which(round(p$pdj,4)<=round(fdr_threshold,4)),]$p.val)
}else{  fdr_threshold=5e-5
p_thershFDR=5e-5
}
cv<-data.frame(p_valueFR=p_thershFDR,FDR=fdr_threshold)
return(cv)
}
```
#going to make some visualizations here
```{r}
library(tidyr)
hist(df_begin$GI_0)
df_begin$Family<-gsub("DH130910","Lightning",df_begin$Family)
df_begin_long<-df_begin%>%mutate(GI_0=ifelse(is.nan(GI_0),NA,GI_0))%>%pivot_longer(names_to="trait",cols=c("GE","GI"))%>%rename(year=Year)
df2<-df_begin_long%>%filter(PM_date%in%c("5","19","47","96","152"))

df2$year<-"2020/2021"

df_begin_long%>%rbind(df2)%>% mutate(headerFacet = 'Dataset Year') %>%
  ggplot(aes(x = PM_date, y = value, group = GID))+
  geom_smooth(na.rm = TRUE,se=FALSE,linewidth=0.2,colour="black")+facet_nested(trait~headerFacet+year, scales = 'free')+
  geom_vline(xintercept = c(12,33,68), color = 'red')+
  labs(title = 'Germination Energy and Index over time by individual genotype')+theme_bw()
```

```{r}

df_begin_long%>%rbind(df2) %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Dormant','Nondormant')) %>% plyr::join(WinterGD[,c('taxa','JHI_Hv50k_2016_8507')]) %>% filter(JHI_Hv50k_2016_8507!= 1) %>% mutate(JHI_Hv50k_2016_8507= ifelse(JHI_Hv50k_2016_8507==2,'minor','ref'))%>%
  filter(Family %nin% c('Lightning'))   %>%
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x = PM_date, y = value, fill = Qsd1))+
 
  geom_boxplot(outlier.shape = NA)+facet_nested(trait~Family+year, scales = 'free')+
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination rates by grouping of AlaAT"))+
theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))

```
#Single Timepoint GWAS

```{r}
GI_all<-df_begin
hist(df_begin$phs)
GI_2020<-GI_all%>%filter(Year=="2020")%>%droplevels()
table(GI_2020$PM_date)
grep("L",colnames(myGD20_prune))

```
#PHS
```{r}
#we need to include
phs<-DH2020_2021%>%mutate(GID=toupper(taxa))%>%filter(!Location=="Spring")%>%left_join(marker_ef,by="GID")%>%mutate(type="base_data_single_step")%>%select(GID,Location,SourcePLOT,Row,Check,Family,Year,phs)%>%left_join(marker_ef,by="GID")

simple<-phs%>%unique()
simple

genoM

gd=as.matrix(genoD);genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE

dim(gd)
print(paste0("data input: ",simple$type%>%unique()))

#should do Genomic prediction for the missing ones


simple$ND<-as.character(simple$Check)
simple[simple$GID=="ENDEAVOR",]$phs<-8
simple$ND
pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "phs", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
fixed="Check"
random=NULL

resid=NULL
pre.aprAA$pheno.data$phs
hist(pre.aprAA$pheno.data$phs)
gwasSTP.phs<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "phs",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = NULL, npc = 4, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data,bonferroni = TRUE,fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')
library(asreml)
wald.asreml(gwasSTP.phs$mod)
gwasSTP.phs$heritability
summary(gwasSTP.phs$mod)$varcomp
FDR_thresholds(gwasSTP)

gwasSTP.phs$heritability
gwasSTP.phs$gwas.sel
plot(gwasSTP$mod)
qq.plot(gwas.table = gwasSTP.phs$gwas.all)
manhattan.plot(gwas.table = gwasSTP.phs$gwas.all, pvalue.thr = FDR_thresholds(gwasSTP)[1,1],
point.size = 1.2)
#change here
FDR_thresholds(gwasSTP)[1,1]
tr
GWAS_phs<-gwasSTP.phs

GWAS_phs$trait<-paste0("PHS","-","2020/2021","-PM_","3")
GWAS_phs$trait
GWAS_phs$FDR_rate<-FDR_thresholds(gwasSTP.phs)
save(GWAS_phs,file = "data/Genotype_data/ASRgwas_results/PHS_GWAS_results.Rdata")


#JHI-Hv50k-2016-367342 MKK3 QTL

dim<-genoD%>%as.matrix()


```

## Germination Index 2020



GI_all

```{r}
## 2020

GI_2020<-GI_all%>%filter(Year=="2020")%>%droplevels()
table(GI_2020$PM_date)
grep("L",colnames(myGD20_prune))

#JHI-Hv50k-2016-367342 MKK3 QTL

dim<-genoD%>%as.matrix()
```

### PM 5
```{r}
simple<-GI_2020%>%filter(PM_date=="5")
simple$PM_date


dim[1:5,1:5]
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE

genoM
print(paste0("data input: ",simple$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones


###
hist(simple$GI)
simple$ND<-as.character(simple$Check)
simple$ND
pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND==2), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
fixed=c("Location","replication")
random=NULL

resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


FDR_thresholds(gwasSTP)

t3<-simple[,c("GI","Year","PM_date")]


tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = FDR_thresholds(gwasSTP)[1,1],
point.size = 1.2)
#change here

GI2020_PM5<-gwasSTP
GI2020_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM5$FDR_rate<-FDR_thresholds(gwasSTP)
GI2020_PM5$FDR_rate
table(GI_2020$PM_date)
```
### 2020 GI 19
```{r}
simple<-GI_2020%>%filter(PM_date=="19")
simple$PM_date%>%unique()


simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)

print(paste0("data input: ",simple$type%>%unique()))


###
table(simple$GID)

pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=c("Location","Check");resid="Location";maf_lim=0.05;p.val=5e-05;

##
pre.aprAA$pheno.data$Row
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 5e-05,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
FDR_thresholds(gwasSTP)[1,1]
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp

t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
t3

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = FDR_thresholds(gwasSTP)[1,1],
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN 
#change here
GI2020_PM19<-gwasSTP
tpm

GI2020_PM19$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM19$FDR_rate<-FDR_thresholds(gwasSTP)
GI_2020$PM_date
GI2020_PM19$trait
```
### GI  47
```{r}
simple<-GI_2020%>%filter(PM_date=="47")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GI,GE)



###
pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=c("Location");resid=NULL;maf_lim=0.05;p.val=5e-05

##

gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 5e-05,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
##
GI2020_PM19$FDR_rate


hist(gwasSTP$mod$residuals)
library(asreml)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp


t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr =5.054344e-06,
point.size = 1.2)

#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]
cor<-as.data.frame(pre.aprAA$map.data)%>%arrange(chrom,pos)%>%filter(marker%in%c(sel.markers))
cor
cor$pos[25]-cor$pos[1]
length(cor$pos)
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,2,18,21)],
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN

#change here
tpm
GI2020_PM47<-gwasSTP
paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM47$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM47$FDR_rate<-FDR_thresholds(gwasSTP)
```
### 96
```{r}
simple<-GI_2020%>%filter(PM_date=="96")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,Day2Germ,Day3Germ,Day4Germ,Day5Germ,scald,Qsd1,GI,GE,GI_0)



###
pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI_0", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=c("replication");resid="replication";maf_lim=0.05;p.val=1e-05


gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=NULL,randomf = NULL,workspace='2Gb')
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp


t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,length(sel.markers))],
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN
#change here
tpm
paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM96<-gwasSTP

GI2020_PM96$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM96$trait
table(GI_2020$PM_date)
```
### 2020 GI 152
```{r}
simple<-GI_2020%>%filter(PM_date=="152")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,Day2Germ,Day3Germ,Day4Germ,Day5Germ,scald,Qsd1,GI,GE,GI_0)



###
pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=NULL;resid=NULL;maf_lim=0.05;p.val=5e-05


gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=NULL,randomf = NULL,workspace='2Gb')
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp


t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,length(sel.markers))],
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN
#change here
tpm
paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM152<-gwasSTP

GI2020_PM152$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM152$trait
table(GI_2020$PM_date)
GI2020_PM5
GI_2020_GWAS=list(GI2020_PM5,GI2020_PM19,GI2020_PM47,GI2020_PM96,GI2020_PM152)
```
### Germination Energy 2020 
```{r}

GE_all<-df_begin


GE_all
## 2020

GE_2020<-GE_all%>%filter(Year=="2020")%>%droplevels()
table(GE_2020$PM_date)

```

### PM 5
```{r}
simple<-GE_2020%>%filter(PM_date=="5")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones


###
hist(simple$GE)
simple$ND<-as.character(simple$Check)
simple$ND
simple
pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="Location"
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#change here
tr
paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM5<-gwasSTP
GE2020_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM5$trait
table(GE_2020$PM_date)
```
### 2020 GE 19
```{r}



simple<-GE_2020%>%filter(PM_date=="19")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones


###
hist(simple$GE)
simple$ND<-as.character(simple$Check)
simple$ND
simple
pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="Location"
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-5,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[, sel.markers[c(1,6)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN 

#change here
tr
tpm
ty
GE2020_PM19<-gwasSTP
GE2020_PM19$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM19$trait
table(GE_2020$PM_date)
```
### GE 47
```{r}
simple<-GE_2020%>%filter(PM_date=="47")

simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,GE5,GI5)%>%mutate(GI5_0=GI5/GE5)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))



###
hist(simple$GE)
simple$ND<-as.character(simple$Check)


pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE_tr", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed=c("Location:replication","Location","replication")
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM47<-gwasSTP
GE2020_PM47$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM47$trait
table(GE_2020$PM_date)
```
### GE 96
```{r}


simple<-GE_2020%>%filter(PM_date=="96")

simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,GE5,GI5)%>%mutate(GI5_0=GI5/GE5)



###
hist(simple$GE)
simple$ND<-as.character(simple$Check)


pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="Location"
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM96<-gwasSTP
GE2020_PM96$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM96$trait
table(GE_2020$PM_date)
### GE 2020 152


simple<-GE_2020%>%filter(PM_date=="152")

simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,GE5,GI5)%>%mutate(GI5_0=GI5/GE5)



###
hist(simple$GE)
simple$ND<-as.character(simple$Check)


pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed=c("Location")
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM152<-gwasSTP
GE2020_PM152$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM152$trait
table(GE_2020$PM_date)

### GE 2020 list

GE_2020_GWAS<-list(GE2020_PM5,GE2020_PM19,GE2020_PM47,GE2020_PM96,GE2020_PM152)
GE2020
```
#SAVING GE2020 and GI2020 Single Timepoint GWAS
```{r}
GI_2020

save(GI_2020_GWAS,file="data/Genotype_data/ASRgwas_results/Single_TP_GI_2020_results.Rdata")
save(GE_2020_GWAS,file="data/Genotype_data/ASRgwas_results/Single_TP_GE_2020_results.Rdata")
```
# 2021 Single Timepoint
## Germination Index
```{r}
GI_all<-df_begin


GI_all
## 2020

GI_2021<-GI_all%>%filter(Year=="2021")%>%droplevels()
table(GI_2021$PM_date)


### 2021 GI PM 5
simple<-GI_2021%>%filter(PM_date=="5")
simple$Maturity_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,Maturity_date)

hist(simple$GI)
simple$ND<-as.character(simple$Check)
simple$GI_l<-simple$GI^1
hist(simple$GI_l)
simple
pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
pre.aprAA$pheno.data$Day1Germ
fixed=NULL
rand=NULL
resid="Location"
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-5,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-05,
point.size = 1.2)
#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM5<-gwasSTP
GI2021_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM5$trait
table(GI_2021$PM_date)
```
### GI PM 12
```{R}
simple<-GI_2021%>%filter(PM_date=="12")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=TRUE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,JHI_Hv50k_2016_308762,GE,GI,GI_0)

hist(simple$GE)
simple$ND<-as.character(simple$Check)

pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

fixed="Location"
random=NULL
resid="Location"
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='1Gb')



summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

gwasSTP
qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-5,
point.size = 1.2)
#significant marker
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers, drop = FALSE]
sel.markers 
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN
#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM12<-gwasSTP
GI2021_PM12$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM12$trait
table(GI_2021$PM_date)
```
### 2021 GI 2019
```{r}
simple<-GI_2021%>%filter(PM_date=="19")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=TRUE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)

hist(simple$GI)
simple$ND<-as.character(simple$Check)

pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.20)
##

fixed=NULL
random=NULL
resid="replication"

gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-5,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

gwasSTP
qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr =1e-5,
point.size = 1.2)
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers, drop = FALSE]
sel.markers 
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN
#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM19<-gwasSTP
GI2021_PM19$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM19$trait
table(GI_2021$PM_date)
```
### 2021 GI 33 PM
```{r}


simple<-GI_2021%>%filter(PM_date=="33")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=TRUE


print(paste0("data input: ",df$type%>%unique()))
simple
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762)

hist(simple$GI)
simple$ND<-as.character(simple$Check)

pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

fixed=c("replication")
random=NULL

resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-5,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-5,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,6)], drop = FALSE]
sel.markers 
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM33
GI2021_PM33<-gwasSTP
GI2021_PM33$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM33$trait
table(GI_2021$PM_date)
```

### GI 2021 47
```{r}
simple<-GI_2021%>%filter(PM_date=="47")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=TRUE


print(paste0("data input: ",GI_2021$type%>%unique()))
simple
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762)

hist(simple$GI)
simple$ND<-as.character(simple$Check)

pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

fixed=NULL
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = NULL, npc = 5, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-05,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-5,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM47
GI2021_PM47<-gwasSTP
GI2021_PM47$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM47$trait
table(GI_2021$PM_date)
```
### 2021 GI68 days
```{r}
simple<-GI_2021%>%filter(PM_date=="68")

gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",df$type%>%unique()))
simple
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)

hist(simple$GI)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)


pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
pre.aprAA$pheno.data
random=NULL
fixed=c("Location")
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI_0",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-5,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers, drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM68
GI2021_PM68<-gwasSTP
GI2021_PM68$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM68$trait
table(GI_2021$PM_date)
```
### 2012 GI 96 PM
```{R}
simple<-GI_2021%>%filter(PM_date=="96")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))
#View(simple)
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,Maturity_date)%>%mutate(GI_lam=GI^1.2)
#View(simple)
hist(simple$GI_lam)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)
simple$GI_0
pre.aprAA <- pre.gwas(pheno.data=simple%>%droplevels(), indiv = "GID",
resp = "GI_0", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
#View(pre.aprAA$pheno.data)
random=NULL
fixed=c("Location:Row","replication")
resid=c("Row")

gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI_0",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-05,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 0.05/13409,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM96
GI2021_PM96<-gwasSTP
GI2021_PM96$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM96$trait


```
### 2021 GI 152
```{r}
simple<-GI_2021%>%filter(PM_date=="152")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=TRUE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))
simple
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)

hist(simple$GI)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI_0", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

random=NULL
fixed="Location"
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI_0",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 5e-05,
bonferroni =FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI_0";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-5,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM152
GI2021_PM152<-gwasSTP
GI2021_PM152$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2021_PM152$trait

```

## Germination Energy 2021
```{r}
GE_all<-df_begin
GE_2021<-GE_all%>%filter(Year=="2021")
## 2021 GE PM 5
simple<-GE_2021%>%filter(PM_date=="5")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=NULL
resid="Location"
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr =0.05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')

hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
library(asreml)

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 0.05/13409,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM5
GE2021_PM5<-gwasSTP
GE2021_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM5$trait
table(GE_2021$PM_date)
```

### 2021 GE 12
```{r}
simple<-GE_2021%>%filter(PM_date=="12")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)

hist(simple$GE)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

random=NULL
fixed=NULL
resid=NULL
covar=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",cov = covar,
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = 1e-5,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
#library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 1e-5,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM12
GE2021_PM12<-gwasSTP
GE2021_PM12$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM12$trait
table(GE_2021$PM_date)
```
### GE 2021 19
```{r}

simple<-GE_2021%>%filter(PM_date=="19")
simple$AssayNumber
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=1e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent,AssayNumber)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))%>%mutate(groups=as.factor(cut(simple$AssayNumber,8)))
hist(simple$GE_bi)
str(simple$AssayNumber)
simple$groups

hist(simple$GE)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
pre.aprAA $pheno.data
rand=NULL
fixed=c("Qsd1","JHI_Hv50k_2016_308762")
resid=NULL
####warning binomial will crash the computer
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, bonferroni = TRUE,
fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
#library(asreml)
wald.asreml(gwasSTP$mod)


t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all,pvalue.thr = 3.82528125009163e-06,point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 


gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM19
GE2021_PM19<-gwasSTP
GE2021_PM19$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM19$trait
table(GE_2021$PM_date)



```
### GE 2021 33

```{r}


simple<-GE_2021%>%filter(PM_date=="33")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=TRUE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))
hist(simple$GE_bi)

hist(simple$GE)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

rand=NULL
fixed=c("Location:Row")
resid=NULL
####warning binomial will crash the computer
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  NULL, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, bonferroni = FALSE,
fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)

###*****There is some error in Location:Row"
#library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)

manhattan.plot(gwas.table = gwasSTP$gwas.all,pvalue.thr = 0.05/dim(pre.aprAA$geno.data)[2],point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 


gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM33
GE2021_PM33<-gwasSTP
GE2021_PM33$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM33$trait
table(GE_2021$PM_date)




```


```{r}
# qq.plot(gwas.table = gwasSTP$gwas.all)
# manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
# point.size = 1.2)
# summary(gwasSTP$mod)$varcomp
# summary(gwasSTP$mod)
# gwasMTP$gwas.sel


# library(asreml)
# ?asreml()
# d3<-pre.aprA$pheno.data
# d3$PM_date_t<-as.numeric(as.character(d3$PM_date))
# d3$PM_date
# model<-asreml(
#   fixed = GI~ 1,
#   random = ~GID+Year:dev(PM_date_t),
#   sparse = ~NULL,
#   residual = ~NULL,data=d3)
# summary(model)$varcomp
###

```
###GE 2021 47

```{r}


simple<-GE_2021%>%filter(PM_date=="47")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=TRUE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))
#hist(simple$GE)

hist(simple$GE_tr)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)

names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
hist(pre.aprAA $pheno.data$GE)

rand=NULL
fixed=NULL
resid=NULL
####warning binomial will crash the computer
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  NULL, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, bonferroni = TRUE,
fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)

#library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)

manhattan.plot(gwas.table = gwasSTP$gwas.all,pvalue.thr = 0.05/dim(pre.aprAA$geno.data)[2],point.size = 1.2)
sel.markers


sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 


gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM47
GE2021_PM47<-gwasSTP
GE2021_PM47$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM47$trait
table(GE_2021$PM_date)




```
### GE 2021 68

```{r}


simple<-GE_2021%>%filter(PM_date=="68")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE)
#hist(simple$GE)

hist(simple$GE)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)

names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
hist(pre.aprAA $pheno.data$GE)
rand=NULL
fixed=c("Location:Row","Row")
resid=NULL
####warning binomial will crash the computer
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  pre.aprAA$Q, npc = 7, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, bonferroni = FALSE,
fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb',pvalue.thr =p.val )

#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)

###*****There is some error in Location:Row"
#library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)

manhattan.plot(gwas.table = gwasSTP$gwas.all,pvalue.thr =5e-05,point.size = 1.2)
sel.markers


sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 


gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM68
GE2021_PM68<-gwasSTP
GE2021_PM68$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM68$trait
table(GE_2021$PM_date)




```
### GE 2021 96
```{r}


simple<-GE_2021%>%filter(PM_date=="96")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent,AssayNumber)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)^-5))%>%mutate(groups=as.factor(cut(simple$AssayNumber,8)))

#hist(simple$GE)

hist(simple$GE)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)

names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##

rand=NULL
fixed=c("Location:Row","Location")
resid="Location"

####warning binomial will crash the computer
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  pre.aprAA$Q, npc = 7, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, bonferroni = FALSE,
fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb',maxiter.update = 10,pvalue.thr = p.val)

#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)

###*****There is some error in Location:Row"
#library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)

manhattan.plot(gwas.table = gwasSTP$gwas.all,pvalue.thr = 0.05/dim(pre.aprAA$geno.data)[2],point.size = 1.2)
sel.markers


sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 


gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm,"did not covnver")
GE2021_PM96
GE2021_PM96<-gwasSTP
GE2021_PM96$trait<-paste0(tr,"-",ty,"-PM_",tpm,": did_not_converge")
GE2021_PM96$trait
table(GE_2021$PM_date)
```
### GE 2021 152
```{r}


simple<-GE_2021%>%filter(PM_date=="152")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_2021$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE)
#hist(simple$GE)

hist(simple$GE)
simple$ND<-as.character(simple$Check)
length(simple$GI_0)

names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
hist(pre.aprAA $pheno.data$GE)
rand=NULL
fixed=c("Row","Location")
resid=NULL
####warning binomial will crash the computer
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =  pre.aprAA$Q, npc = 7, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, bonferroni = TRUE,
fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')

#rm(gwasSTP)
hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)

###*****There is some error in Location:Row"
#library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)

manhattan.plot(gwas.table = gwasSTP$gwas.all,pvalue.thr = 0.05/dim(pre.aprAA$geno.data)[2],point.size = 1.2)
sel.markers


sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 


gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here
paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM152
GE2021_PM152<-gwasSTP
GE2021_PM152$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2021_PM152$trait
table(GE_2021$PM_date)
```
#SAVING 2021 GI and GE
```{r}


GI_2021_GWAS=list(GI2021_PM5,GI2021_PM19,GI2021_PM33,GI2021_PM47,GI2021_PM68,GI2021_PM96,GI2021_PM152)
save(GI_2021_GWAS,file="data/Genotype_data/ASRgwas_results/Single_TP_GI_2021_results.Rdata")

GE_2021_GWAS=list(GE2021_PM5,GE2021_PM12,GE2021_PM33,GE2021_PM68,GE2021_PM96,GE2021_PM152)
save(GE_2021_GWAS,file="data/Genotype_data/ASRgwas_results/Single_TP_GE_2021_results.Rdata")

```
# COMBINED

## GI

### GI PM 5
```{r}
GI_all<-df_begin
GI_all$Year
##GI PM 5
simple<-GI_all%>%filter(PM_date=="5")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)


print(paste0("data input: ",GI_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)
pre.aprAA$Kinv
pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=c("Year:Location")
resid="Year"
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here


paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GI_PM5<-gwasSTP
GI_PM5$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GI_PM5$trait

```

### GI PM 19

```{r}
GI_all<-df_begin
GI_all$Year
##GI PM 5
simple<-GI_all%>%filter(PM_date=="19")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=c("Year:Location")
resid="Year"
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = TRUE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp
library(asreml)
wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)

GI_19<-gwasSTP
GI_19$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GI_19$trait
table(GI_2020$PM_date)
```
## GI 47 PM
```{r}
GI_all<-df_begin
GI_all$Year
##GI PM 5
simple<-GI_all%>%filter(PM_date=="47")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)
length(simple$GI_0)
names(simple)

pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=c("Year:replication")
resid=NULL
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)

GI_47<-gwasSTP
GI_47$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GI_47$trait
table(GI_2020$PM_date)
```
### GI 96
```{r}
GI_all<-df_begin
GI_all$Year
##GI PM 96
simple<-GI_all%>%filter(PM_date=="96")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)



pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed="Year:replication"
resid=NULL
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)

GI_96<-gwasSTP
GI_96$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GI_96$trait

```
### GI 152
```{r}
GI_all<-df_begin
GI_all$Year
##GI PM 96
simple<-GI_all%>%filter(PM_date=="152")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GI_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)



pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=c("Year","Year:replication")
resid=NULL
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)

GI_152<-gwasSTP
GI_152$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GI_152$trait

```
#SAVE GI combined
```{r}
GI_STP_combined=list(GI_PM5,GI_19,GI_47,GI_96,GI_152)

save(GI_STP_combined,file="data/Genotype_data/ASRgwas_results/Single_TP_GI_cpmbined_results.Rdata")
```
## GE
### GE 5
```{r}
GE_all<-df_begin
GE_all$Year
##GI PM 5
simple<-GE_all%>%filter(PM_date=="5")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GE_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)



pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=c("Year","Year:Location")
resid=NULL
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)


t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)

GE_5<-gwasSTP
GE_5$trait<-paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)
GE_5$trait
table(GE_all$PM_date)
```
### GE 19
```{r}
GE_all<-df_begin
GE_all$Year
##GI PM 5
simple<-GE_all%>%filter(PM_date=="19")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GE_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)



pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=NULL

resid="Year"
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 0.05/13401,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)

GE_19<-gwasSTP
GE_19$trait<-paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)
GE_19$trait
table(GE_all$PM_date)
```
### GE 47
```{r}
GE_all<-df_begin
GE_all$Year

simple<-GE_all%>%filter(PM_date=="47")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GE_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)#%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)))

simple$ND<-as.character(simple$Check)



pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=c("Year","Year:Location")
resid=NULL
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 5, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 0.05/13401,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)

GE_47<-gwasSTP
GE_47$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GE_47$trait
table(GE_all$PM_date)
```
### GE 96
```{r}
GE_all<-df_begin
GE_all$Year
##GI PM 5
simple<-GE_all%>%filter(PM_date=="96")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GE_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)),GE_lam=GE^8)


simple$ND<-as.character(simple$Check)
pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE_tr", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed="Year:Location"
resid="Year"
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 7, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 0.05/13401,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)

GE_96<-gwasSTP
GE_96$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GE_96$trait
table(GE_all$PM_date)
```

### GE 152
```{r}
GE_all<-df_begin
GE_all$Year
##GI PM 5
simple<-GE_all%>%filter(PM_date=="152")
simple
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE
#View(simple)

print(paste0("data input: ",GE_all$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,JHI_Hv50k_2016_308762,Row,Column,spot_bloch,ws_percent)%>%mutate(GE_tr=abs(GE - mean(GE,na.rm = TRUE)),GE_lam=GE^8)


simple$ND<-as.character(simple$Check)
pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND=="2")%>%droplevels(), indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##




random=NULL
fixed=NULL
resid=NULL
table(pre.aprAA$pheno.data$Year)
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 7, geno.data = pre.aprAA$geno.data,bonferroni = FALSE,
map.data = pre.aprAA$map.data, pvalue.thr =5e-05,
fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')


hist(gwasSTP$mod$residuals)
summary(gwasSTP$mod)$varcomp

wald.asreml(gwasSTP$mod)
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 0.05/13401,
point.size = 1.2)
sel.markers

sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprAA$geno.data[,sel.markers[c(1,7)], drop = FALSE]
sel.markers 
Msel 

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN



#change here

paste0(paste0("GE-",ty[1],"/",ty[2]),"-","PM_",tpm)

GE_152<-gwasSTP
GE_152$trait<-paste0(paste0("GI-",ty[1],"/",ty[2]),"-","PM_",tpm)
GE_152$trait
table(GE_all$PM_date)
```
#GE COMBINED
```{r}
GE_STP_combined=list(GE_5,GE_19,GE_47,GE_96,GE_152)
GE_STP_combined[[4]]$trait
save(GE_STP_combined,file="data/Genotype_data/ASRgwas_results/Single_TP_GE_combined_results.Rdata")
```
# Malting Quality GWAS
```{r}

Sys.info()['sysname']
if( .Platform$OS.type == "unix" ){
  patht <- "/home/karl/git/"
}
if( .Platform$OS.type == "windows" ){
  patht <- "C:/Users/kars8/git/"
}

patht<-paste0(patht,"NY-winter-barley-analysis/")
load("data/MQ/WMB21_Master_Cornell.Rdata")
View(MQ_WMB21)
dim<-genoD%>%as.matrix()

```

```{r}
View(MQ_WMB21)
MQ<-MQ_WMB21%>%rename(GID=gid)%>%filter(!Treatment=="SpringTP2-4")%>%left_join(marker_ef,by="GID")%>%mutate(type="base_data_single_step")
MQ[MQ$GID=="DH130910",]$GID<-"LIGHTNING"
library(tidyr)
MQ%>%pivot_longer(names_to="MQ_trait",cols=c(BG,AA,DP,FAN,sp_mean,ME,MP,ST))%>%ggplot(aes(value)) + 
        # it sets bins intead of breaks, so add 1
        geom_histogram(bins = 51) + 
        # make a new "facet" for each value of `variable` (formerly column names), and 
        # use a convenient x-scale instead of the same for all 3
        facet_wrap(~Timepoint+MQ_trait,scales = "free_x")+theme_bw()


gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",MQ$type%>%unique()))

#simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones
```
#may need to do two step analysis
```{r}

pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "AA", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.20)


asreml_step1 <- function(d2, groupvars) {
d2<-MQ_long%>%filter(trait=="AA",Timepoint=="TP2")


model0<-asreml(fixed=value~1,random=~GID,residual=~units,data=d2,na.action = na.method(x = "include"))
M
model_narrow<-asreml(fixed=value~1,random=~vm(GID, pre.aprAA$Kinv$GID),residual=~units,data=d2,na.action = na.method(x = "include"))
BLUP<-summary(model_narrow,coef=TRUE)$coef.random
d2$value
dim(BLUP)
cor(BLUP,d2$value,method = "pearson",use="complete.obs")
var_comp_narrow<-as.data.frame(summary(model_narrow)$varcomp)
var_comp_narrow
var_Comp<-as.data.frame(summary(model0)$varcomp)

t<-as.data.frame(table(d2$GID));t<-t[!t$Var1%in%c("R","NACL"),];r<-mean(t$Freq)
r
var_Comp
H2<-round(var_Comp["GID",]$component/((var_Comp["GID",]$component+var_Comp["units!R",]$component)/r),4)
h2<-round(var_comp_narrow[1,]$component/((var_comp_narrow[1,]$component+var_comp_narrow["units!R",]$component)/r),4)
h2<-as.data.frame(h2)
H2<-as.data.frame(H2)
H2
predict
predictV<-predict(model0,classify = "GID")

object<-as.data.frame(list(predictV,H2,h2))
object
return(object)
step1_MQ.0<-data%>%group_by(Timepoint,trait)%>%group_modify(asreml_step1)%>%ungroup()
}


MQ_long<-MQ%>%mutate(GID=as.factor(GID))%>%pivot_longer(names_to="trait",cols=c(AA,BG,DP,FAN,ME,sp_mean,ST,MP))%>%group_by(Timepoint,trait)



```
#GBLUP
```{r}
asreml_step1 <- function(d2, groupvars) {
MQ_long$
df<-MQ_long%>%filter(trait%in%c("BG"))
  pre.aprGBLP <- pre.gwas(pheno.data=d2, indiv = "GID",
resp = "value", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.20)

 pre.aprGBLP$pheno.data

#model0<-asreml(fixed=value~Timepoint,random=~GID,residual=~units,data=d2,na.action = na.method(x = "include"))

print(groupvars)
model_narrow<-asreml(fixed=value~Timepoint,random=~vm(GID, pre.aprGBLP$Kinv$GID),residual=~units,data=pre.aprGBLP$pheno.data,na.action = na.method(x = "include"))
var_comp_narrow<-as.data.frame(summary(model_narrow)$varcomp)
model_narrow<-update.asreml(model_narrow)
BLUP<-summary(model_narrow,coef=TRUE)$coef.random
pheno<-pre.aprGBLP$pheno.data%>%group_by(GID)%>%summarize(mean=mean(value))%>%ungroup()

BLUP_cor<-data.frame(BLUP_cor=cor(BLUP,pheno$mean,method = "pearson",use = "complete.obs"))

var_Comp<-as.data.frame(summary(model0)$varcomp)

t<-as.data.frame(table(d2$GID));t<-t[!t$Var1%in%c("R","NACL"),];r<-mean(t$Freq)
H2<-round(var_Comp["GID",]$component/((var_Comp["GID",]$component+var_Comp["units!R",]$component)/r),4)
h2<-round(var_comp_narrow[1,]$component/((var_comp_narrow[1,]$component+var_comp_narrow["units!R",]$component)/r),4)
h2<-as.data.frame(h2)
H2<-as.data.frame(H2)
predictV<-predict(model0,classify = "GID")

object<-list(predictV,H2,h2,BLUP_cor)

return(object)

}

step1_MQ.0<-MQ_long%>%filter(!trait%in%c("ST","sp_mean"))%>%group_by(trait)%>%group_map(asreml_step1,.keep = TRUE)

```

Biplot
```{r}


%>% group_modify(~{
    PCA = .x %>% select(AA:GITP6) %>%
      princomp(., cor=T)
    X <- PCA$loadings
    X[,1:2]
    loadings = data.frame((X[,1:3]))
    loadings %>% rownames_to_column(var = 'trait') %>% mutate(PercentVar2PC = sum((PCA$sdev^2)[1:2])/sum(PCA$sdev^2) )
  })


```
```{r}


###


simple<-MQ%>%rename(crop_year="Crop Year")

genoMp
table(step1_MQ1$trait)
#d3<-step1_MQ1%>%rename(GID=taxa,value=pvals.predicted.value)%>%filter(trait=="TotalProtein")
View(MQ)
names(MQ)
pre.aprAA <- pre.gwas(pheno.data=MQ, indiv = "GID",
resp = "FAN", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.20)
##v

fixed=c("Treatment")
random=NULL
#View(pre.aprAA$pheno.data)
resid="Timepoint"

gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "MP",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 2, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr =1e-05,bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = random,workspace='2Gb')
gwasSTP$heritability
gwasSTP$heritability
FDR_thresholds(gwasSTP)
summary(gwasSTP$mod)$varcomp
wald.asreml(gwasSTP$mod)



qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = FDR_thresholds(gwasSTP)[1,1],
point.size = 1.2)


tr<-"ST";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
t3<-simple[,c("AA","Timepoint")]
#change here

GI2020_PM5<-gwasSTP
GI2020_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM5$FDR_rate<-FDR_thresholds(gwasSTP)
GI2020_PM5$FDR_rate
table(GI_2020$PM_date)
```