---
title: "GWAS_analysis"
author: "Karl Kunze"
date: "2023-03-18"
output: html_document
editor_options: 
  chunk_output_type: console
---
#packages
```{r}
library(readxl)
library(ggplot2)

library(ASRgwas)


```
#data upload
```{r}
load("data/Phenotype_Data/AllDHBluesPerYear_first_step.Rdata")
load("data/Phenotype_Data/Heritability_first_step.Rdata")

load("data/Genotype_data/wmb_file.Rdata")
load("data/Genotype_data/numeric_raw.Rdata")
load("data/WMB_DH_Germination_data2020_2021.RData")#DH2020_2021
DH2020_2021<-DH2020_2021%>%filter(!Location=="Spring")

DH2020_2021[DH2020_2021$taxa%in%c("Charles","Endeavor"),]$Check<-"2"

```



```{r}
# AllDHBluesPerYear=AllDHBluesPerYear%>%mutate(family = mapvalues(substr(taxa,1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
#                             to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
#       
#                                                                 'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")))
#         
# AllDHBluesPerYear$family

 
```
## H2 visualization
```{r}
DH_H2%>%
   ggplot(aes(x = TP, y = ah2, fill = trait)) +geom_bar(stat = 'identity', position = 'dodge')+
  facet_wrap(vars(Year), ncol = 1)+theme_bw()+labs(title= 'Narrow sense heritability\nover time and datasets')

#other ideas to expand here, perhaps calculate narrow sense heritability, since we have the genotype info
#adding genotype data

table(AllDHBluesPerYear$taxa)
#AllDHBluesPerYear= rbind(DH2020Estimates, DH2021Estimates,DHCombined) %>% filter(type =='BLUE') %>%mutate(PM_date=as.double(PM_date),year=as.character(year))%>%ungroup()
#load("/home/karl/git/TimeSeriesGermination/WinterBarley/Analysis/AllDHBluesPerYear.RData")
```

## Genotype data

```{r}


# geno<-wmb_num$GD
# 
# geno=as.data.frame(apply(geno[-1],1, function(x) { 
#   x[which(x == 0)] = -1
#   x[which(x == 1)] <- 0
#   x[which(x == 2)] = 1
#   x
# }))
# geno
# geno$SNP<-rownames(geno)
# geno=geno%>%relocate(SNP,.before=1)


#geno<-WinterGM%>%full_join(geno,by="SNP")%>%arrange(Chromosome,Position)%>%dplyr::rename(marker=SNP,Chrom=Chromosome,bp=Position)%>%as.data.frame()
#rownames(geno)<-geno$marker
genoD_raw<-wmb_num$GD
genoM_raw<-wmb_num$GM

wmb_num$GM[wmb_num$GM$SNP=="Qsd1",]
wmb_num$GD[,"Qsd1"]
```
#winterGD and WinterGM
```{r}
#load('data/Genotype_data/myGM20_LDprune.Rdata')
#load('data/Genotype_data/myGD20_LDprune.Rdata')

WinterGM=wmbGAPIT$GM;WinterGD=as.data.frame(wmbGAPIT$GD)
#rm(myGM20_prune);rm(myGD20_prune)
WinterGM = WinterGM %>% arrange(Chromosome, Position)
#my taxa IDs have "-" instead of an underscore which I prefer
#taxa = gsub(pattern = '-',replacement = '_',taxa)
rownames(WinterGD)
WinterGD$taxa<-rownames(WinterGD)

str(WinterGD)
WinterGD=WinterGD %>%relocate(taxa,.before = 1)%>%

  mutate(taxa = gsub(pattern = '\\ ', replacement = '_',taxa)) %>% tibble::remove_rownames()



WinterGD
#summarise(taxa = "Endeavor",Qsd1 = 2)%>%bind_rows(WinterGD, .)%>%mutate(Qsd1=replace(Qsd1,Qsd1==1,2))
WinterGD$taxa
WinterGD[WinterGD$taxa=="ENDEAVOR",]$Qsd1
WinterGD[WinterGD$taxa=="CHARLES",]$Qsd1

dim(WinterGM)
colnames(WinterGD)[!colnames(WinterGD)%in%WinterGM$SNP]
dim(WinterGD)
# Make sure things are in the right order
# Sum should = 8384
dim(WinterGD)

WinterRelationship=wmbGAPIT$A
WinterPCA = eigen(WinterRelationship)
substr(WinterGD$taxa,1,3)
WinterPVEPCA = WinterPCA$values/sum(WinterPCA$values)
data.frame(ordinal = 1:10, PVE = WinterPVEPCA[1:10]) %>%plot(., xlab = 'PC', col = 'red') 
winterlinePCAvalues = WinterPCA$vectors %>% data.frame()%>% 
  
  mutate(family = mapvalues(substr(WinterGD$taxa,1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
                            to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
      
                                                                'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")),
         taxa = WinterGD$taxa,
         shapes = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), taxa, 'Lines'),
         size = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), 3, 2))



winterlinePCAvalues %>% filter(family !=c('Check')) %>%
  ggplot(aes(x = X1, y = X2, color = family, shape = shapes)) + geom_point(aes(size = size))+theme_bw() +guides(size = "none")+
  xlab('PC1')+ylab('PC2')+ggtitle(lab="PCA by family of NY \n WMB DH population")


setwd(rprojroot::find_rstudio_root_file())


#Travis' Time series perspective
AllDHBluesPerYear
AllDHBluesPerYear  %>%  filter(type == 'BLUE') %>% mutate(headerFacet = 'Data source') %>%
  ggplot(aes(x = PM_date, y = value, group = taxa))+
  geom_smooth()+facet_nested(trait~headerFacet+year, scales = 'free')+
  geom_vline(xintercept = c(12,33,68), color = 'red')+
  labs(title = 'GE and GI BLUEs over time with differeing datasets')+theme_bw()

DHCombined %>% filter(type == 'BLUE' & trait == 'GI') %>% ggplot(aes(x = PM_date, y = value, group = taxa))+geom_line()+
  geom_vline(xintercept = c(12,33,68), color = 'red')
#filter(Family %nin% c('DH130910'))%>%

#png('plots/Time_series/Sup_BluesByFamilyQsd1AllYears.png', 2400, 1500, res =120)



#png('WinterBarley/WinterDHGerminationPaper/picsPNGforQsd1Effects_paper/BluesByFamilyQsd12020_2021.png', 1400, 800, res =120)
#2020
table(AllDHBluesPerYear$PM_date,AllDHBluesPerYear$year)

AllDHBluesPerYear
 filter(type == 'BLUE') %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Dormant','Nondormant')) %>%
  filter(family %nin% c('Lightning')) %>% filter(year %in% c('2020'))  %>%
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x = TP, y = value, fill = Qsd1))+
 
  geom_boxplot()+facet_nested(trait~family, scales = 'free')+
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination Rates for 2020 Year"))+
theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))

#2021
AllDHBluesPerYear %>%  filter(type == 'BLUE') %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Dormant','Nondormant')) %>%
  filter(Family %nin% c('Lightning')) %>% filter(year %in% c('2021'))  %>%
  filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x = TP, y = value, fill = Qsd1))+  
  geom_boxplot()+facet_nested(trait~Family, scales = 'free')+ 
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination Rates for 2021 Year"))+
  theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))
#2020/2021
AllDHBluesPerYear %>%  filter(type == 'BLUE') %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  join(AlaT[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Dormant','Nondormant')) %>%
  filter(Family %nin% c('Lightning')) %>% filter(year %in% c('2020/2021'))  %>%
  filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x = TP, y = value, fill = Qsd1))+  
  geom_boxplot()+facet_nested(trait~Family, scales = 'free_y')+
  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination Rates for Combined 2020/2021 Years"))+
  theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))

dev.off()
#####


#library(GAPIT3)
DF_OperationsV3 = function(df){
  df = df %>% arrange(P.value)  %>% mutate(logPercentileQQplot = -log10(c(1:length(df$SNP))/length(df$SNP)),
                                           rank = c(1:length(df$SNP))) %>% arrange(Chrom, as.numeric(Pos)) %>%
    mutate(log10PVal = -log10(P.value),ordinal = c(1:length(df$SNP)))
  return(df)
}
#I am assuming this is to help make manhattan plots

GWA_MLMM_fortidyR = function(df, groupvars) {
  #fg<- DH2020_1 %>% rbind(DH2021_1, DHcomb_1) %>%  filter(type == 'BLUE') %>%
  # ungroup()%>%filter(year=="2020",TP=="TP1",trait=="GI")

  
  
  # %>% group_by(year, TP, trait)
  
  GWAS = GAPIT(Y = df%>% dplyr::select(taxa, value) %>% as.data.frame(),GD=WinterGD, GM=WinterGM,PCA.total = 2,
               Geno.View.output=F, model="MLMM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05)
    GWAS 
  Out = DF_OperationsV3(GWAS$GWAS) %>% arrange(P.value) %>% slice_head(n=1000)
  return(Out)
}
#GWA MLMM per timepoint,
# GWA_MLMM_fortidyR.perFamily = function(df, groupvars) {
#   WinterGD_sub = WinterGD %>% filter(taxa %in% df$taxa)
#   Out = tryCatch(
#     {GWAS = suppressWarnings(GAPIT(Y = df %>%ungroup() %>% select(taxa, value) %>% as.data.frame(),GD=WinterGD_sub, GM=WinterGM,PCA.total = 0,
#                                    Geno.View.output=F, model="MLMM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05))
#     DF_OperationsV3(GWAS$GWAS) %>%
#       arrange(P.value) %>% slice_head(n = 100)
#     }, error = function(e){data.frame() } )
#   return(Out)
# }
#per family, I'm not going to worry about this one for now
# GWA_MLM_fortidyR = function(df, groupvars) {
#   GWAS = GAPIT(Y = df %>% ungroup() %>% select(taxa, value) %>% as.data.frame(),GD=WinterGD, GM=WinterGM,PCA.total = 2,
#                Geno.View.output=F, model="MLM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05)
#   Out = (GWAS$GWAS) %>% arrange(P.value)  %>%
#     mutate(logPercentileQQplot = -log10(c(1:length(GWAS$GWAS$SNP))/length(GWAS$GWAS$SNP)),
#            rank = c(1:length(GWAS$GWAS$SNP))) %>% arrange(Chromosome, 'Position') %>%
#     mutate(log10PVal = -log10(P.value), ordinal = c(1:length(GWAS$GWAS$SNP)))%>% 
#     arrange(P.value) %>% slice_head(n = 1000)
#   return(Out)
# }
#not going to worry about this one either for germination traits at least
GWA_MLM_fortidyR.perFamily = function(df, groupvars) {
  WinterGD_sub = WinterGD %>% filter(taxa %in% df$taxa)
   Out = tryCatch(
    {GWAS = suppressWarnings(GAPIT(Y = df %>% ungroup() %>% select(taxa, value) %>% as.data.frame(),GD=WinterGD_sub, GM=WinterGM,PCA.total = 0,
                                    Geno.View.output=F, model="MLM", Major.allele.zero = F, file.output=F,SNP.MAF = 0.05))
    (GWAS$GWAS) %>% arrange(P.value)  %>%
      mutate(logPercentileQQplot = -log10(c(1:length(GWAS$GWAS$SNP))/length(GWAS$GWAS$SNP)),
             rank = c(1:length(GWAS$GWAS$SNP))) %>%  arrange(Chromosome, 'Position') %>%
       mutate(log10PVal = -log10(P.value), ordinal = c(1:length(GWAS$GWAS$SNP)))%>% 
       arrange(P.value) %>% filter(P.value < 0.01)
    }, error = function(e){data.frame() } )
   return(Out)
 }

# Add_DH130910_familyStructure = function(df3, groupvar){
#   DH130910_rows = df3 %>% filter(taxa == 'Lightning') %>% select(!family)
#   df3= df3 %>% filter(taxa != 'Lightning') %>%
#     rbind(DH130910_rows %>% mutate(Family='Flavia/DH130910'),
#           DH130910_rows %>% mutate(Family='SY_Tepee/DH130910'),
#           DH130910_rows %>% mutate(Family='Scala/DH130910'),
#           DH130910_rows %>% mutate(Family='Wintmalt/DH130910')
#     ) %>%
#     filter(Family %in% c('Flavia/DH130910','SY_Tepee/DH130910','Scala/DH130910','Wintmalt/DH130910'))
#   return(df3)
# }


#Per time point for MLMM
##takes some time
#SAVE THE RESULTS so that we dont have to run again
print("MLMM GWA model ahead")

library(GAPIT3)

 WinterPerTPGWAS = AllDHBluesPerYear %>%  filter(type == 'BLUE') %>%
  ungroup() %>% group_by(year, TP, trait)%>%group_modify(GWA_MLMM_fortidyR)
save(WinterPerTPGWAS ,file="data/GWA_results/WinterPerTPGWAS.Rdata")

```
```{r}
# 
# geno<-WinterGD
# rownames(geno)<-WinterGD$taxa
# geno=as.data.frame(t(geno[-1]))
# geno$SNP<-rownames(geno)
# geno=geno%>%relocate(SNP,.before=1)
# geno<-WinterGM%>%full_join(geno,by="SNP")%>%arrange(Chromosome,Position)%>%dplyr::rename(marker=SNP,chrom=Chromosome,bp=Position)
# geno

```



#asreml GWAS example
```{r}

library("ASRgenomics")

data(geno.pine655)
geno.pine655[1:5, 1:5]
M_filter <- qc.filtering(M = geno.pine655, base = FALSE, ref = NULL,
marker.callrate = 0.2, ind.callrate = 0.2, maf = 0.05, heterozygosity = 0.95,
Fis = 1, impute = FALSE, na.string = "-9", plots = TRUE)
M_filter$M.clean[1:5, 1:5]
dim(M_filter$M.clean)
G <- G.matrix(M = M_filter$M.clean, method = "VanRaden", na.string = NA)$G
#use Yang method for inbreds
data(ped.pine)
head(ped.pine)
tail(ped.pine)
A <- AGHmatrix::Amatrix(data = ped.pine)
A
A[601:605, 601:605]
check_G <- kinship.diagnostics(K = G, diagonal.thr.small = 0.8,
diagonal.thr.large = 1.2, duplicate.thr = 0.95)
check_G$list.diagonal
check_G$list.duplicate
check_G$plot.diag
check_G$plot.offdiag

####
G2A <- match.G2A(A = A, G = G, clean = TRUE, ord = TRUE, mism = TRUE,
RMdiff = TRUE)
G2A$Aclean
G2A$Aclean[343:347, 343:347]
G2A$Aclean[343:347, 343:347]
G2A$plotG2A
#clean up
G_align <- G.tuneup(G = G2A$Gclean, A = G2A$Aclean, align = TRUE)$Gb

Ga2A <- match.G2A(A = A, G = G_align, clean = TRUE, ord = TRUE,
mism = TRUE, RMdiff = TRUE)
dim(Ga2A$RM[Ga2A$RM$absdiff > 0.2, ])
check_G_align <- kinship.diagnostics(K = G_align)
check_G_align$plot.diag
check_G_align$plot.offdiag
#kinship
kinship.heatmap(K = G_align, dendrogram = TRUE, row.label = FALSE,
col.label = FALSE)

pca_pine <- kinship.pca(K = G_align, ncp = 15)
pca_pine$eigenvalues
pca_pine$plot.scree
pca_pine$plot.pca

```
#example again
```{r}
library(ASRgenomics)
head(pheno.salmon)
head(geno.salmon)
pre.salmonA <- pre.gwas(pheno.data = pheno.salmon,
indiv = "ID", resp = "amoebic_load", geno.data = geno.salmon,
Q.method = "K", maf = 0.05, marker.callrate = 0.2,
ind.callrate = 0.2, method = "VanRaden", heterozygosity = 0.8,
Fis = 0.98, impute = TRUE)

pre.salmonD <- pre.gwas(pheno.data = pheno.salmon,
indiv = "ID", resp = "amoebic_load", geno.data = geno.salmon,
Q.method = "K", maf = 0.05, marker.callrate = 0.2,
ind.callrate = 0.2, method = "Su", heterozygosity = 0.8,
Fis = 0.98, impute = TRUE)

pheno.AD<-pre.salmonD$pheno.data
pheno.AD$IDA <- pheno.AD$ID
pheno.AD$IDD <- pheno.AD$ID

Ma <- round(pre.salmonA$geno.data, 0)
Md <- round(pre.salmonD$geno.data, 0)
colnames(Md) <- paste("D", colnames(Md), sep = "_")
Md[Md == 2] <- 0
dim(Ma)
dim(Md)
Mad <- cbind(Ma, Md)


gwasS <- gwas.asreml(pheno.data = pheno.AD, resp = "amoebic_load",
gen = c("IDA", "IDD"), Kinv = list(IDA = pre.salmonA$Kinv$ID,
IDD = pre.salmonD$Kinv$ID), Q = pre.salmonA$Q,
npc = 7, geno.data = Mad, map.data = NULL, pvalue.thr = 5e-04,
bonferroni = FALSE, P3D = TRUE)

summary(gwasS$mod)$varcomp
gwasS$heritability
gwasS$gwas.sel




```
#asreml actual GWAS
```{r}
library("ASRgwas")


genoD<-genoD_raw
dim<-as.matrix(genoD)
dim[1:5,1:5]
rownames(dim)%in%c("ENDEAVOR")
library(dplyr)
genoM<-genoM_raw%>%dplyr::rename(marker=SNP,chrom="Chromosome","pos"="Position")%>%filter(marker%in%colnames(dim))

table(DH2020_2021$PM_date,DH2020_2021$Year)


#marker effects
marker_ef<-dim%>%as.data.frame()%>%dplyr::select("Qsd1")%>%tibble::rownames_to_column()%>%rename(GID=rowname)%>%mutate(Qsd1=as.factor(Qsd1))
df_begin<-DH2020_2021%>%mutate(GID=toupper(taxa))%>%filter(!Location=="Spring")%>%left_join(marker_ef,by="GID")%>%mutate(type="base_data_single_step")
table(df_begin$GID)
###
df_begin<-df_begin%>%mutate(replication=as.factor(replication),Location=as.factor(Location),Check=as.factor(Check),GI_0=GI/GE)
df_begin
#Single Timepoint ASRgwas
```
#Single Timepoint GWAS
## Germination Index 2020
```{r}
GI_all<-df_begin


GI_all
## 2020

GI_2020<-GI_all%>%filter(Year=="2020")%>%droplevels()
table(GI_2020$PM_date)


### PM 5
simple<-GI_2020%>%filter(PM_date=="5")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones


###
hist(simple$GI)
simple$ND<-as.character(simple$Check)
simple$ND
pre.aprAA <- pre.gwas(pheno.data=simple%>%filter(!ND==2), indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
fixed=c("Location","Check")
random=NULL
resid="Location"
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 8, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GI","Year","PM_date")]
t3
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#change here

GI2020_PM5<-gwasSTP
GI2020_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM5$trait
table(GI_2020$PM_date)

### 19

simple<-GI_2020%>%filter(PM_date=="19")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)



###
table(simple$GID)

pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=c("Location","Check");resid="Location";maf_lim=0.05;p.val=5e-05

##
pre.aprAA$pheno.data$Row
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =NULL, npc = 4, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp

t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
t3

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]
sel.markers
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,15)],
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN 
#change here
GI2020_PM19<-gwasSTP
tpm
GI2020_PM19$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI_2020$PM_date
GI2020_PM19$trait
### 47
simple<-GI_2020%>%filter(PM_date=="47")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GI,GE)



###
pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=c("Location");resid=NULL;maf_lim=0.05;p.val=5e-05

##

gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q =NULL, npc = 4, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp


t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]
sel.markers 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,2,18,21)],
ref.vc = 1, pvalue.thr = 0.01, maxiter.update = 4)
gwasFIN
#change here
tpm
GI2020_PM47<-gwasSTP
paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM47$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM47$trait
### 96
simple<-GI_2020%>%filter(PM_date=="96")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,Day2Germ,Day3Germ,Day4Germ,Day5Germ,scald,Qsd1,GI,GE,GI_0)



###
pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI_0", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=c("replication");resid="replication";maf_lim=0.05;p.val=5e-05


gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 6, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=NULL,randomf = NULL,workspace='2Gb')
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp


t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,length(sel.markers))],
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN
#change here
tpm
paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM96<-gwasSTP

GI2020_PM96$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM96$trait
table(GI_2020$PM_date)

# 2020 GI 152
simple<-GI_2020%>%filter(PM_date=="152")
simple$PM_date%>%unique()


print(paste0("data input: ",df$type%>%unique()))
simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,Day2Germ,Day3Germ,Day4Germ,Day5Germ,scald,Qsd1,GI,GE,GI_0)



###
pre.aprAA <- pre.gwas(pheno.data = simple, indiv = "GID",
resp = "GI", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = FALSE, Q.method = "K",heterozygosity = 0.20)
##
gd=dim;genoMp=genoM;rand=NULL;fixed=NULL;resid=NULL;maf_lim=0.05;p.val=5e-05


gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GI",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 6, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = FALSE,fixedf = fixed,residual=NULL,randomf = NULL,workspace='2Gb')
##
hist(gwasSTP$mod$residuals)
wald.asreml(gwasSTP$mod)
summary(gwasSTP$mod)$varcomp


t3<-simple[,c("Year","PM_date")]
tr<-"GI";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()


qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#marker effects
sel.markers <- gwasSTP$gwas.sel$marker
Msel <- pre.aprAA$geno.data[, sel.markers, drop = FALSE]

gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel[,c(1,length(sel.markers))],
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)
gwasFIN
#change here
tpm
paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM152<-gwasSTP

GI2020_PM152$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GI2020_PM152$trait
table(GI_2020$PM_date)
GI_2020=list(GI2020_PM5,GI2020_PM19,GI2020_PM47,GI2020_PM96,GI2020_PM152)
```
### 2020 Germination Energy
```{r}

GE_all<-df_begin


GE_all
## 2020

GE_2020<-GE_all%>%filter(Year=="2020")%>%droplevels()
table(GE_2020$PM_date)


### PM 5
simple<-GE_2020%>%filter(PM_date=="5")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones


###
hist(simple$GE)
simple$ND<-as.character(simple$Check)
simple$ND
simple$
pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="Location"
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 4, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#change here
tr
GE2020_PM5<-gwasSTP
GE2020_PM5$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM5$trait
table(GE_2020$PM_date)
### 2020 GE 19




simple<-GE_2020%>%filter(PM_date=="19")
simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0)
#should do Genomic prediction for the missing ones


###
hist(simple$GE)
simple$ND<-as.character(simple$Check)
simple$ND
simple
pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="Location"
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 4, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM19<-gwasSTP
GE2020_PM19$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM19$trait
table(GE_2020$PM_date)
##### 47

simple<-GE_2020%>%filter(PM_date=="47")

simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,GE5,GI5)%>%mutate(GI5_0=GI5/GE5)



###
hist(simple$GE)
simple$ND<-as.character(simple$Check)


pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="replication"
random=NULL
resid="replication"
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = NULL, npc = 4, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM47<-gwasSTP
GE2020_PM47$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM47$trait
table(GE_2020$PM_date)
######## 96

simple<-GE_2020%>%filter(PM_date=="96")

simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,GE5,GI5)%>%mutate(GI5_0=GI5/GE5)



###
hist(simple$GE)
simple$ND<-as.character(simple$Check)


pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed="Location"
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 8, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM96<-gwasSTP
GE2020_PM96$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM96$trait
table(GE_2020$PM_date)
### GE 2020 152


simple<-GE_2020%>%filter(PM_date=="152")

simple$PM_date
gd=dim;genoMp=genoM;maf_lim=0.05;p.val=5e-05;bonfer=FALSE


print(paste0("data input: ",df$type%>%unique()))

simple<-simple%>%dplyr::select(SourcePLOT,GID,Year,PM_date,replication,Location,Check,Day1Germ,scald,Qsd1,GE,GI,GI_0,GE5,GI5)%>%mutate(GI5_0=GI5/GE5)



###
hist(simple$GE)
simple$ND<-as.character(simple$Check)


pre.aprAA <- pre.gwas(pheno.data=simple, indiv = "GID",
resp = "GE", geno.data = gd,map.data = genoMp,
maf = maf_lim, marker.callrate = 0.2, ind.callrate = 0.2,
method = "VanRaden", rename.markers = TRUE, impute = TRUE, Q.method = "K",heterozygosity = 0.05)
##
fixed=NULL
random=NULL
resid=NULL
gwasSTP<- gwas.asreml(pheno.data = pre.aprAA$pheno.data,resp = "GE",
 gen = "GID", Kinv = pre.aprAA$Kinv,
Q = pre.aprAA$Q, npc = 8, geno.data = pre.aprAA$geno.data,
map.data = pre.aprAA$map.data, pvalue.thr = p.val,
bonferroni = bonfer,fixedf = fixed,residual=resid,randomf = rand,workspace='2Gb')
t3<-simple[,c("GE","Year","PM_date")]
t3
tr<-"GE";ty<-as.character(t3$Year)%>%unique();tpm<-as.character(t3$PM_date)%>%unique()
tpm
wald.asreml(gwasSTP$mod)

qq.plot(gwas.table = gwasSTP$gwas.all)
manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
point.size = 1.2)
#signif marker analysis
sel.markers <- gwasSTP$gwas.sel$marker
sel.markers
Msel <- pre.aprA$geno.data[, sel.markers[c(1:5)], drop = FALSE]
Msel 
gwasFIN <- select.marker(gwas.object = gwasSTP, geno.data.sel = Msel,
ref.vc = 1, pvalue.thr = 0.0001, maxiter.update = 4)


#change here
tr
tpm
ty
GE2020_PM152<-gwasSTP
GE2020_PM152$trait<-paste0(tr,"-",ty,"-PM_",tpm)
GE2020_PM152$trait
table(GE_2020$PM_date)

### GE 2020 combined

GE_2020<-list(GE2020_PM5,GE2020_PM19,GE2020_PM47,GE2020_PM96,GE2020_PM152)
GE2020
```
#SAVING GE2020 and GI2020 Single Timepoint GWAS
```{r}
save(GI_2020,file="data/Genotype_data/ASRgwas_results/Single_TP_GI_2020_results.Rdata")
save(GE_2020,file="data/Genotype_data/ASRgwas_results/Single_TP_GE_2020_results.Rdata")
```
2021 Germination Index
```{r}









# qq.plot(gwas.table = gwasSTP$gwas.all)
# manhattan.plot(gwas.table = gwasSTP$gwas.all, pvalue.thr = 5e-05,
# point.size = 1.2)
# summary(gwasSTP$mod)$varcomp
# summary(gwasSTP$mod)
# gwasMTP$gwas.sel


# library(asreml)
# ?asreml()
# d3<-pre.aprA$pheno.data
# d3$PM_date_t<-as.numeric(as.character(d3$PM_date))
# d3$PM_date
# model<-asreml(
#   fixed = GI~ 1,
#   random = ~GID+Year:dev(PM_date_t),
#   sparse = ~NULL,
#   residual = ~NULL,data=d3)
# summary(model)$varcomp
###

```
