---
title: "Haplotype_analysis"
author: "Karl Kunze"
date: "2/9/2021"
output: html_document
editor_options: 
  chunk_output_type: console
---
read in phenotypes
```{r}
setwd("C:/Users/Karl/git/BarleyGermination-KK")
load(here("data/Winter_DH/TP1_WDH20_BLUE_all.RData"))
load(here("data/Winter_DH/TP2_WDH20_BLUE_all.RData"))
load(here("data/Winter_DH/TP3_WDH20_BLUE_all.RData"))
load(here("data/Winter_DH/TP4_WDH20_BLUE_all.RData"))
load(here("data/Winter_DH/TP5_WDH20_BLUE_all.RData"))
load(here("data/Winter_DH/Field_WDH20_BLUE_all.Rdata"))

load(here("data/Winter_DH/TP1_WDH20_BLUP_all.RData"))
load(here("data/Winter_DH/TP2_WDH20_BLUP_all.RData"))
load(here("data/Winter_DH/TP3_WDH20_BLUP_all.RData"))
load(here("data/Winter_DH/TP4_WDH20_BLUP_all.RData"))
load(here("data/Winter_DH/TP5_WDH20_BLUP_all.RData"))
load(here("data/Winter_DH/Field_WDH20_BLUP_all.Rdata"))

load(here("data/Winter_DH/WMB DH 2020 all data for analysis.RData"))
load("C:/Users/Karl/git/BarleyGermination-KK/R_scripts/data/GWAS.Field.hits2.Rdata")
load("C:/Users/Karl/git/BarleyGermination-KK/R_scripts/data/GWAS.TP.hits2.Rdata")

```
read in Genotypes
```{r}
getwd()
list.files("data/Genotype_data")
load(here("data/Genotype_data/myGD20_LDprune.Rdata"))
load(here("data/Genotype_data/myGM20_LDprune.Rdata"))
load("C:/Users/Karl/git/BarleyGermination-KK/R_scripts/data/GWAS.Field.hits2.Rdata")
load("C:/Users/Karl/git/BarleyGermination-KK/R_scripts/data/GWAS.TP.hits2.Rdata")
unique_list
#View(GWAS.sign.TP_all[with(GWAS.sign.TP_all,order(P.value)),])


myGD20<-myGD20_prune
myGM20<-myGM20_prune
rm(myGD20_prune);rm(myGM20_prune)



TP_all_WDH20_BLUE<-rbind(TP1_WDH20_BLUE_all,TP2_WDH20_BLUE_all,TP3_WDH20_BLUE_all,TP4_WDH20_BLUE_all,TP5_WDH20_BLUE_all)
TP_all_WDH20_BLUE$Pedigree<-All_pheno_TPall$Pedigree[match(TP_all_WDH20_BLUE$taxa,All_pheno_TPall$Entry)]
TP_all_WDH20_BLUE$Family<-All_pheno_TPall$Family[match(TP_all_WDH20_BLUE$taxa,All_pheno_TPall$Entry)]

```

LD heatmap
```{r}

library(LDcorSV)
library(ggplot2)
library(reshape2)

ld_heatmap=function(df){
  ld <- as.matrix(round(df,0))
  
  if(c(-1,3,4) %in% ld){
    ld[which(ld==3)]=2
    ld[which(ld==4)]=2
    ld[which(ld== -1)]=0
  }
  
  LD <- LD.Measures(donnees=ld,  na.presence=F)
  #LD$loc1=as.character(LD$loc1); LD$loc2=as.character(LD$loc2)
  r2 <- matrix(0, nrow=ncol(df), ncol=ncol(df))
  r2[lower.tri(r2, diag=FALSE)] <- LD$r2
  r2 <- t(r2)
  r2 <- as.data.frame(round(r2, 5))
  diag(r2) <- 1
  r2[lower.tri(r2)] = NA
  rownames(r2)=colnames(df); colnames(r2)=rownames(r2)
  r_2=melt(as.matrix(r2), na.rm=T)
  
  graphic = ggplot(r_2, aes(Var2, Var1, fill = value))+
    geom_tile(color = "white")+
    scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       midpoint = 0.5, limit = c(0,1), space = "Lab", name="r2") +
    theme_classic() #+    ggtitle(paste("LD r2 from",colnames(r2)[1],"-", colnames(r2)[length(colnames(r2))], sep=" " ))
  return(graphic)
}

#Top significant markers
#Chr 5
#JHI-Hv50k-2016-311497 475342665
grep("JHI-Hv50k-2016-311497", colnames(myGD20))
#JHI-Hv50k-2016-316599 493408279
#BOPA1_5004-375 493200949
#Chr 4
#JHI-Hv50k-2016-229031 9491790
#Chr 2
#JHI-Hv50k-2016-10635057 7412613               





#QsD1 block
grep("Qsd1", colnames(myGD20))
grep("JHI-Hv50k-2016-308652", colnames(myGD20))


#TP1 5H marker
#TP2 5H marker
grep("JHI-Hv50k-2016-311497",colnames(myGD20))

#grep("JHI-Hv50k-2016-312094", colnames(myGD20))
grep("JHI-Hv50k-2016-311960", colnames(myGD20))




#to remove the marker labels on each axis, use theme_void() instead of theme_classic()
#test of some random markers
#TP1 5H marker

#replace with
grep("JHI-Hv50k-2016-316599", colnames(myGD20))

#TP2 5H marker
grep("JHI-Hv50k-2016-311497",colnames(myGD20))
ld_heatmap(myGD20[,c(5149:5150,5289,5441)])
###

ld_heatmap(myGD20[,c(5149:5150,5289,5441)]) + ggtitle('LD between Qsd1 marker \n and most significant hit at 5H') +
  theme(axis.text.x = element_text(angle = 90))+
  

  scale_x_discrete(label = c("Qsd1","308652","311497","316599"), breaks = c("Qsd1","JHI-Hv50k-2016-308652","JHI-Hv50k-2016-311497","JHI-Hv50k-2016-316599"))+
  scale_y_discrete(label = c("Qsd1","JHI-Hv50k-2016-308652","JHI-Hv50k-2016-311497","JHI-Hv50k-2016-316599"), breaks = c("Qsd1","JHI-Hv50k-2016-308652","JHI-Hv50k-2016-311497","JHI-Hv50k-2016-316599"))+theme(axis.text.x = element_text(angle = 90,size = 10))+
  xlab(NULL)+ylab(NULL)


```

```{r}
#Top significant markers
#Chr 5
#JHI-Hv50k-2016-311497 475342665
marker_list<-list('JHI-Hv50k-2016-311497','JHI-Hv50k-2016-316599','JHI-Hv50k-2016-229031','JHI-Hv50k-2016-316599','JHI-Hv50k-2016-106350','JHI-Hv50k-2016-198312')

#JHI-Hv50k-2016-316599 493408279
#BOPA1_5004-375 493200949
#Chr 4
#JHI-Hv50k-2016-229031 9491790
#Chr 2
#JHI-Hv50k-2016-106350 57412613  


trait<-"GE"
nrow(TP_all_WDH20_BLUE)
TP_all_WDH20_BLUE_all=TP_all_WDH20_BLUE
WDH20_PHS=Field_WDH20_BLUE_all[,c('taxa','PHS')]
WDH20_PHS
TP_all_WDH20_BLUE_all[TP_all_WDH20_BLUE_all$TP=="TP3",]$GE

#if we want to negate outliers, however it has not changed much
#TP_all_WDH20_BLUE_all$GE[TP_all_WDH20_BLUE_all$TP =="TP3"& TP_all_WDH20_BLUE_all$GE<=-0.25]<-NA

#TP_all_WDH20_BLUE_all$GE[TP_all_WDH20_BLUE_all$TP =="TP5"& TP_all_WDH20_BLUE_all$GE<=-0.15]<-NA
table(Graph_haplotype$Qsd1)

Graph_haplotype=function(marker,trait){
  df_marker=myGD20[,c('taxa',marker)]
  df_haplotype=as.data.frame(merge(TP_all_WDH20_BLUE_all,df_marker,by='taxa',all.x = TRUE))
  
  
  
  df_haplotype=df_haplotype[!is.na(df_haplotype[,marker]),]
  
  df_haplotype=df_haplotype[!df_haplotype[,marker]=="1",]
  df_haplotype[,marker]=as.factor(df_haplotype[,marker])
  df_haplotype_m=melt(df_haplotype, id.vars =c('taxa','TP','Pedigree',marker), measure.vars = c(trait))
  plot=ggplot(data=df_haplotype_m, aes(x=TP, y=value,color=df_haplotype_m[,marker])) +
    geom_boxplot()+
    #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
    theme_bw() + 
    ggtitle(paste0("Fixed Effect ",trait," Haplotype WDH"))+
    scale_fill_manual(values=c("#cccccc", "#636363")) +
    guides(color=guide_legend(paste(marker)))
    table=table(df_haplotype[,marker],df_haplotype[,'Pedigree'])
  plot
  return(plot)
  }
plot4=Graph_haplotype('JHI-Hv50k-2016-198312','GIscale')
plot4

#plot.PHS
TP1GEqsd1=Graph_haplotype_family('Qsd1','GE','TP1')
TP1GIqsd1=Graph_haplotype_family('Qsd1','GIscale','TP1')
TP1GE_5Dqsd1=Graph_haplotype_family('Qsd1','GE_5D','TP1')
figure<-grid.arrange(plot.PHS,TP1GEqsd1,TP1GIqsd1,TP1GE_5Dqsd1, 
             layout_matrix=rbind(c(1,1,2,2), c(3,3,4,4)))
anno<-annotate_figure(figure,
                top = text_grob(paste("Haplotype fixed effects for ",marker), face = "bold", size = 12),
                bottom = text_grob("Family", size = 12))
anno

Graph_haplotype_family=function(marker,trait,TP_value){

#plot=Graph_haplotype_family=function(marker,trait,TP){
df_marker=myGD20[,c('taxa',marker)]
df_haplotype=as.data.frame(merge(TP_all_WDH20_BLUE_all,df_marker,by='taxa',all.x = TRUE))
df_haplotype=df_haplotype[!is.na(df_haplotype[,marker]),]
df_haplotype=df_haplotype[!df_haplotype[,marker]=="1",]
df_haplotype[,marker]=as.factor(df_haplotype[,marker])
df_haplotype_m=df_haplotype[df_haplotype$TP==TP_value,]
df_haplotype_m=melt(df_haplotype, id.vars =c('taxa','Family',marker), measure.vars = c(trait))
plot=ggplot(data=df_haplotype_m, aes(x=Family, y=value,color=df_haplotype_m[,marker])) +
  geom_jitter()+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle(paste0(trait," of '",marker,"' for ",TP_value))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5,size=7),axis.text.x = element_text(angle = 20,size = 7),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
  
plot
return(plot)
}
Graph_haplotype_family_hist=function(marker,trait,TP_value,fam_value){

#plot=Graph_haplotype_family=function(marker,trait,TP){
df_marker=myGD20[,c('taxa',marker)]
df_haplotype=as.data.frame(merge(TP_all_WDH20_BLUE_all,df_marker,by='taxa',all.x = TRUE))
df_haplotype=df_haplotype[!is.na(df_haplotype[,marker]),]
df_haplotype=df_haplotype[!df_haplotype[,marker]=="1",]
df_haplotype[,marker]=as.factor(df_haplotype[,marker])
df_haplotype_m=df_haplotype[df_haplotype$TP==TP_value,]
df_haplotype_m=melt(df_haplotype_m, id.vars =c('taxa','Family',marker), measure.vars = c(trait))
df_haplotype_m=df_haplotype_m[df_haplotype_m$Family==fam_value,]
plot=ggplot(data=df_haplotype_m, aes(x=value,color=df_haplotype_m[,marker])) +
  geom_density(alpha=0.2)+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle(paste0(trait," of '",marker,"' for ",TP_value," of ",fam_value))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5,size=7),axis.text.x =element_blank(),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
  
plot
return(plot)
}
Graph_haplotype_family_hist(marker_list[[2]],'GIscale','TP4','Wintmalt')
marker_list
marker_list[[2]]
#############
marker_list[[4]]
Graph_haplotype=function(marker,trait){

df_marker=myGD20[,c('taxa',marker)]
df_haplotype=as.data.frame(merge(TP_all_WDH20_BLUE_all,df_marker,by='taxa',all.x = TRUE))



df_haplotype=df_haplotype[!is.na(df_haplotype[,marker]),]

df_haplotype=df_haplotype[!df_haplotype[,marker]=="1",]
df_haplotype[,marker]=as.factor(df_haplotype[,marker])
df_haplotype_m=melt(df_haplotype, id.vars =c('taxa','TP','Pedigree',marker), measure.vars = c(trait))
plot=ggplot(data=df_haplotype_m, aes(x=TP, y=value,color=df_haplotype_m[,marker])) +
  geom_boxplot()+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle(paste0("Fixed Effect ",trait," Haplotype WDH"))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste(marker)))
  table=table(df_haplotype[,marker],df_haplotype[,'Pedigree'])
plot
return(plot)
}
#plot.PHS
TP2GI_qsd1=Graph_haplotype_family('Qsd1','GIscale','TP2')
TP2GI_qsd1
TP2GI.311497=Graph_haplotype_family(marker_list[[1]],'GIscale','TP2')
TP2GE.311497=Graph_haplotype_family(marker_list[[1]],'GE','TP2')
TP3GI.311497=Graph_haplotype_family(marker_list[[1]],'GIscale','TP3')

figure<-grid.arrange(TP2GI_qsd1,TP2GI.311497,TP2GE.311497,TP3GI.311497, 
             layout_matrix=rbind(c(1,1,2,2), c(3,3,4,4)))
anno<-annotate_figure(figure,
                top = text_grob(paste("Comparison of Qsd1 and 50k-311497 Marker"), face = "bold", size = 12),
                bottom = text_grob("Family", size = 12))
anno











#############

marker_list[1]
marker<-marker_list[[1]]
marker<-'Qsd1'
phs<-merge(WDH20_PHS,TP_all_WDH20_BLUE_all[TP_all_WDH20_BLUE_all$TP=="TP1",],by='taxa',all = TRUE)
phs=phs[!is.na(phs$PHS),]

df_marker=myGD20[,c('taxa',marker)]
df_haplotype1=as.data.frame(merge(phs,df_marker,by='taxa',all.x = TRUE))

df_haplotype1

#df_haplotype=df_haplotype[!is.na(df_haplotype[,marker]),]

df_haplotype1=df_haplotype1[!df_haplotype1[,marker]=="1",]
df_haplotype1[,marker]=as.factor(df_haplotype1[,marker])

df_haplotype_m1=melt(df_haplotype1, id.vars =c('taxa','TP','Family',marker), measure.vars = c('PHS',trait))
df_haplotype_m1
plot.PHS=ggplot(data=df_haplotype_m1, aes(x=Family, y=value,color=df_haplotype_m1[,marker])) +
  geom_boxplot()+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle(paste0("PHS"))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5),axis.text.x = element_text(angle = 20,size = 10),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
plot.PHS

#Top significant markers
#Chr 5
#JHI-Hv50k-2016-311497 475342665

#JHI-Hv50k-2016-316599 493408279
#BOPA1_5004-375 493200949
#Chr 4
#JHI-Hv50k-2016-229031 9491790
#Chr 2
#JHI-Hv50k-2016-106350 57412613  
  

plot=Graph_haplotype('JHI-Hv50k-2016-311497','GIscale')
plot
plot2=Graph_haplotype('JHI-Hv50k-2016-316599','GIscale')
plot2
plot3=Graph_haplotype('Qsd1','GIscale')
plot3
plot4=Graph_haplotype('JHI-Hv50k-2016-229031','GIscale')
plot4
plot5=Graph_haplotype('JHI-Hv50k-2016-106350','GIscale')
plot5
```

```{r}
library(gridExtra)
get_legend<-function(myggplot){
  tmp <- ggplot_gtable(ggplot_build(myggplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  return(legend)
}

```

```{r}
marker<-'JHI-Hv50k-2016-311497'
trait<-'GE'
model_used<-geom_boxplot()
Graph_haplotype_pedigree('Qsd1','GE',geom_boxplot(),"BLUE")
Graph_haplotype_pedigree=function(marker,trait,model_used,model){
  model<-model
df_marker=myGD20[,c('taxa',marker)]
df_haplotype=as.data.frame(merge(TP_all_WDH20_BLUE_all,df_marker,by='taxa',all.x = TRUE))
df_haplotype=df_haplotype[!is.na(df_haplotype[,marker]),]

df_haplotype=df_haplotype[!df_haplotype[,marker]=="1",]
df_haplotype[,marker]=as.factor(df_haplotype[,marker])
df_haplotype_m=melt(df_haplotype, id.vars =c('taxa','TP','Family',marker), measure.vars = c(trait))
MeltTP=df_haplotype_m[df_haplotype_m$TP=="TP1",]
plot(legend)
###
plotT=ggplot(data=MeltTP, aes(x=Family, y=value,color=MeltTP[,marker])) +
  model_used+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 60))+
  ggtitle(paste0("TP1",trait))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste("Haplo ",marker)))

legend=get_legend(plotT)

#Timepoint 1

meltTP=as.data.frame(df_haplotype_m[df_haplotype_m$TP=="TP1",])

#Timepoint 1
meltTP1=df_haplotype_m[df_haplotype_m$TP=="TP1",]
plot1=ggplot(data=meltTP1, aes(x=Family, y=value,color=meltTP1[,marker])) +
  model_used+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 60))+
  ggtitle(paste0("TP1 ",trait))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
plot1
#Timepoint 2
meltTP=df_haplotype_m[df_haplotype_m$TP=="TP2",]
plot2=ggplot(data=meltTP, aes(x=Family, y=value,color=meltTP[,marker])) +
  model_used+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 60))+
  ggtitle(paste0("TP2 ",trait))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
plot3
#Timepoint 3
meltTP=df_haplotype_m[df_haplotype_m$TP=="TP3",]
plot3=ggplot(data=meltTP, aes(x=Family, y=value,color=meltTP[,marker])) +
  model_used+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 60))+
  ggtitle(paste0("TP3 ",trait))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
plot3
#Timepoint 4
meltTP=df_haplotype_m[df_haplotype_m$TP=="TP4",]
plot4=ggplot(data=meltTP, aes(x=Family, y=value,color=meltTP[,marker])) +
  model_used+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 60))+
  ggtitle(paste0("TP4 ",trait))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())
meltTP=df_haplotype_m[df_haplotype_m$TP=="TP5",]
plot5=ggplot(data=meltTP, aes(x=Family, y=value,color=meltTP[,marker])) +
  model_used+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  theme(axis.text.x = element_text(angle = 60))+
  ggtitle(paste0("TP5 ",trait))+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  guides(color=guide_legend(paste(marker)))+
  theme(plot.title = element_text(hjust = 0.5),legend.position = "none",axis.title.x=element_blank(),axis.title.y=element_blank())


#grid.arrange(blankPlot,ncol=3,nrow=2)
plot1
figure<-grid.arrange(plot2,plot3,plot4,plot5, 
             layout_matrix=rbind(c(1,1,2,2), c(3,3,4,4)))
#return(figure)



anno<-annotate_figure(figure,
                top = text_grob(paste("Haplotype",marker), face = "bold", size = 10),
               bottom = text_grob("Family", size = 12))
anno

plot(legend)




return(plot)



```
```{r}

"JHI-Hv50k-2016-312121"
Qsd1<-myGD20[,c('taxa','Qsd1')]

Hv_311497<-myGD20[,c('taxa','JHI-Hv50k-2016-311497')]
TP_all_WDH20_BLUE_all[,c(8)]
str(myGD20)
colnames(TP_all_WDH20_BLUE_all)
colnames(Hv_311497_haplotype)
Hv_311497_haplotype<-merge(TP_all_WDH20_BLUE_all,Hv_311497,by='taxa',all.x = TRUE)
table(Hv_311497_haplotype$`JHI-Hv50k-2016-311497`,Hv_311497_haplotype$Pedigree)
Qsd_haplotype<-merge(TP_all_WDH20_BLUE_all,Qsd1,by='taxa',all.x = TRUE)
table(Qsd_haplotype$Qsd1,Qsd_haplotype$Pedigree)
table(Qsd_haplotype[!Qsd_haplotype$Qsd1=="0" & !Qsd_haplotype$Qsd1=="2",]$Qsd1)


Qsd_haplotype<-Qsd_haplotype[!is.na(Qsd_haplotype$Qsd1),]
Qsd_haplotype=Qsd_haplotype[!Qsd_haplotype$Qsd1=="1",]

colnames(Qsd_haplotype)
Qsd_haplotype


Qsd_GE=melt(Qsd_haplotype, id.vars =c("taxa","TP","Qsd1"), measure.vars = c("GE"))
ggplot(data=Qsd_GE, aes(x=TP, y=value,color=Qsd1)) +
  geom_boxplot()+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle("Fixed Effect GE Haplotype WDH")+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  
  xlab("Qsd Haplotype") + ylab("BLUE GE") 

Qsd_GI=melt(Qsd_haplotype, id.vars =c("taxa","TP","Qsd1"), measure.vars = c("GIscale"))
ggplot(data=Qsd_GI, aes(x=TP, y=value,color=Qsd1)) +
  geom_boxplot()+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle("Fixed effect GI Qsd1 haplotype WDH")+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  
  xlab("Qsd Haplotype") + ylab("BLUE GI Scale") 

#HV
Hv_311497_haplotype$`JHI-Hv50k-2016-311497`<-as.factor(Hv_311497_haplotype$`JHI-Hv50k-2016-311497`)
Hv_311497_haplotype$`JHI-Hv50k-2016-311497`

Hv_311497_GE=melt(Hv_311497_haplotype, id.vars =c("taxa","TP","JHI-Hv50k-2016-311497"), measure.vars = c("GE"))
ggplot(data=Hv_311497_GE, aes(x=TP, y=value,color=`JHI-Hv50k-2016-311497`)) +
  geom_boxplot()+
  #facet_grid(cols=vars(dataset), scales="free", rows=vars(variable)) +
  theme_bw() + 
  ggtitle("Fixed Effect GE Haplotype WDH")+
  scale_fill_manual(values=c("#cccccc", "#636363")) +
  
  xlab("JHI-Hv50k-2016-311497 Haplotype") + ylab("BLUE GE")



```

#extra from genotype processing
```{r}
```{r}                 
#SD1 and SD2
ld_heatmap(myGD[,c(10780:10800, 13000:13075)])


c("JHI-Hv50k-2016-312217","JHI-Hv50k-2016-312121","JHI-Hv50k-2016-316817","JHI-Hv50k-2016-316934")

ld_heatmap(myGD20[,c(9043:9055,9290:9302)]) + ggtitle('LD between significant marker regions at 5H') +
  theme(axis.text.x = element_text(angle = 90))+
  
#opts(axis.title.x = theme_text(vjust=-2))+
  scale_x_discrete(label = c("312217","312121","316817","316934"), breaks = c("JHI-Hv50k-2016-312217","JHI-Hv50k-2016-312121","JHI-Hv50k-2016-316817","JHI-Hv50k-2016-316934"))+
  scale_y_discrete(label = c("JHI-Hv50k-2016-312217","JHI-Hv50k-2016-312121","JHI-Hv50k-2016-316817","JHI-Hv50k-2016-316934"), breaks = c("JHI-Hv50k-2016-312217","JHI-Hv50k-2016-312121","JHI-Hv50k-2016-316817","JHI-Hv50k-2016-316934"))+ 
  xlab(NULL)+ylab(NULL)

ld_heatmap(myGD20[,c(10785:10795,13000:13010,13055:13070)]) + ggtitle('LD between AlaAt, GA20Ox, and MKK3 markers') +
 
  scale_x_discrete(label = c('GA20Ox','MKK3','AlaAt'), 
                   breaks = c('SCRI_RS_121526','JHI-Hv50k-2016-367253','JHI-Hv50k-2016-308584'))+
  scale_y_discrete(label = c('GA20Ox','MKK3','AlaAt'), 
                   breaks = c('SCRI_RS_121526','JHI-Hv50k-2016-367253','JHI-Hv50k-2016-308584'))+ 
  xlab('470-670 Mb, selected markers')+ylab(NULL)

```