---
title: "post_GWAS_plots"
author: "Karl Kunze"
date: "2023-03-27"
output: html_document
editor_options: 
  chunk_output_type: console
---
#post gwas plotting
```{r}
library(dplyr)
library(asreml)

#2020
load("data/Genotype_data/ASRgwas_results/Single_TP_GI_2020_results.Rdata")
load("data/Genotype_data/ASRgwas_results/Single_TP_GE_2020_results.Rdata")
#2021
load("data/Genotype_data/ASRgwas_results/Single_TP_GI_2021_results.Rdata")
load("data/Genotype_data/ASRgwas_results/Single_TP_GE_2021_results.Rdata")
#combined
load("data/Genotype_data/ASRgwas_results/Single_TP_GI_cpmbined_results.Rdata")#GI_STP_combined,
load("data/Genotype_data/ASRgwas_results/Single_TP_GE_combined_results.Rdata")#GE_STP_combined



convert<-function(df){
var_test<-df[[1]]$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(),fdr.pvalues=p.adjust(p.value,method="fdr"))

var_test$ID<-df[[1]]$trait

#env_n<-length(table(BSR.df$Env))
for(i in (1:length(df))-1){
#i<-3
app<-df[[1+i]]$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(),fdr.pvalues=p.adjust(p.value,method="fdr"))
app
app$ID<-df[[i+1]]$trait
  var_test<- rbind(var_test,app)
}
return(var_test)
}

str(GE_2020_GWAS)
GE_2020_GWAS[[2]]$gwas.sel

convert<-function(df){
var_test<-df[[1]]$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(.),fdr.pvalues=p.adjust(p.value,method="fdr"))

var_test$ID<-df[[1]]$trait

#env_n<-length(table(BSR.df$Env))
for(i in (2:length(df))-1){
#i<-3
app<-df[[1+i]]$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(.),fdr.pvalues=p.adjust(p.value,method="fdr"))
app
app$ID<-df[[i+1]]$trait
  var_test<- rbind(var_test,app)
}
return(var_test)
}



GE20<-convert(GE_2020_GWAS)
table(GE20$ID)
GI20<-convert(GI_2020_GWAS)
table(GI20$ID)
GE21<-convert(GE_2021_GWAS)
table(GE21$ID)
GI21<-convert(GI_2021_GWAS)
table(GI21$ID)
GI_C<-convert(GI_STP_combined)
table(GI_C$ID)
GE_C<-convert(GE_STP_combined)
library(stringr)
STP_all<-rbind(GE20,GI20,GE21,GI21,GI_C,GE_C)%>%mutate(notes="none")

STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$notes<-"GE-2021-PM_96: did_not_converge"
STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$ID<-"GE-2021-PM_96"


STP_all[grep("GI_0-2021-PM_152",STP_all$ID),]$ID<-"GI0-2021-PM_152"
STP_all<-STP_all%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),year=str_split_i(ID,"-",2))%>%mutate(log10PVal=-log10(p.value))%>%arrange(year,trait,PM)
table(STP_all$ID)
###

WTP<-STP_all%>%dplyr::rename(Chromosome=chrom,Position=pos)

table_chr<-WTP%>%dplyr::select(Chromosome,Position)%>%arrange(Chromosome,Position)%>%unique()%>%mutate(ordinal=1:n())
table_chr
library(tidyr)
names(table_chr)
table_chr$Chromosome
table_chr_min<-table_chr%>%group_by(Chromosome)%>%summarise(min=min(Position),min_ord=min(ordinal))
table_chr_min
table_chr_max<-table_chr%>%group_by(Chromosome)%>%summarise(max=max(Position),max_ord=max(ordinal))
table_chr<-merge(table_chr_min,table_chr_max,by="Chromosome")
Chr_breaks=table_chr$max_ord-table_chr$min_ord
ChrTable<-Chr_breaks
ChrTable

chrLabel = c(1:7, 'UN')
winterOrdinalBreaks = c(ChrTable[1]/2)

winterChrLines = ChrTable[1]#for the first position

for (i in 2:8){
 winterOrdinalBreaks[i] = sum(ChrTable[1:i-1])+ChrTable[i]/2
 winterChrLines[i] = sum(ChrTable[1:i])
}

WinterPerTPGWAS2<-STP_all%>%ungroup()
table(WTP$ID)



library(stringr)
winterChrLines
library(ggplot2)
table(WinterPerTPGWAS2$PM)
0.05/9238

WinterPerTPGWAS2%>%ungroup()%>%mutate(trait=ifelse(trait=="GI0","GI",trait))%>% ggplot(aes(ordinal, log10PVal, color = as.factor(PM), shape = year))+geom_point()+
  geom_vline(xintercept = winterChrLines, color = 'black')+
 # geom_vline(xintercept = 4780, color = 'red',linetype='longdash')+
  #annotate(geom= 'text', x = 4780, y = 30, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+
  scale_x_continuous(label = c("1H","2H", "3H", "4H", "5H", "6H", "7H", "UN"),
                     breaks = winterOrdinalBreaks)+
  ylab('-log(p-value)')+xlab('Chromosome')+ geom_hline(yintercept = -log10(5.412427e-06) )+theme_bw()+
 facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()
a=-log10(1e-5)
a
CH3<-WinterPerTPGWAS2 %>%filter(log10PVal>a)%>%filter(chrom=="3")
CH3<-colnames(WinterGD)[which(CH3$marker%in%colnames(WinterGD))]

mar<-WinterPerTPGWAS %>% arrange(P.value) %>% ungroup()%>% select(SNP, Chromosome, Position) %>% unique()


```

#sig markers
```{r}
sig_markers(GE_2020)
sig_markers<-function(df){
  #df<-GE_2020
  names(sel)
  sel<-data.frame(1,1)
  sel
  ?data.frame()
  sel
sel<-df[[2]]$gwas.sel

sel$ID<-df[[1]]$trait
sel
#env_n<-length(table(BSR.df$Env))
for(i in (1:length(df))-1){
#i<-3
abb<-df[[i+1]]$gwas.sel

abb$ID<-df[[i+1]]$trait
  sel_test<- rbind(sel,abb)
}
return(sel_test)
}

```