---
title: "post_GWAS_plots"
author: "Karl Kunze"
date: "2023-03-27"
output: html_document
editor_options: 
  chunk_output_type: console
---
#post gwas plotting
```{r}
library(dplyr)
library(asreml)
getwd()
setwd("C:/Users/kars8/git/BarleyGermination-KK/")
#2020
load("data/Genotype_data/ASRgwas_results/Single_TP_GI_2020_results.Rdata")

load("data/Genotype_data/ASRgwas_results/Single_TP_GE_2020_results.Rdata")
#2021
load("data/Genotype_data/ASRgwas_results/Single_TP_GI_2021_results.Rdata")
str(GI_2021_GWAS)
load("data/Genotype_data/ASRgwas_results/Single_TP_GE_2021_results.Rdata")
#combined
load("data/Genotype_data/ASRgwas_results/Single_TP_GI_cpmbined_results.Rdata")#GI_STP_combined,
load("data/Genotype_data/ASRgwas_results/Single_TP_GE_combined_results.Rdata")#GE_STP_combined
#phs
load("data/Genotype_data/ASRgwas_results/PHS_GWAS_results.Rdata")

rep_str = c('GI'='GE')
library(stringr)
GE_STP_combined[[5]]$trait<-str_replace_all(GE_STP_combined[[5]]$trait,rep_str)
GE_STP_combined[[3]]$trait<-str_replace_all(GE_STP_combined[[3]]$trait,rep_str)
GE_STP_combined[[4]]$trait<-str_replace_all(GE_STP_combined[[4]]$trait,rep_str)

```
# need to get phenotypic distrbution file
```{r}
load("data/WMB_DH_Germination_data2020_2021.RData")#DH2020_2021
library(tidyr)
sum<-DH2020_2021%>%mutate(GI_0=GI/GE)%>%pivot_longer(values_to = "value",names_to = "trait",cols = c(GE,GI,GI_0))%>%group_by(Year,trait,PM_date)%>%reframe(mean=format(round(mean(value,na.rm=TRUE),3),nsmall=2),range=paste0(format(round(min(value,na.rm=TRUE),2),nsmall=2),"-",format(round(max(value,na.rm=TRUE),2),nsmall=2)))%>%filter(!trait=="GI_0"|PM_date%in%c("152","96")|Year=="2021")%>%filter(!trait=="GI_0"|PM_date%in%c("152","96"))
#View(sum)
sum_across<-DH2020_2021%>%pivot_longer(values_to = "value",names_to = "trait",cols = c(GE,GI))%>%filter(!PM_date%in%c("12","33","68"))%>%mutate(Year="2020/2021")%>%group_by(Year,trait,PM_date)%>%reframe(mean=format(round(mean(value,na.rm=TRUE),3),nsmall=2),range=paste0(format(round(min(value,na.rm=TRUE),2),nsmall=2),"-",format(round(max(value,na.rm=TRUE),2),nsmall=2)))
sum$trait
sum_stats=rbind(sum,sum_across)%>%mutate(PM_date=as.numeric(as.character(PM_date)))
sum_stats$trait<-gsub("GI_0","GI0",sum_stats$trait)

gvH2_fromList<-function(df){
  
H2<-df[[1]]$heritability


varcomp<-summary(df[[1]]$mod)$varcomp
o2g<-varcomp[grep("GID",rownames(varcomp)),"component"]


var_est<-data.frame(trait=df[[1]]$trait,o2g=round(o2g,4),h2=round(H2$h2.vc[1,"Estimate"],4),h2_PEV=round(H2$h2.pev[1,"Estimate"],4))

#env_n<-length(table(BSR.df$Env))
for(i in (2:length(df))){

  H2<-df[[i]]$heritability


varcomp<-summary(df[[i]]$mod)$varcomp
o2g<-varcomp[grep("GID",rownames(varcomp)),"component"]


app<-data.frame(trait=df[[i]]$trait,o2g=round(o2g,4),h2=round(H2$h2.vc[1,"Estimate"],4),h2_PEV=round(H2$h2.pev[1,"Estimate"],4))


  var_est<- rbind(var_est,app)

}
return(var_est)
}


gvH2=rbind(gvH2_fromList(GE_2020_GWAS),gvH2_fromList(GI_2020_GWAS),gvH2_fromList(GE_2021_GWAS),gvH2_fromList(GI_2021_GWAS),gvH2_fromList(GE_STP_combined),gvH2_fromList(GI_STP_combined))%>%mutate(ID=trait)
#adding phs
bn<-data.frame(GWAS_phs$trait,summary(GWAS_phs$mod)$varcomp[1,1],GWAS_phs$heritability$h2.vc[1],GWAS_phs$heritability$h2.pev[1],GWAS_phs$trait);colnames(bn)<-names(gvH2)
gvH2<-gvH2%>%add_row(bn)

gvH2$ID<-gsub("GI_0","GI0",gvH2$ID)

gvH2[grep("did",gvH2$ID),]$ID<-"GE-2021-PM_96"
gvH2<-gvH2%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),Year=str_split_i(ID,"-",2))%>%arrange(Year,trait,PM)%>%rename(PM_date=PM)

sum_table<-gvH2%>%left_join(sum_stats,by=c("Year","trait","PM_date"))%>%arrange(Year,trait,PM_date)%>%relocate(ID,trait,Year,PM_date,.before=1)%>%relocate(mean,range,.before = o2g)
mr_phs<-DH2020_2021%>%dplyr::select(phs,Location,Year)%>%reframe(mean=round(mean(phs,na.rm=TRUE),2),range=paste0(min(phs,na.rm=TRUE),"-",max(phs,na.rm = TRUE)))

sum_table[sum_table$trait=="PHS",c("mean","range")]<-mr_phs[1,]
sum_table[sum_table$trait=="PHS",c("mean","range")]
#View(sum_table)
save(sum_table,file="ms/Rdata/genetic_variance_h2.Rdata")

```
```{r}
df3<-DH2020_2021%>%mutate(GID=toupper(taxa))%>%filter(!Location=="Spring")%>%mutate(type="base_data_single_step")

df3[!df3$PM_date==5,]$phs<-0
df_phs<-df3%>%filter(PM_date==5)%>%dplyr::select(PM_date,GID,phs,Year)%>%group_by(Year,GID)%>%summarise(phs_m=mean(phs,na.rm=TRUE))%>%mutate(PM_date=as.factor(5))%>%ungroup()
df_phs
df3
df3$PM_date
cor<-df3%>%dplyr::select(GID,SourcePLOT,PM_date,phs,Year,GE,GI)%>%ungroup()%>%unique()%>%arrange(PM_date)%>%group_by(PM_date,Year,GID)%>%reframe(GI=mean(GI),GE=mean(GE),phs=mean(phs))%>%mutate(PM_date=paste("PM",as.character(PM_date)))%>%pivot_wider(names_from ="PM_date",values_from = c("GE","GI","phs"))
cor

phs_5<-cor[grep("phs",colnames(cor))]$"phs_PM 5"

vb<-cor%>%dplyr::select(!grep("phs",(colnames(cor))))%>%cbind(phs=phs_5)%>%relocate(phs,.after=GID)
vb

names(vb)
vb_m<-as.matrix(vb[,-c(1,2)])
vb_m
pheno_cor<-round(cor(vb_m,use="pairwise.complete.obs",method="spearman"),2)

pheno_object<-Hmisc::rcorr(as.matrix(pheno_cor),type="pearson")

#?rcorr()
diag(pheno_object$r) <- NA 
pheno_object$r[upper.tri(pheno_object$r)] <- NA
diag(pheno_object$P) <- NA 
pheno_object$P[upper.tri(pheno_object$P)] <- NA

round(pheno_object$r,2)%>%
  as.data.frame() %>%
  tibble::rownames_to_column("Variable") %>%
  flextable::flextable(theme="zebra")

```
#need to extract values here for genetic correlations
```{r}



BLUP_vals<-function(df){
#df<-GE_2020_GWAS

BLUP_var<-df[[1]]$BLUPS$pvals

BLUP_var$ID<-df[[1]]$trait

#env_n<-length(table(BSR.df$Env))
for(i in (2:length(df))-1){
#i<-3
app<-df[[1+i]]$BLUPS$pvals%>%unique()
app$ID<-df[[i+1]]$trait
  BLUP_var<- rbind(BLUP_var,app)
}
return(BLUP_var)
}


GI20_BLUPS<-BLUP_vals(GI_2020_GWAS)
GE20_BLUPS<-BLUP_vals(GE_2020_GWAS)

GI21_BLUPS<-BLUP_vals(GI_2021_GWAS)%>%unique()
dim(GI21_BLUPS[GI21_BLUPS$ID=="GI-2021-PM_68",])
table(GI21_BLUPS$ID)
GE21_BLUPS<-BLUP_vals(GE_2021_GWAS)

GI_C_BLUPS<-BLUP_vals(GI_STP_combined)


GE_C_BLUPS<-BLUP_vals(GE_STP_combined)
#accidently labeled GE with GI here, see if I can change this


PHS_BLUP<-GWAS_phs$BLUPS$pvals

PHS_BLUP$ID<-GWAS_phs$trait
GE21_BLUPS[grep("did",GE21_BLUPS$ID),]$ID<-"GE-2021-PM_96"

STP_all_BLUP<-rbind(GE20_BLUPS,GI20_BLUPS,GE21_BLUPS,GI21_BLUPS%>%unique(),GI_C_BLUPS,GE_C_BLUPS,PHS_BLUP)%>%mutate(notes="BLUP values")
STP_all_BLUP$ID<-gsub("GI_0","GI",STP_all_BLUP$ID)

STP_all_BLUP<-STP_all_BLUP%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),year=str_split_i(ID,"-",2))%>%arrange(year,trait,PM)
PM2021<-STP_all_BLUP%>%filter(year=="2021",PM%in%c("12","33","68"))%>%droplevels()
table(PM2021$PM)
comb<-STP_all_BLUP%>%filter(year=="2020/2021")%>%rbind(PM2021)%>%mutate(year="2020/2021")%>%arrange(PM)
comb[comb$PM==3,]$PM<-5
test<-comb%>%mutate(PM_date=paste("PM",as.character(PM)))%>%dplyr::select(!c(status,ID,std.error))%>%pivot_wider(names_from ="trait",values_from = c("predicted.value"))%>%dplyr::select(!PM)%>%pivot_wider(names_from = "PM_date",values_from = c("PHS","GE","GI"))
#View(test)
phs<-test%>%dplyr::select(grep("PHS",colnames(test)))%>%dplyr::select("PHS_PM 5")%>%rename(phs="PHS_PM 5")
#View(test)
gen_cor<-test%>%dplyr::select(!grep("PHS",colnames(.)))%>%cbind(phs)%>%relocate(phs,.after ="year" )
#View(gen_cor)
library(Hmisc)
?rcorr()
gen_cor[,"GI_PM 33"]
#

rcor_object<-rcorr(as.matrix(gen_cor[,-c(1:3)]))
rcor_object$r[lower.tri(rcor_object$r)] <- NA
rcor_object$P[lower.tri(rcor_object$P)] <- NA
rcor_object$r[lower.tri(rcor_object$r)]
new <- pheno_object$r
#View(new)

new[upper.tri(new)] <- rcor_object$r[upper.tri(rcor_object$r)]

ne_p<-pheno_object$P
pheno_object$P
ne_p[upper.tri(ne_p)] <- rcor_object$P[upper.tri(rcor_object$P)]
#View(ne_p)
#View(object$r)
library(flextable)
save(new,file="ms/Rdata/geno_corr.Rdata")
save(ne_p,file="ms/Rdata/p_corr.Rdata")
round(new,2)%>%
  as.data.frame() %>%
  tibble::rownames_to_column("Variable") %>%flextable()
 
round(ne_p,2)%>%
  as.data.frame() %>%
  tibble::rownames_to_column("Variable") %>%
  flextable::flextable(theme="zebra")



```

```{r}
convert<-function(df){
var_test<-df[[1]]$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(),fdr.pvalues=p.adjust(p.value,method="fdr"))

var_test$ID<-df[[1]]$trait

#env_n<-length(table(BSR.df$Env))
for(i in (1:length(df))-1){
#i<-3
app<-df[[1+i]]$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(),fdr.pvalues=p.adjust(p.value,method="fdr"))
app
app$ID<-df[[i+1]]$trait
  var_test<- rbind(var_test,app)
}
return(var_test)
}






GE20<-convert(GE_2020_GWAS)%>%unique()

table(GE20$ID)
GI20<-convert(GI_2020_GWAS)%>%unique()
table(GI20$ID)
GE21<-convert(GE_2021_GWAS)%>%unique()
table(GE21$ID)
GI21<-convert(GI_2021_GWAS)%>%unique()
table(GE21$ID)

GI_C<-convert(GI_STP_combined)%>%unique()


table(GI_C$ID)
GE_C<-convert(GE_STP_combined)%>%unique()
#accidently labeled GE with GI here, see if I can change this

GE_C$ID<-gsub("GI","GE",GE_C$ID)

PHS<-GWAS_phs$gwas.all%>%arrange(chrom,pos)%>%mutate(ordinal=row_number(.),fdr.pvalues=p.adjust(p.value,method="fdr"))

PHS$ID<-GWAS_phs$trait

STP_all<-rbind(GE20,GI20,GE21,GI21,GI_C,GE_C,PHS)%>%mutate(notes="none")%>%group_by(ID)%>%arrange(chrom,pos)%>%mutate(ordinal=1:n())
hist(STP_all$pos)
#STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge"&STP_all$p.value<0.0004&!is.na(STP_all$p.value),]$p.value<-1

save(STP_all,file="data/GWA_results/STP_all.Rdata")
```


```{r}
#STP_all
table_chr<-STP_all%>%ungroup()%>%dplyr::select(chrom,pos,ordinal)%>%arrange(chrom,pos)%>%unique()
table_chr
detach("package:Hmisc",unload=TRUE)

table_chr_min<-table_chr%>%group_by(chrom)%>%summarise(min=min(pos),min_ord=min(ordinal))
table_chr_max<-table_chr%>%group_by(chrom)%>%summarise(max=max(pos),max_ord=max(ordinal))
table_chr<-merge(table_chr_min,table_chr_max,by="chrom")%>%tidyr::drop_na(chrom)
table_chr

STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$notes<-"GE-2021-PM_96: did_not_converge"
STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$ID<-"GE-2021-PM_96"
table(STP_all$ID)

STP_all$ID<-gsub("GI_0","GI0",STP_all$ID)

STP_all<-STP_all%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),year=str_split_i(ID,"-",2))%>%mutate(log10PVal=-log10(p.value))%>%arrange(year,trait,PM)
WinterPerTPGWAS2<-STP_all
hist(STP_all$fdr.pvalues)
###

detach("package:Hmisc",unload = TRUE)

Chr_breaks=table_chr$max_ord-table_chr$min_ord
ChrTable<-Chr_breaks
ChrTable

chrLabel = c(1:7, 'UN')
winterOrdinalBreaks = c(ChrTable[1]/2)

winterChrLines = ChrTable[1]#for the first position
winterChrLines 
for (i in 2:8){
 winterOrdinalBreaks[i] = sum(ChrTable[1:i-1])+ChrTable[i]/2
 winterChrLines[i] = sum(ChrTable[1:i])
}
winterChrLines

table(STP_all$ID)
winterOrdinalBreaks

WinterPerTPGWAS2<-STP_all%>%ungroup()
library(stringr)
winterChrLines
library(ggplot2)
table(WinterPerTPGWAS2$ID)

hist(WinterPerTPGWAS2$log10PVal)

hist(WinterPerTPGWAS2[WinterPerTPGWAS2$ID=="GE-2021-PM_96"&WinterPerTPGWAS2$log10PVal&!is.na(WinterPerTPGWAS2$log10PVal),]$log10PVal)
ord_list<-STP_all[STP_all$ID=="GI-2020/2021-PM_152",c("chrom","pos","marker","ordinal"),]
WinterPerTPGWAS2<-STP_all%>%ungroup()%>%mutate(ordinal=plyr::mapvalues(marker,from=ord_list$marker,to=ord_list$ordinal),ordinal=as.numeric(ordinal))
#this GWAS TP is giving weird assocations, so I just removed it

#False discovery rate
#all P values are FDR adjusted, to find the p-value of the FDR rate, I found the max p value for which the FDR p value is less than 0.05, code is below
WinterPerTPGWAS2[WinterPerTPGWAS2$p.value>0.95&!is.na(WinterPerTPGWAS2$p.value),]$fdr.pvalues<-1
FDR_rate<-max(WinterPerTPGWAS2[WinterPerTPGWAS2$fdr.pvalues<0.05,]$p.value,na.rm=TRUE)
FDR_rate
#even the line is slightly more conservative than the true FDR rate since the nearest FDR value below 0.05 is 0.03. This is important because while we do not want false positive(QQ plots should be used for this), we also shouldn't be over stringent with Bonferonni cutoff, we elimate
ALaAT_ord<-unique(WinterPerTPGWAS2[WinterPerTPGWAS2$marker=="Qsd1",]$ordinal)
ALaAT_ord
ALaAT_ord<-median(ALaAT_ord)


WinterPerTPGWAS2%>%ungroup()%>%filter(!trait=="PHS")%>%mutate(trait=ifelse(trait=="GI0","GI",trait))%>% ggplot(aes(ordinal, log10PVal, color = as.factor(PM), shape = year))+geom_point()+ggtitle(label="Genome wide association for Germination Energy and Index \n across timepoints and years")+
  geom_vline(xintercept = winterChrLines, color = 'black')+
 geom_vline(xintercept = ALaAT_ord, color = 'red',linetype='longdash')+
  annotate(geom= 'text', x = ALaAT_ord, y = 12, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+
  scale_x_continuous(label = c("1H","2H", "3H", "4H", "5H", "6H", "7H", "UN"),
                     breaks = winterOrdinalBreaks)+
  ylab('-log(p-value)')+xlab('Chromosome')+ geom_hline(yintercept = -log10(FDR_rate),linetype="longdash")+geom_hline(yintercept = -log10(0.05/9658) ,alpha=0.2,color="red")+theme_bw()+
 facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()
  
WinterPerTPGWAS2$fdr.pvalues_round<-round(WinterPerTPGWAS2$fdr.pvalues,3)
WinterPerTPGWAS2$fdr.pvalues_round



mar<-WinterPerTPGWAS2 %>% arrange(P.value) %>% ungroup()%>% select(SNP, Chromosome, Position) %>% unique()
WinterPerTPGWAS2
WinterPerTPGWAS2%>%filter(fdr.pvalues<0.05)

```

#sig markers looping
```{r}
sig_markers<-function(df){
bj<-c(df[[1]]$trait,!is.null(df[[1]]$gwas.sel),1)
names(bj)<-c("trait","sign_markers","i_value")
for(i in 2:length(df)){
  bo<-c(df[[i]]$trait,!is.null(df[[i]]$gwas.sel),i)
  bj<-rbind(bo,bj)

}
return(bj)
}
sig_all<-function(df){
sig_markers<-function(df){
bj<-c(df[[1]]$trait,!is.null(df[[1]]$gwas.sel),1)
names(bj)<-c("trait","sign_markers","i_value")
for(i in 2:length(df)){
  bo<-c(df[[i]]$trait,!is.null(df[[i]]$gwas.sel),i)
  bj<-rbind(bo,bj)

}
return(bj)
}

vb<-sig_markers(df)%>%as.data.frame()%>%mutate(i_value=as.numeric(i_value))%>%filter(sign_markers==TRUE)%>%arrange(i_value)

ro<-df[[vb[1,"i_value"]]]$gwas.sel
ro$trait<-df[[vb[1,"i_value"]]]$trait


for ( i in c(vb[2:nrow(vb),"i_value"])){
rt<-df[[i]]$gwas.sel
rt$trait<-df[[i]]$trait
    ro<-rbind(ro,rt)                   
}
return(ro)
}

GE20_sig<-sig_all(GE_2020_GWAS)
GE21_sig<-sig_all(GE_2021_GWAS)
GI21_sig<-sig_all(GI_2021_GWAS)
GI20_sig<-sig_all(GI_2020_GWAS)
GI_sig<-sig_all(GI_STP_combined)
GE_sig<-sig_all(GE_STP_combined)

GWAS_phs$gwas.sel$trait<-GWAS_phs$trait

names(GWAS_phs)
names(GWAS_phs$gwas.sel)
names(GI21_sig)
phs_df<-GWAS_phs$gwas.sel




sig<-rbind(GE20_sig,GE21_sig,GE_sig,GI20_sig,GI21_sig,GI20_sig,phs_df)%>%filter(!trait%in%c("GE-2021-PM_96: did_not_converge"))%>%mutate(ID=trait)
sig


sig$ID<-gsub("GI_0","GI0",sig$ID)

sig<-sig%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),year=str_split_i(ID,"-",2))%>%mutate(log10PVal=-log10(p.value))%>%arrange(year,trait,PM)
sig
markers1<-sig%>%filter(p.value<FDR_rate)
FDR_rate
t<-WinterPerTPGWAS2%>%filter(p.value<0.0003872807)%>%dplyr::select(names(markers1))

markers=rbind(markers1,t)%>%unique()%>%filter(maf>0.05)

#View(markers)
markers
library(readxl)
getwd()
Genetic_map <- read_excel("/home/karl/git/naked_barley_GWAS/data/Genetic/Genetic_map.xlsx")%>%dplyr::select(snp,"physical map chr","genetic map chr",cM)%>%rename(SNP=snp)%>%mutate(cM= if_else(is.na(cM), lead(cM), cM)) %>%tidyr::fill(cM,.direction="downup")%>%mutate(SNP=gsub("-","_",SNP))%>%rename(marker=SNP,chrom="physical map chr",chromG="genetic map chr")%>%mutate(chrom=as.numeric(substr(chrom,1,1)),chromG=as.numeric(substr(chromG,1,1)))



marker_table<-markers%>%filter(maf>0.05,p.value<FDR_rate)%>%arrange(chrom,pos,year,PM)%>%left_join(Genetic_map,by=c("chrom","marker"))%>%unique()%>%reframe(.by = c(chrom,marker),pos_range=ifelse(min(pos)==max(pos),max(pos),paste0(min(pos),"-",max(pos))),cm_range=ifelse(min(cM)==max(cM),min(cM),paste0(min(cM),"-",max(cM))),PM_dates = toString(unique(PM)),trait=toString(unique(trait)),year=toString(unique(year)),testmin=round(min(maf),3),testmax=round(max(maf),3),PVEmin=round(min(expl.var),3),PVEmax=round(max(expl.var),3),Pvalue=max(round(p.value,4)))%>%tidyr::drop_na(chrom)
markers$p.value

```
#LD analysis
```{r}
load("data/Genotype_data/wmb_file.Rdata")
load("data/Genotype_data/numeric_raw.Rdata")
WinterGM=wmbGAPIT$GM;WinterGD=as.data.frame(wmbGAPIT$GD)
dim(WinterGM)
library(LDheatmap)
# install.packages("snpStats")
# install.packages("genetics")
#removed from CRAN library, download from archive
#https://cran.r-project.org/src/contrib/Archive/LDheatmap/

rgb.palette <- colorRampPalette(rev(c("blue", "orange", "red")), space = "rgb")

### Chromosome 1
gdt<-WinterGD%>%dplyr::select(marker_table[marker_table$chrom=="1",]$marker)%>%as.matrix()
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="1",]$marker))
gmt
min_marker<-gmt[which(gmt$Position==min(gmt$Position))+1,]

max_marker<-gmt[which(gmt$Position==max(gmt$Position))-1,]
max_marker$Position-min_marker$Position
?as()
library(snpStats)
gdat<-as(gdt,"SnpMatrix")
gdat
LDheatmap(gdat,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 1",SNP.name = c(min_marker$SNP,max_marker$SNP))

gdt0<-gdt[,c(1:3)]
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="1",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")
LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 1",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)


CH1_LD_groupA=data.frame(SNP=gmt$SNP,Ch="1",group="A",Position=gmt$Position,ld=0.95)


#they are all in the same LD block, except for the outer two markers
```
```{r}
#CH 2 

marker_table[marker_table$chrom=="2",]$marker
gdt<-WinterGD%>%dplyr::select(marker_table[marker_table$chrom=="2",]$marker)%>%as.matrix()
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="2",]$marker))

min_marker<-gmt[which(gmt$Position==min(gmt$Position)),]
gmt
max_marker<-gmt[which(gmt$Position==max(gmt$Position)),]
max_marker$Position-min_marker$Position
gdat<-as(gdt,"SnpMatrix")
dim(gdat)
max_marker$SNP
LDheatmap(gdat,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 2",SNP.name = c(min_marker$SNP,max_marker$SNP),add.map = FALSE,text=FALSE)
#group A
gdt0<-gdt[,c(8:29)]
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="2",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")
LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 2",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)


CH2_LD_groupA=data.frame(SNP=gmt$SNP,Ch="2",group="A",Position=gmt$Position,ld=0.95)
#group B
# gdt0<-gdt[,c(26:94)]
# gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="2",]$marker))
# gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
# gdat0<-as(gdt0,"SnpMatrix")
# LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 2",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)
# 
# 
# CH2_LD_groupB=data.frame(SNP=gmt$SNP,Ch="2",group="B",Position=gmt$Position,ld=0.95)


CH2_LD=rbind(CH2_LD_groupA)

```
### 3H
```{r}
marker_table[marker_table$chrom=="3",]$marker
gdt<-WinterGD%>%dplyr::select(marker_table[marker_table$chrom=="3",]$marker)%>%as.matrix()
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="3",]$marker))

min_marker<-gmt[which(gmt$Position==min(gmt$Position)),]
gmt
max_marker<-gmt[which(gmt$Position==max(gmt$Position)),]
max_marker$Position-min_marker$Position
gdat<-as(gdt,"SnpMatrix")
max_marker$SNP
LDheatmap(gdat,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 3",SNP.name = c(min_marker$SNP,max_marker$SNP),add.map = FALSE,text=FALSE)
##3a
gdt0<-gdt[,c(10:18)]
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="3",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")
LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 3",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)


CH3_LD_groupA=data.frame(SNP=gmt$SNP,Ch="3",group="A",Position=gmt$Position,ld=0.95)


```
### 4H
```{r}
marker_table[marker_table$chrom=="4",]$marker
gdt<-WinterGD%>%dplyr::select(marker_table[marker_table$chrom=="4",]$marker)%>%as.matrix()
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="4",]$marker))

min_marker<-gmt[which(gmt$Position==min(gmt$Position)),]
gmt
max_marker<-gmt[which(gmt$Position==max(gmt$Position)),]
max_marker$Position-min_marker$Position
gdat<-as(gdt,"SnpMatrix")
max_marker$SNP
LDheatmap(gdat,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 4",SNP.name = c(min_marker$SNP,max_marker$SNP),add.map = FALSE,text=FALSE)

dim(gdt)

gdt0<-gdt[,c(3:17)]

gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="4",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")

LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 4",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)

gmt$SNP
CH4_LD_groupA_real=data.frame(SNP=gmt$SNP,Ch="4",group="A",Position=gmt$Position,ld=0.93)
#4b
# gdt0<-gdt[,c(4:7)]
# 
# gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="4",]$marker))
# gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
# gdat0<-as(gdt0,"SnpMatrix")
# 
# LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 4",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)
# 
# gmt$SNP
# CH4_LD_groupB_real=data.frame(SNP=gmt$SNP,Ch="4",group="B",Position=gmt$Position,ld=0.93)
CH4_LD<-rbind(CH4_LD_groupA_real)

```

### 5H
```{r}
gdt<-WinterGD%>%dplyr::select(marker_table[marker_table$chrom=="5",]$marker)%>%as.matrix()
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="5",]$marker))

min_marker<-gmt[which(gmt$Position==min(gmt$Position)),]
gmt
max_marker<-gmt[which(gmt$Position==max(gmt$Position)),]
max_marker$Position-min_marker$Position

gdat<-as(gdt,"SnpMatrix")

LDheatmap(gdat,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 5",SNP.name = c(min_marker$SNP,max_marker$SNP,"Qsd1"),add.map = FALSE,text=FALSE)
## need to make one for paper
###
WinterGD_0<-WinterGD%>%rename(AlaAT_L214F=Qsd1)
WinterGM_0<-WinterGM
WinterGM_0[WinterGM_0$SNP=="Qsd1",]$SNP<-"AlaAT_L214F"
marker_table_01<-marker_table
marker_table_01[marker_table_01$marker=="Qsd1",]$marker<-"AlaAT_L214F"
gdt<-WinterGD_0%>%dplyr::select(marker_table_01[marker_table_01$chrom=="5",]$marker)%>%as.matrix()
gmt<-WinterGM_0%>%filter(SNP%in%c(marker_table_01[marker_table_01$chrom=="5",]$marker))

#View(marker_table[marker_table$chrom=="5",])
H5_1<-c("JHI_Hv50k_2016_282096","SCRI_RS_104699","JHI_Hv50k_2016_283960")
AlaAT_L214F<-c("AlaAT_L214F","JHI_Hv50k_2016_308768","JHI_Hv50k_2016_308899")
PHS_GE<-c("JHI_Hv50k_2016_310470","JHI_Hv50k_2016_311435","JHI_Hv50k_2016_31683","JHI_Hv50k_2016_31683","JHI_Hv50k_2016_312374")
gdat<-as(gdt,"SnpMatrix")


CH5_LD<-LDheatmap(gdat,genetic.distances=gmt$Position,color=grey.colors(20),title="Linkage Disequilribium on Chr 5H",SNP.name = c(H5_1,AlaAT_L214F,PHS_GE),add.map = FALSE,text=FALSE,flip = FALSE)

LDheatmap.highlight(CH5_LD, i = 1, j = 15, col = "red",lwd=2,flipOutline=FALSE, crissCross=FALSE)
LDheatmap.highlight(CH5_LD, i = 16, j = 31, col = "red",lwd=2,flipOutline=FALSE, crissCross=FALSE)
LDheatmap.highlight(CH5_LD, i = 33, j = 88, col = "red",lwd=2,flipOutline=FALSE, crissCross=FALSE)

####




# I see 23high LD and we will need to look at Qsd1
#group A
gdt0<-gdt[,c(1:11)]
colnames(gdt0)
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="5",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")

LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 5",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)


CH5_LD_groupA_real=data.frame(SNP=gmt$SNP,Ch="5",group="A",Position=gmt$Position,ld=0.91)
## group 2
gdt0<-gdt[,c(13:19)]
colnames(gdt0)
colnames(gdt0)
gmt<-WinterGM_0%>%filter(SNP%in%c(marker_table[marker_table$chrom=="5",]$marker,"AlaAT_L214F"))
WinterGM_0[WinterGM_0$SNP=="AlaAT_L214F",]

gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")
colnames(gdat0)
gmt$SNP
LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 5",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)
CH5_LD_groupBQsd1_real=data.frame(SNP=gmt$SNP,Ch="5",group="B",Position=gmt$Position,ld=0.98)
CH5_LD_groupBQsd1_real
## Group C
gdt0<-gdt[,c(27:85)]

gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="5",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")

LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 5",SNP.name = c(gmt$SNP),add.map = FALSE,text=FALSE)

gmt$SNP
CH5_LD_groupC_real=data.frame(SNP=gmt$SNP,Ch="5",group="C",Position=gmt$Position,ld=0.91)

#group d
#phs
# gdt0<-gdt[,c(85:96)]
# 
# gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="5",]$marker))
# gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
# gdat0<-as(gdt0,"SnpMatrix")
# 
# LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 5",SNP.name = c(gmt$SNP),add.map = TRUE,text=FALSE)
# CH5_LD_groupD_real=data.frame(SNP=gmt$SNP,Ch="5",group="D_phs",Position=gmt$Position,ld=0.91)

CH5_LD=rbind(CH5_LD_groupA_real,CH5_LD_groupBQsd1_real,CH5_LD_groupC_real)
View(CH5_LD)
```

```{r}

gdt<-WinterGD%>%dplyr::select(marker_table[marker_table$chrom=="7",]$marker)%>%as.matrix()
gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="7",]$marker))

min_marker<-gmt[which(gmt$Position==min(gmt$Position)),]
gmt
max_marker<-gmt[which(gmt$Position==max(gmt$Position)),]
max_marker$Position-min_marker$Position

gdat<-as(gdt,"SnpMatrix")
max_marker$SNP
LDheatmap(gdat,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 7",SNP.name = c(min_marker$SNP,max_marker$SNP),add.map = FALSE,text=FALSE)

##7A
gdt0<-gdt[,c(2:5)]

gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="7",]$marker))
gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
gdat0<-as(gdt0,"SnpMatrix")

LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 7",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)



# ## 7A
# gdt0<-gdt[,c(5:8)]
# 
# gmt<-WinterGM%>%filter(SNP%in%c(marker_table[marker_table$chrom=="7",]$marker))
# gmt<-gmt%>%filter(SNP%in%colnames(gdt0))
# gdat0<-as(gdt0,"SnpMatrix")
# 
# LDheatmap(gdat0,genetic.distances=gmt$Position,color=rgb.palette(18),title="LD on CH 7",SNP.name = c(gmt$SNP),add.map = TRUE,text=TRUE)


CH7_LD=data.frame(SNP=gmt$SNP,Ch="7",group="A",Position=gmt$Position,ld=0.91)



```


```{r}

#closet marker to MKK3
WinterGM%>%filter(SNP=="JHI_Hv50k_2016_367317")
WinterGD%>%dplyr::select(any_of(v))


```


## LD and combined marker table
```{r}


LD_groups<-rbind(CH1_LD_groupA,CH2_LD,CH3_LD_groupA,CH4_LD,CH5_LD,CH7_LD)

#LD_groups[LD_groups$group=="D_phs",]$group<-"C"
collapse_table<-LD_groups%>%group_by(Ch,group)%>%arrange(Position)%>%reframe(SNP_marker=paste0(SNP[c(1)],", ",SNP[c(n()/2)],", ",SNP[c(n())]),bp_range=paste0(min(Position),"-",max(Position)),ld)%>%unique() %>%ungroup()  

collapse_table
conversion<-LD_groups[,1:3]%>%ungroup()%>%left_join(collapse_table,by=c("Ch","group"),relationship = "many-to-many")
##
markers
m3<-markers
FDR_rate
rownames(m3)=NULL


test<-Genetic_map%>%right_join(m3%>%dplyr::select(marker,chrom,pos),by=c("chrom","marker"))%>%tidyr::drop_na(cM)%>%unique()

m3[m3$marker=="Qsd1",]$marker<-"AlaAT_L214F"
marker_table2<-m3%>%filter(p.value<FDR_rate)%>%left_join(test%>%dplyr::select(!pos),by=(c("chrom","marker")))%>%arrange(chrom,pos)%>%mutate(marker_collapse=plyr::mapvalues(marker,from=conversion$SNP,to=conversion$SNP_marker))%>%dplyr::select(!marker)%>%unique()%>%reframe(.by = c(chrom,marker_collapse),pos_range=paste0(min(pos),"-",max(pos)),cm_range=paste0(min(cM),"-",max(cM)),PM_dates = toString(unique(PM)),trait=toString(unique(trait)),year=toString(unique(year)),testmin=round(min(maf),3),testmax=round(max(maf),3),PVEmin=round(min(expl.var),3),PVEmax=round(max(expl.var),3),Pvalue=max(round(p.value,4)))%>%



mutate(MAF=ifelse(testmax==testmin,testmax,paste0(round(testmin,3),"-",round(testmax,3))),PVE_r=ifelse(PVEmin==PVEmax,PVEmax,paste0(round(PVEmin,3),"-",round(PVEmax,3))))%>%dplyr::select(!c("testmin","testmax","PVEmin","PVEmax"))%>%tidyr::drop_na(chrom)%>%rename(Chr=chrom,SNP=marker_collapse,Position=pos_range,Timepoints=PM_dates,PVE=PVE_r)


View(marker_table2)

```
#lets do the GALLO thing
# Gallo analysis
```{r}
library("GALLO")
data("QTLmarkers")
setwd("/home/karl/git/naked_barley_GWAS/")
# we have to do this on the linux
#also do not add columns, that screws things up later. If you want adtnl columns merge them and use relocate at the end
#QTL_Disease<-sig_hits_markers%>%mutate(SNP.2=SNP,Breed="1")%>%dplyr::select(SNP,SNP.2,trait,Chr,Pos,Breed,signif)%>%rename(Associated.marker=SNP,SNP.reference=SNP.2,Trait=trait,CHR=Chr,BP=Pos,Reference=signif)%>%

QTL_Disease<-markers%>%mutate(Associated.marker=marker,SNP.reference=marker,Breed="1",CHR=chrom,BP=pos,Reference=ID)%>%select(Associated.marker,SNP.reference,trait,Breed,CHR,BP,Reference)%>%arrange(CHR,BP)
names(QTL_Disease)
QTL_Disease$CHR
v3<-import_gff_gtf(db_file="data/Reference_seq/V3/Hv_Morex.pgsb.Jul2020.gff3",file_type = "gff")

# hc_ch1<-import_gff_gtf(db_file="data/Reference_seq/Chr1H/High-Confidence-chr1H-1..522466905.gff3",file_type = "gff")
# hc_ch2<-import_gff_gtf(db_file="data/Reference_seq/Chr2H/High-Confidence-chr2H-1..675310294.gff3",file_type = "gff")
# hc_ch3<-import_gff_gtf(db_file="data/Reference_seq/Chr3H/High-Confidence-chr3H-1..628753756.gff3",file_type = "gff")
# hc_ch4<-import_gff_gtf(db_file="data/Reference_seq/Chr4H/High-Confidence-chr4H-1..624247919.gff3",file_type = "gff")
# hc_ch5<-import_gff_gtf(db_file="data/Reference_seq/Chr5H/High-Confidence-chr5H-1..599018945.gff3",file_type = "gff")
# hc_ch6<-import_gff_gtf(db_file="data/Reference_seq/Chr6H/High-Confidence-chr6H-1..573247234.gff3",file_type = "gff")
# hc_ch7<-import_gff_gtf(db_file="data/Reference_seq/Chr7H/High-Confidence-chr7H-1..634667502.gff3",file_type = "gff")
hc_v3<-import_gff_gtf(db_file="data/Reference_seq/V3/Hv_Morex.pgsb.Jul2020.gff3",file_type = "gff")
hc<-hc_v3
rm(hc_v3)
#hc=rbind(hc_ch1,hc_ch2,hc_ch3,hc_ch4,hc_ch5,hc_ch6,hc_ch7)
#rm(hc_ch1,hc_ch2,hc_ch3,hc_ch4,hc_ch5,hc_ch6,hc_ch7)


# CDS_path<-paste0("data/Reference_seq/",list.files("data/Reference_seq",pattern="CDS-",recursive = TRUE))
# 
# cds_ch1<-import_gff_gtf(db_file=CDS_path[1],file_type = "gff")
# cds_ch2<-import_gff_gtf(db_file=CDS_path[2],file_type = "gff")
# cds_ch3<-import_gff_gtf(db_file=CDS_path[3],file_type = "gff")
# cds_ch4<-import_gff_gtf(db_file=CDS_path[4],file_type = "gff")
# cds_ch5<-import_gff_gtf(db_file=CDS_path[5],file_type = "gff")
# cds_ch6<-import_gff_gtf(db_file=CDS_path[6],file_type = "gff")
# cds_ch7<-import_gff_gtf(db_file=CDS_path[7],file_type = "gff")
# cds<-rbind(cds_ch1,cds_ch2,cds_ch3,cds_ch4,cds_ch5,cds_ch6,cds_ch7)
# rm(cds_ch1,cds_ch2,cds_ch3,cds_ch4,cds_ch5,cds_ch6,cds_ch7)
# mrna_path<-paste0("data/Reference_seq/",list.files("data/Reference_seq",pattern="mRNA",recursive = TRUE))
# mrna_path
# mrna1<-import_gff_gtf(db_file=mrna_path[1],file_type = "gff")
# mrna2<-import_gff_gtf(db_file=mrna_path[2],file_type = "gff")
# mrna3<-import_gff_gtf(db_file=mrna_path[3],file_type = "gff")
# mrna4<-import_gff_gtf(db_file=mrna_path[4],file_type = "gff")
# mrna5<-import_gff_gtf(db_file=mrna_path[5],file_type = "gff")
# mrna6<-import_gff_gtf(db_file=mrna_path[6],file_type = "gff")
# mrna7<-import_gff_gtf(db_file=mrna_path[7],file_type = "gff")
# 
# mrna<-rbind(mrna1,mrna2,mrna3,mrna4,mrna5,mrna6,mrna7)
# rm(mrna1,mrna2,mrna3,mrna4,mrna5,mrna6,mrna7)
mrna<-v3%>%filter(QTL_type=="mRNA")
cds<-v3%>%filter(QTL_type=="CDS")

mrna$chr<-substr(mrna$chr,4,4)#
cds$chr<-substr(cds$chr,4,4)#
hc$chr<-substr(hc$chr,4,4)#
source("R_scripts/GALLO_custom/splitQTL_comment.R")

out.genes_mrna<-find_genes_qtls_around_markers_custom(db_file=mrna,
marker_file=QTL_Disease, method = "qtl", 
marker = "snp", interval = 2000000)
out.genes_cds<-find_genes_qtls_around_markers_custom(db_file=cds,
marker_file=QTL_Disease, method = "qtl", 
marker = "snp", interval = 2000000)
out.genes_hc<-find_genes_qtls_around_markers_custom(db_file=hc,
marker_file=QTL_Disease, method = "qtl", 
marker = "snp", interval = 2000000)
out.genes_hc

splitQTL_comment_mrna<-function(df){


  output_qtls<-as.data.frame(df)
  output_qtls$QTL_type<-as.character(output_qtls$QTL_type)
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_Association",output_qtls$QTL_type),"QTL_type"]), "_Association"),'[',1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_Association",output_qtls$QTL_type),"Association_type"]<-"Association"
    output_qtls[grep("_Association",output_qtls$QTL_type),"QTL_type"]<-tmp.type
  }
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_QTL",output_qtls$QTL_type),"QTL_type"]), "_QTL"), '[', 1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_QTL",output_qtls$QTL_type),"Association_type"]<-"QTL"
    output_qtls[grep("_QTL",output_qtls$QTL_type),"QTL_type"]<-tmp.type
  }
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_Mendelian",output_qtls$QTL_type),"QTL_type"]), "_Mendelian"), '[', 1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_Mendelian",output_qtls$QTL_type),"Association_type"]<-"Mendelian"
    output_qtls[grep("_Mendelian",output_qtls$QTL_type),"QTL_type"]<-tmp.type
    cat("problem test")
  }
  cat("Test")
  output_qtls$extra_info<-iconv(output_qtls$extra_info, "latin1", "ASCII", "")

output_qtls
  
  tmp.split<-stringr::str_split_fixed(output_qtls$extra_info, ";",n=Inf)
tt<-cbind(output_qtls,tmp.split)%>%as.data.frame()%>%distinct()

names(tt)[14]<-"gene_id"
names(tt)[15]<-"parent_id"
names(tt)[16]<-"confidence"
names(tt)[17]<-"biotype"
names(tt)[18]<-"significance_scores"

names(tt)[19]<-"description"
names(tt)[20]<-"ontology"
names(tt)[21]<-"interfam"
names(tt)[22]<-"interpro"
tt$Ordinal<-1:nrow(tt)
#View(tt)

# descr2<-tt%>%dplyr::filter(grepl('_Description2', test2),grepl('_Gene_id', test3))%>%rename(gene_description2=16,gene_ID=17,gene_ID_2=18,GO_term=19,Confidence_primary=20,Confidence_secondary=21)%>%dplyr::select(!c("9"))
# descr2
# descr3<-tt%>%filter(grepl('_Description3', test3),grepl('_Gene_id', test4))%>%rename(gene_description2=16,gene_description3=17,gene_ID=18,gene_ID_2=19,GO_term=20,Confidence_primary=21,Confidence_secondary=22)%>%dplyr::select(!c("10"))
# 
# descr4<-tt%>%filter(grepl('_Description4', test4),grepl('_Gene_id', test5))%>%rename(gene_description2=16,gene_description3=17,gene_description4=18,gene_ID=19,gene_ID_2=20,GO_term=21,Confidence_primary=22,Confidence_secondary=23)#%>%dplyr::select(!c("11"))
# #View(descr5)
# 
# descr5<-tt%>%filter(grepl('_Description5',test5))%>%rename(gene_description2=16,gene_description3=17,gene_description4=18,gene_description5=19,gene_ID=20,gene_ID_2=21,GO_term=2,Confidence_primary=23,Confidence_secondary=24)
# 
# descr<-plyr::rbind.fill(descr2,descr3,descr4,descr5)%>%dplyr::select(-"9")
# 
# 
# tt_descr1<-tt%>%filter(!Ordinal%in%descr$Ordinal)%>%rename(significance_scores=14,gene_description1=15,gene_ID=16,gene_ID_2=17,GO_term=18,Confidence_primary=19,Confidence_secondary=20)%>%dplyr::select(!c("8","9","10"))
# 
output_qtls<-tt%>%unique()



# sapply(strsplit(as.character(h$tes), "\\."), "[[", 2)
#sapply(strsplit(output_qtls[,c(7:13,15:18)], "="), "[[", 1)
output_qtls2<-output_qtls

for(i in 14:22){
  
  output_qtls2[,i]<-stringr::str_split_fixed(output_qtls[,i], "=",n=2)[,2]
  
}

mrna_qtl<-output_qtls

  return(mrna_qtl)

  #output_qtls[,"trait_ID"]<-stringr::str_split_fixed(tmp.split[,6], "=",n=2)[,2]

  #output_qtls[,"Name"]<-stringr::str_split_fixed(tmp.split[,2], "=",n=2)[,2]
  #utput_qtls[,"Abbrev"]<-stringr::str_split_fixed(tmp.split[,3], "=",n=2)[,2]


  #output_qtls[,"pubmed_id"]<-stringr::str_split_fixed(tmp.split[,4], "=",n=2)[,2]

}

mrna_output<-splitQTL_comment_mrna(df=out.genes_mrna)

#View(mrna_output)
mrna_output<-mrna_output%>%unique()


splitQTL_comment_cds<-function(df){

  output_qtls<-as.data.frame(df)
  output_qtls
  output_qtls$QTL_type<-as.character(output_qtls$QTL_type)
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_Association",output_qtls$QTL_type),"QTL_type"]), "_Association"),'[',1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_Association",output_qtls$QTL_type),"Association_type"]<-"Association"
    output_qtls[grep("_Association",output_qtls$QTL_type),"QTL_type"]<-tmp.type
  }
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_QTL",output_qtls$QTL_type),"QTL_type"]), "_QTL"), '[', 1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_QTL",output_qtls$QTL_type),"Association_type"]<-"QTL"
    output_qtls[grep("_QTL",output_qtls$QTL_type),"QTL_type"]<-tmp.type
  }
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_Mendelian",output_qtls$QTL_type),"QTL_type"]), "_Mendelian"), '[', 1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_Mendelian",output_qtls$QTL_type),"Association_type"]<-"Mendelian"
    output_qtls[grep("_Mendelian",output_qtls$QTL_type),"QTL_type"]<-tmp.type
    cat("problem test")
  }
  cat("Test")
  output_qtls$extra_info<-iconv(output_qtls$extra_info, "latin1", "ASCII", "")

  tmp.split<-stringr::str_split_fixed(output_qtls$extra_info, ";",n=Inf)

  output_qtls[,"gene_ID"]<-stringr::str_split_fixed(tmp.split[,1], "=",n=2)[,2]
   output_qtls[,"gene_parent"]<-stringr::str_split_fixed(tmp.split[,2], "=",n=2)[,2]

return(output_qtls)

  #output_qtls[,"trait_ID"]<-stringr::str_split_fixed(tmp.split[,6], "=",n=2)[,2]

  #output_qtls[,"Name"]<-stringr::str_split_fixed(tmp.split[,2], "=",n=2)[,2]
  #utput_qtls[,"Abbrev"]<-stringr::str_split_fixed(tmp.split[,3], "=",n=2)[,2]


  #output_qtls[,"pubmed_id"]<-stringr::str_split_fixed(tmp.split[,4], "=",n=2)[,2]
  return(output_qtls[,-which(colnames(output_qtls)%in%"extra_info")])
 
}
#View(hc)
out.genes_cds
cds_output<-splitQTL_comment_cds(df=out.genes_cds)
out.genes_hc


splitQTL_comment_hc<-function(df){
#df<-test<-out.genes%>%filter(QTL_type=="CDS")
  output_qtls<-hc%>%filter(!QTL_type%in%c("mRNA","CDS"))
  output_qtls
  output_qtls$QTL_type<-as.character(output_qtls$QTL_type)
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_Association",output_qtls$QTL_type),"QTL_type"]), "_Association"),'[',1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_Association",output_qtls$QTL_type),"Association_type"]<-"Association"
    output_qtls[grep("_Association",output_qtls$QTL_type),"QTL_type"]<-tmp.type
  }
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_QTL",output_qtls$QTL_type),"QTL_type"]), "_QTL"), '[', 1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_QTL",output_qtls$QTL_type),"Association_type"]<-"QTL"
    output_qtls[grep("_QTL",output_qtls$QTL_type),"QTL_type"]<-tmp.type
  }
  tmp.type<-vapply(strsplit(as.character(output_qtls[grep("_Mendelian",output_qtls$QTL_type),"QTL_type"]), "_Mendelian"), '[', 1,FUN.VALUE=character(1))
  if(length(tmp.type)!=0){
    output_qtls[grep("_Mendelian",output_qtls$QTL_type),"Association_type"]<-"Mendelian"
    output_qtls[grep("_Mendelian",output_qtls$QTL_type),"QTL_type"]<-tmp.type
    cat("problem test")
  }
  cat("Test")
    output_qtls
  output_qtls$extra_info<-iconv(output_qtls$extra_info, "latin1", "ASCII", "")
  #gene


output_qtls_exon<-output_qtls%>%filter(QTL_type=="exon")

tmp.split_gene<-stringr::str_split_fixed(output_qtls_exon$extra_info, ";",n=Inf)

 output_qtls_exon$extra_info<-iconv( output_qtls_exon$extra_info, "latin1", "ASCII", "")

  tmp.split<-stringr::str_split_fixed( output_qtls_exon$extra_info, ";",n=Inf)

  output_qtls_exon[,"gene_ID"]<-stringr::str_split_fixed(tmp.split[,1], "=",n=2)[,2]
 output_qtls_exon[,"gene_parent"]<-stringr::str_split_fixed(tmp.split[,2], "=",n=2)[,2]
 output_qtls<- output_qtls_exon


return(output_qtls)

  #output_qtls[,"trait_ID"]<-stringr::str_split_fixed(tmp.split[,6], "=",n=2)[,2]

  #output_qtls[,"Name"]<-stringr::str_split_fixed(tmp.split[,2], "=",n=2)[,2]
  #utput_qtls[,"Abbrev"]<-stringr::str_split_fixed(tmp.split[,3], "=",n=2)[,2]


  #output_qtls[,"pubmed_id"]<-stringr::str_split_fixed(tmp.split[,4], "=",n=2)[,2]
  return(output_qtls[,-which(colnames(output_qtls)%in%"extra_info")])
 
}
hc_output<-splitQTL_comment_hc(df=out.genes_hc%>%filter(!QTL_type%in%c("mRNA","CDS")))
library(stringr)
cds_output
names(mrna_output)
#View(object)
mrna_output[,"gene_id"]<-stringr::str_split_fixed(mrna_output[,"gene_id"], "=",n=2)[,2]
mrna_output$gene_id
object<-plyr::rbind.fill(mrna_output%>%rename(gene_ID=gene_id,gene_description=description),hc_output,cds_output)%>%tidyr::drop_na(SNP.reference)
names(object)
object[,"gene_description"]<-stringr::str_split_fixed(object[,"gene_description"], "=",n=2)[,2]
library(stringr)
names(object)

tt<-object%>%group_by(trait,SNP.reference,gene_ID)%>%dplyr::select(grep("gene_description", names(object), value=TRUE))%>%ungroup() %>% mutate_if(is.character, str_trim)%>%unique() %>% mutate_if(is.character, str_trim)

# tt$gene_description<-trimws(paste0(tt$gene_description1,",",tt$gene_description2,",",tt$gene_description3,",",tt$gene_description4,",",tt$gene_description5))
tt
ww<-tt%>%dplyr::select(trait,SNP.reference,gene_description,gene_ID)%>%unique()%>%group_by(SNP.reference)#%>%pivot_wider(values_from="gene_description")

ss<-object%>%dplyr::select(SNP.reference,CHR,BP,gene_ID,start_pos,end_pos)%>%unique()
table(ss$CHR)
ww
ee<-ww%>%full_join(ss,by=c("SNP.reference","gene_ID"),relationship ="many-to-many")%>%tidyr::drop_na(trait)%>%arrange(trait,CHR,BP)%>%group_by(SNP.reference,gene_description)%>%slice(1)%>%ungroup()%>%arrange(CHR,BP)%>%unique()
ee
setwd("~/git/BarleyGermination-KK/")

library(openxlsx)
write.xlsx(ee,file = "ms/supplementary_data/gene_annotation_file_0.xlsx")

```
#time to make a formated table
```{r}
save(marker_table2,file="data/marker_table2.Rdata")
marker_table2<-marker_table2%>%filter(MAF>0.05)
marker_table2$Gene_candidate<-NA
marker_table2
marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_14932",]
#CHR 2
marker_table2[marker_table2$Chr=="2",]$Position
marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_64721",]$Gene_candidate<-c("protein kinase family protein(HORVU.MOREX.r3.2HG0098680.1)")
marker_table2[marker_table2$Chr=="2",]
marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_78167, JHI_Hv50k_2016_78425, JHI_Hv50k_2016_78947",]$Gene_candidate<-"Anthocyanin 3-O-beta-glucosyltransferase	HORVU.MOREX.r3.2HG0113240.1"


marker_table2[marker_table2$SNP=="SCRI_RS_135633, JHI_Hv50k_2016_90999, SCRI_RS_132839",]$Gene_candidate<-"Dimeric alpha-amylase inhibitor(HORVU.MOREX.r2.2HG0176960), mitogen-activated protein kinase 1(HORVU.MOREX.r3.2HG0135700.1),Beta-amylase(HORVU.MOREX.r3.2HG0138420.1),1,4-alpha-glucan-branching enzyme(HORVU.MOREX.r3.2HG0165780.1)"

# 3H
marker_table2[marker_table2$Chr=="3",]


marker_table2[marker_table2$SNP=="SCRI_RS_16934",]$Gene_candidate<-"IAA-amino acid hydrolase ILR1,putative(HORVU.MOREX.r2.3HG0235350)"
#4 H
marker_table2[marker_table2$Chr=="4",]
marker_table2[marker_table2$Chr=="4",]$SNP
marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_262688, JHI_Hv50k_2016_262688, JHI_Hv50k_2016_262937",]$Gene_candidate<-"Alpha amylase inhibitor protein(HORVU.MOREX.r3.4HG0409620.1).Alpha-1,4-glucan-protein synthase [UDP-forming] 1(HORVU.MOREX.r3.4HG0409970.1), QGpc.StMo-4H(Protein Content)"
marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_275586, JHI_Hv50k_2016_275686, JHI_Hv50k_2016_276209",]$Gene_candidate<-c("	
QDp.DiMo-4H(Diastatic Power)/QBgnm.MT2-4H(Beta Glucan)")


#CH5
marker_table2[marker_table2$Chr=="5",]$SNP



marker_table2[marker_table2$SNP%in%c("JHI_Hv50k_2016_282781","JHI_Hv50k_2016_282768"),]$Gene_candidate<-"JHI_Hv50k_2016_282781	glucan synthase-like 7(HORVU.MOREX.r2.5HG0353380)"

marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_281309",]$Gene_candidate<-"GRAM-containing/ABA-responsive protein(HORVU.MOREX.r2.5HG0352660)"

marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_311161",]$Gene_candidate<-"root meristem growth factor(HORVU.MOREX.r2.5HG0403260"


marker_table2[marker_table2$Chr=="5",]$SNP
#marker_table2[marker_table2$SNP%in%c("JHI_Hv50k_2016_281309"),]$Gene_candidate<-"GRAM-containing/ABA-responsive protein(HORVU.MOREX.r3.5HG0424400.1)"
marker_table2[marker_table2$SNP%in%c("JHI_Hv50k_2016_282781","JHI_Hv50k_2016_282768","JHI_Hv50k_2016_283159","SCRI_RS_136706","SCRI_RS_136706, JHI_Hv50k_2016_283749, JHI_Hv50k_2016_283875"),]$Gene_candidate<-"Aminotransferase-like, plant mobile domain-containing protein(HORVU.MOREX.r3.5HG0425960.1)"

marker_table2[marker_table2$SNP=="Qsd1",]$Gene_candidate<-"HvAlaAT-Aminotransferase(HORVU.MOREX.r3.5HG0481320.1)"


marker_table2[marker_table2$SNP=="AlaAT_L214F, JHI_Hv50k_2016_308790, JHI_Hv50k_2016_308899",]$Gene_candidate<-"HvAlaAT-Aminotransferase(HORVU.MOREX.r3.5HG0481320.1), abscisic aldehyde oxidase 3(HORVU.MOREX.r2.5HG0399320)"

marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_310278, JHI_Hv50k_2016_312229, JHI_Hv50k_2016_314246",]$Gene_candidate<-"GRAM-containing/ABA-responsive protein(HORVU.MOREX.r3.5HG0487770.1), 1,4-alpha-glucan branching enzymeGlgB(HORVU.MOREX.r3.5HG0488090.1),root meristem growth factor(HORVU.MOREX.r3.5HG0486840.1)"

marker_table2[marker_table2$SNP=="JHI_Hv50k_2016_310358, JHI_Hv50k_2016_312229, JHI_Hv50k_2016_314246",]$Gene_candidate<-"Beta-glucosidase, putative	HORVU.MOREX.r3.5HG0490890.1,Mitogen-activated protein kinase(HORVU.MOREX.r3.5HG0491520.1"
View(marker_table2[marker_table2$Chr=="5",])

marker_table2[marker_table2$Chr=="7",]$SNP

marker_table2[marker_table2$SNP%in%c("JHI_Hv50k_2016_450348", "JHI_Hv50k_2016_483509", "JHI_Hv50k_2016_486280"),]$Gene_candidate<-"beta-amylase(HORVU.MOREX.r2.7HG0614220)/ABA responsive binding factor(HORVU.MOREX.r2.7HG0533970)/ Glucan endo-1,3-beta-glucosidase,putative(HORVU.MOREX.r2.7HG0535140)/ Seed maturation protein(HORVU.MOREX.r2.7HG0535020)"







marker_table2[marker_table2$SNP%in%c("JHI_Hv50k_2016_486280"),]$Gene_candidate<-"Endoglucanase(HORVU.MOREX.r2.7HG0584740)"



marker_table2[marker_table2$SNP%in%c("JHI_Hv50k_2016_507218, JHI_Hv50k_2016_507370, JHI_Hv50k_2016_508103"),]$Gene_candidate<-"beta glucosidase 17(HORVU.MOREX.r2.7HG0609800)"




View(marker_table2)
marker_tableR<-marker_table2
marker_tableR$SNP<-gsub("Qsd1","AlaAT_L214F",marker_tableR$SNP)

save(marker_tableR,file="ms/Rdata/Significant_marker_table.Rdata")

```
# literature
```{r}


```


