---
title: "post_GWAS_analysis"
author: "Karl Kunze"
date: "2023-03-20"
output: html_document
editor_options: 
  chunk_output_type: console
---
```{r setup}



knitr::opts_knit$set(root.dir = '/home/karl/git/BarleyGermination-KK')

```
## POST GWAS Analysis
```{r}
  getwd()
library(dplyr)
library(ggplot2)
load("data/GWA_results/WinterPerTPGWAS.Rdata")#WinterPerTPGWAS


#moved results to end of logistics, can load the data 

table_chr<-WinterPerTPGWAS%>%ungroup()%>%dplyr::select(Chromosome,Position,ordinal)%>%arrange(Chromosome,Position)%>%unique()
library(dplyr)
table_chr_min<-table_chr%>%group_by(Chromosome)%>%summarize(min=min(Position),min_ord=min(ordinal))
table_chr_max<-table_chr%>%group_by(Chromosome)%>%summarize(max=max(Position),max_ord=max(ordinal))
table_chr<-merge(table_chr_min,table_chr_max,by="Chromosome")

Chr_breaks=table_chr$max_ord-table_chr$min_ord
ChrTable<-Chr_breaks
ChrTable

chrLabel = c(1:7, 'UN')
winterOrdinalBreaks = c(ChrTable[1]/2)

winterChrLines = ChrTable[1]#for the first position

for (i in 2:8){
 winterOrdinalBreaks[i] = sum(ChrTable[1:i-1])+ChrTable[i]/2
 winterChrLines[i] = sum(ChrTable[1:i])
}
WinterPerTPGWAS
WinterPerTPGWAS2<-WinterPerTPGWAS%>%group_by(year,trait,TP)%>%mutate(fdr.pvalues=p.adjust(P.value,method="fdr"))%>%ungroup()


W

WinterPerTPGWAS2 %>% ggplot(aes(ordinal, log10PVal, color = TP, shape = year))+geom_point()+
  geom_vline(xintercept = winterChrLines, color = 'black')+
  geom_vline(xintercept = 4780, color = 'red',linetype='longdash')+
  annotate(geom= 'text', x = 4780, y = 30, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+
  scale_x_continuous(label = c("1H","2H", "3H", "4H", "5H", "6H", "7H", "UN"),
                     breaks = winterOrdinalBreaks)+
  ylab('-log(p-value)')+xlab('Chromosome')+ geom_hline(yintercept = -log10(0.05/13400) )+
  facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()


WinterPerTPGWAS %>% arrange(P.value) %>% ungroup()%>% select(SNP, Chromosome, Position) %>% unique()

#(P.value<5e-6)




#I will worry about the other models for a d
View(WinterPerTPGWAS%>%filter(P.value<5e-6)%>%group_by(SNP,Chromosome,Position))
#data %>% group_by(factor1, factor2) 
Sig_hitsW<-WinterPerTPGWAS%>%filter(P.value<5e-6)%>%group_by(SNP,Chromosome,Position)%>%dplyr::summarize(count=n(),.groups="keep")%>%
  arrange(Chromosome,Position)
table(Sig_hitsW$Chromosome)
nrow(Sig_hitsW)
#look at LD of these significant markers
WinterGD[,c(1,3,4)]
ld_heatmap=function(df){

  ld <- as.matrix(round(df,0))
  
  if(c(-1,3,4) %in% ld){
    ld[which(ld==3)]=2
    ld[which(ld==4)]=2
    ld[which(ld== -1)]=0
  }
  
  LD <- LD.Measures(donnees=ld,  na.presence=F)
  #LD$loc1=as.character(LD$loc1); LD$loc2=as.character(LD$loc2)
  r2 <- matrix(0, nrow=ncol(df), ncol=ncol(df))
  r2[lower.tri(r2, diag=FALSE)] <- LD$r2
  r2 <- t(r2)
  r2
  r2 <- as.data.frame(round(r2, 4))
  diag(r2) <- 1
  r2[lower.tri(r2)] = NA

  rownames(r2)=colnames(df); colnames(r2)=rownames(r2)
  r_2=melt(as.matrix(r2), na.rm=T)
  
  graphic = ggplot(r_2, aes(Var2, Var1, fill = value,label = value))+
    geom_tile(color = "white")+
    geom_text(color = "black",size=3)+
    scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                         midpoint = 0.5, limit = c(0,1), space = "Lab", name="r2") +
    theme_classic() +    ggtitle(paste("LD r2 from",colnames(r2)[1],"-", colnames(r2)[length(colnames(r2))], sep=" " ))
  #return(print("The R2 is ",list(r_2)))
  return(graphic)
}
#Chromosome 1
Sig_hitsW%>%filter(Chromosome==1)
grep("JHI-Hv50k-2016-8002", colnames(WinterGD))#2

Sig_hitsW%>%filter(Chromosome==2)
grep("JHI-Hv50k-2016-59425",colnames(WinterGD))#2

grep("JHI-Hv50k-2016-101167",colnames(WinterGD))#3
#no ld there
Sig_hitsW%>%filter(Chromosome==3)

grep("JHI-Hv50k-2016-165725",colnames(WinterGD))#2

#Chromsome 4
Sig_hitsW%>%filter(Chromosome==4)
grep("JHI-Hv50k-2016-228126",colnames(WinterGD))

Sig_hitsW%>%filter(Chromosome==4)
grep("JHI-Hv50k-2016-273301",colnames(WinterGD))
grep("JHI-Hv50k-2016-273434",colnames(WinterGD))
grep("JHI-Hv50k-2016-276836",colnames(WinterGD))#24
grep("JHI-Hv50k-2016-276836",colnames(WinterGD))
grep("JHI-Hv50k-2016-276707",colnames(WinterGD))
grep("JHI-Hv50k-2016-276688",colnames(WinterGD))
grep("JHI-Hv50k-2016-276604",colnames(WinterGD))
ld_heatmap(WinterGD[,c(4570,4571,4583:4588)])

grep("JHI-Hv50k-2016-276836",colnames(WinterGD))
#all but the first marker on Chr 4 are in high LD
Sig_hitsW%>%filter(Chromosome==4)

grep("Qsd1",colnames(WinterGD))#31
grep("JHI-Hv50k-2016-308652",colnames(WinterGD))#13
ld_heatmap(WinterGD[,c(5149,5150)])
#high LD
grep("JHI-Hv50k-2016-308712",colnames(WinterGD))#1<-this could be a genotyping error
ld_heatmap(WinterGD[,c(5149,5151:5155)])
#looks like there could be an independent LD block here
#0.4348
grep("JHI-Hv50k-2016-308899",colnames(WinterGD))#1
grep("JHI-Hv50k-2016-309905",colnames(WinterGD))#1
ld_heatmap(WinterGD[,c(5168,5219)])#0.4617
grep("JHI-Hv50k-2016-311700",colnames(WinterGD))#4
grep("JHI-Hv50k-2016-311908",colnames(WinterGD))#2
grep("JHI-Hv50k-2016-312060",colnames(WinterGD))#2
ld_heatmap(WinterGD[,c(5296,5305,5311)])#high LD here
#This LD block I believe is real
grep("JHI-Hv50k-2016-336814",colnames(WinterGD))#1
grep("JHI-Hv50k-2016-337656",colnames(WinterGD))#1
ld_heatmap(WinterGD[,c(5843,5866)])#low LD here
Sig_hitsW%>%filter(Chromosome==7)
grep("JHI-Hv50k-2016-507370",colnames(WinterGD))#2
sig<-c("JHI-Hv50k-2016-8002","JHI-Hv50k-2016-59425","JHI-Hv50k-2016-101167","JHI-Hv50k-2016-165725",
  "JHI-Hv50k-2016-228126","JHI-Hv50k-2016-276836","Qsd1","JHI-Hv50k-2016-311700","JHI-Hv50k-2016-337656","JHI-Hv50k-2016-507370")
c<-WinterPerTPGWAS[WinterPerTPGWAS$SNP%in%sig,]%>%select(SNP,Chromosome,Position,year,TP,trait)%>%group_by(SNP,Chromosome,Position)%>%dplyr::summarize(count=n(),.groups="keep")%>%
  arrange(Chromosome,Position)
View(c)
write.csv(c,"data/GWA_results/Significant_hits.csv")
######
```