---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

Workflow:
Winter_DH_data_consolidation.Rmd ->analysis_script.R->WDH_GenotypeData_Processing.Rmd

Filling in Missing Chr positions
```{r}
library(readxl)
Paired_samples<- read.delim("data/Genotype_data/Paired Sample Table WDH 50K.txt",stringsAsFactors = FALSE)
str(Paired_samples)

Barley50K<-read_xlsx("data/Genotype_data/50K_Physical_positions.xlsx", sheet = 'EXCAP markers')
#there is also a 9K sheet if needed, although not all positions from the 9K are validated
Barley50K=Barley50K[,c(1:3)]
library(stringr)
Barley50K$`EXCAP marker name`<-str_split_fixed(Barley50K$`EXCAP marker name`,"-0_" , 2)[,1]
colnames(Barley50K)<-c("Name","Chr","bp")
Barley50K$Type<-"50K"

#9K
Barley9K<-as.data.frame(read_xlsx("data/Genotype_data/50K_Physical_positions.xlsx", sheet = '9k markers'))
str(Barley9K)
Barley9K<-Barley9K[,c(1,2)]
str(Barley9K)
Barley9K$chr<-sapply(strsplit(Barley9K$`final chromosome:position`,":"), "[", 1)
Barley9K$chr<-sapply(strsplit(Barley9K$chr,"r"), "[", 2)
Barley9K$bp<-sapply(strsplit(Barley9K$`final chromosome:position`,":"), "[", 2)
Barley9K$bp<-sapply(strsplit(Barley9K$bp,"\\("), "[", 1)
Barley9K<-Barley9K[,-2]
str(Barley9K)
colnames(Barley9K)<-c("Name","Chr","bp")
Barley9K$Type<-"9K"
Barley50_9K<-rbind(Barley50K,Barley9K)
table(Barley50_9K$Chr)
Barley50_9K<-Barley50_9K[with(Barley50_9K, order(Chr, bp)),]


Paired_samples$Position
str(Paired_samples)
hist(Paired_samples$Position)
table(Paired_samples$Chr)
table(Barley50_9K$Chr)
str(Paired_samples)
allsnps<-merge(Barley50_9K,Paired_samples[,c(1,2,3,4)],all=T)
allsnps<-allsnps[with(allsnps,order(Chr, bp)),]
#when merging, particularly with Chr, I'm not sure what is happening with Chr and Positions, but I would want to retain the positions from the downloaded 50k and 9k data

allsnps$Name

allsnps=allsnps[which(allsnps$Name %in% Paired_samples$Name),]
str(allsnps)
hist(as.numeric(allsnps$bp))

#there are more callend positions at 4H for the pairwise than 50k +9k data

```

Winter DH 50K GENOTYPING PROCESSING

```{r}

getwd()

library(here)
load(here("WMB DH 2020 all data for analysis.RData"))
WDH20_pheno<-All_pheno_TPall

WDH20_pheno$Entry
wdh<-read.table("data/Genotype_data/Cornell2020Sorrells-50k Project_Standard1Filte-numeric.hmp.txt",skip=1, header=T,sep ='\t')
#need to skip because there is an extra header, first being numeric 'skip=1' skips the <numeric> 
```
Check Marker names
```{r}
#Check marker names
#txt tab delimited files replace "-" character with ".", need to replace "." with "-"
colnames(wdh)=gsub("\\.","-",colnames(wdh))
which(colnames(wdh)%in%allsnps$Name)
ncol(wdh)

#13142 entries match
#allsnps retained 39,621 but we only want the 13,143 in the genotype file beacause those are the filtered ones

#remove markers in allsnps(paired sample file) not in the genotype file

allsnps<-allsnps[which(allsnps$Name%in%colnames(wdh)),]

```
Check Entry Names
```{r}
wdh[1:8,1:8]
colnames(wdh)[1]<-"Entry"
colnames(wdh)[1]
wdh[,1]
rownames(wdh)<-wdh[,1]
wdh<-wdh[,-1]
#Entry names appear unchanged in genotype file from phenotype entry list(maybe
#every line we use should have a non name entry)

```
Change coding
This step should be modified if using GAPIT or rrBLUP
```{r}
#convert to 0/1/2 coding
nrow(wdh)
colnames(wdh)
wdh[c(1:8),c(1:8)]
wdh[c(1:8),c(1:8)]


wdh=as.data.frame(apply(wdh,1, function(x) { 
  
  x[which(x == 1)] = 2
  x[which(x == 0.5)] <- NA
  x[which(x == 0)] = 0
  x
  }))
#be careful not to run this recursively(i e more than once)
#this aparently inverted the matrix, which you want

#check for entries as rows, markers as columns

wdh=as.data.frame(t(wdh))
wdh[c(1:5),c(1:5)]
```
add marker data if available
```{r}
library(rrBLUP)
wdh=as.data.frame(A.mat(wdh-1, return.imputed = T)$imputed+1)

v2<-read.delim("C:/Users/Karl/50k_data/Winter DH population/barley50kMarkerPositions_Morex2019Assembly/barley50kMarkerPositions_Morex2019Assembly.gff",header = F,skip = 1)

colnames(v2)<-c("Chromo","Platform","Type","pos","pos2","unknown_1","strand","unknown_2","SNP")
head(v2)
v2$Chromo<-sapply(strsplit(v2$Chromo,"r"), "[", 2)
#v2$Chromo<-sapply(strsplit(v2$Chromo,"H"), "[",1)
v2$Chromo

v2$SNP<-sapply(strsplit(v2$SNP,"D="), "[", 2)


myGM20=data.frame(SNP=colnames(wdh))
myGM20

myGM20$Chromosome = v2$Chromo[match(myGM20$SNP, v2$SNP)]; myGM20$Chromosome = substr(myGM20$Chromosome ,1,1)
myGM20$Chromosome
myGM20$Chromosome[which(myGM20$Chromosome=="U")]="8"

myGM20$Position = as.character(v2$pos[match(myGM20$SNP, v2$SNP)])

myGD20=as.data.frame(cbind(rownames(wdh), wdh)); colnames(myGD20)[1]="taxa"
str(myGD20)
str(myGM20)


```

```{r}
library(tidyverse)

DH_Entry<-WDH20_pheno[,c(1,2,3,7,10)]
DH_Entry<-DH_Entry[!duplicated(DH_Entry['PLOT']),]


library(gdsfmt)
library(SNPRelate)
myGM20$Chromosome
chr=as.integer(myGM20$Chromosome)
str(wdh)
snpgdsCreateGeno("test.gds", genmat = as.matrix(round(wdh,0)), sample.id=row.names(wdh), snp.id = colnames(wdh), snp.chromosome = chr,
                 snp.position = as.integer(myGM20$Position), snpfirstdim = F)

gdsfile <- snpgdsOpen("test.gds")

pca <- snpgdsPCA(gdsfile)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))
plot(1:10,pc.percent[1:10])

library(ggplot2)
test=data.frame(pca$eigenvect)
DH_Entry$Pedigree
test$pedigree=DH_Entry$Pedigree[match(pca$sample.id, DH_Entry$Entry)]




ggplot(test, aes(x=X1, y=X2, color=pedigree))+
  geom_point(size=2) +
  theme_bw() +
  ggtitle("PCA by Pedigree")+
  xlab("PC1: 17.26%") +
  ylab("PC2: 5.08%")



```