---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

Workflow:
Winter_DH_data_consolidation.Rmd ->analysis_script.R->WDH_GenotypeData_Processing.Rmd

Filling in Missing Chr positions
```{r}
library(readxl)
Paired_samples<- read.delim("data/Genotype_data/Paired Sample Table WDH 50K.txt",stringsAsFactors = FALSE)
str(Paired_samples)

Barley50K<-read_xlsx("data/Genotype_data/50K_Physical_positions.xlsx", sheet = 'EXCAP markers')
#there is also a 9K sheet if needed, although not all positions from the 9K are validated
Barley50K=Barley50K[,c(1:3)]
library(stringr)
Barley50K$`EXCAP marker name`<-str_split_fixed(Barley50K$`EXCAP marker name`,"-0_" , 2)[,1]
colnames(Barley50K)<-c("Name","Chr","bp")
Barley50K$Type<-"50K"

#9K
Barley9K<-as.data.frame(read_xlsx("data/Genotype_data/50K_Physical_positions.xlsx", sheet = '9k markers'))
str(Barley9K)
Barley9K<-Barley9K[,c(1,2)]
str(Barley9K)
Barley9K$chr<-sapply(strsplit(Barley9K$`final chromosome:position`,":"), "[", 1)
Barley9K$chr<-sapply(strsplit(Barley9K$chr,"r"), "[", 2)
Barley9K$bp<-sapply(strsplit(Barley9K$`final chromosome:position`,":"), "[", 2)
Barley9K$bp<-sapply(strsplit(Barley9K$bp,"\\("), "[", 1)
Barley9K<-Barley9K[,-2]
str(Barley9K)
colnames(Barley9K)<-c("Name","Chr","bp")
Barley9K$Type<-"9K"
Barley50_9K<-rbind(Barley50K,Barley9K)
table(Barley50_9K$Chr)
Barley50_9K<-Barley50_9K[with(Barley50_9K, order(Chr, bp)),]


hist(Paired_samples$Position)
str(Paired_samples)
hist(Paired_samples$Position)
table(Paired_samples$Chr)
table(Barley50_9K$Chr)
str(Paired_samples)
allsnps<-merge(Barley50_9K,Paired_samples[,c(1,2,3,4)],all=T)
allsnps<-allsnps[with(allsnps,order(Chr, bp)),]
#when merging, particularly with Chr, I'm not sure what is happening with Chr and Positions, but I would want to retain the positions from the downloaded 50k and 9k data

allsnps$Name

allsnps=allsnps[which(allsnps$Name %in% Paired_samples$Name),]
str(allsnps)
hist(as.numeric(allsnps$bp))

#there are more callend positions at 4H for the pairwise than 50k +9k data

```

Winter DH 50K GENOTYPING PROCESSING

```{r}

getwd()

library(here)
load(here("WMB DH 2020 all data for analysis.RData"))
WDH20_pheno<-All_pheno_TPall


wdh<-read.table("data/Genotype_data/Cornell2020Sorrells-50k Project_Standard1Filte-numeric.hmp.txt",skip=1, header=T,sep ='\t')
list.files("data/Genotype_data")
wdh_parents<-read.table("data/Genotype_data/WMB_WMB_50K_misc.txt",skip=1,header = T,sep = '\t')
str(wdh_parents)
wdh_parents[1:10,1:10]
#need to skip because there is an extra header, first being numeric 'skip=1' skips the <numeric> 
```
Check Marker names
```{r}
#Check marker names
#txt tab delimited files replace "-" character with ".", need to replace "." with "-"
colnames(wdh)=gsub("\\.","-",colnames(wdh))

which(colnames(wdh)%in%allsnps$Name)
ncol(wdh)

#13142 entries match
#allsnps retained 39,621 but we only want the 13,143 in the genotype file beacause those are the filtered ones

#remove markers in allsnps(paired sample file) not in the genotype file

allsnps<-allsnps[which(allsnps$Name%in%colnames(wdh)),]

#parent genotype names
colnames(wdh_parents)=gsub("\\.","-",colnames(wdh_parents))



```
retain relevant wdh_parents for analysis
```{r}
#we want to keep DH1301910, SY Tepee, KWS Scala, Flavia, Wintmalt, Charles, and Endeavor(will be for 20-21)


```
Add AlAT marker data
```{r}

AlaT_markers<-read_excel("data/Marker_data/WMB_DH_AlaAT_KASP_rawdata.xlsx",sheet = 'Results')
AlaT_markers
AlaT_markers<-AlaT_markers[,c(1:12)]
colnames(AlaT_markers)[12]<-"AlaT_Allele"
hist(AlaT_markers$AlaT_Allele)
AlaT_markers<-AlaT_markers[!AlaT_markers$AlaT_Allele==1,]
#remove het called markers, most likely a mistake
#AlaT_markers$AlaT_Allele<-as.factor(AlaT_markers$AlaT_Allele)
#cant be a factor once its in the genotype table
AlaT_markers[,c(4,12)]

colnames(AlaT_markers)[4]<-"Entry"
#in 0 and 2 coding
```

Check Entry Names

```{r}
wdh[1:8,1:8]
colnames(wdh)[1]<-"Entry"
colnames(wdh)[1]


#added ALaT marker to genotype matrix
colnames(wdh)[ncol(wdh)-1]
rownames(wdh)<-wdh[,1]

wdh<-wdh[,-1]
wdh[1:4,c(13121:ncol(wdh))]


library(tidyverse)
#wdh<-as.data.frame(merge(wdh,AlaT_markers[,c(4,12)],by="Entry",all.x = TRUE))
which( colnames(wdh)=="JHI-Hv50k-2016-308659")#8890
#determined that this is closet marker upstream of QsD1 position
wdh=add_column(wdh,AlaT_markers$AlaT_Allele[match( rownames(wdh), AlaT_markers$Entry)],.before="JHI-Hv50k-2016-308659");colnames(wdh)[8889]="Qsd1"



wdh[c(1:5),8887:8890]





#Entry names appear unchanged in genotype file from phenotype entry list(maybe
#every line we use should have a non name entry)

```

Change coding
This step should be modified if using GAPIT or rrBLUP
```{r}
#convert to 0/1/2 coding
nrow(wdh)
#QsD1 is already in 0/2 coding

str(wdh$BK_01)
wdh=as.data.frame(apply(wdh,1, function(x) { 
  x[which(x == 2)] = 2
  x[which(x == 0)] = 0
  x[which(x == 1)] = 2
  x[which(x == 0.5)] <- NA
  
  
  x
  }))
#be careful not to run this recursively(i e more than once)
#this aparently inverted the matrix, which you want

#check for entries as rows, markers as columns
colnames(wdh)
rownames(wdh)

wdh=as.data.frame(t(wdh))
wdh[c(1:5),c(1:5)]
wdh$Qsd1

```

```{r}
library(rrBLUP)

wdh=as.data.frame(A.mat(wdh-1, return.imputed = T)$imputed+1)
str(wdh)
v2<-read.delim("data/Genotype_data/barley50kMarkerPositions_Morex2019Assembly.gff",header = F,skip = 1)


colnames(v2)<-c("Chromo","Platform","Type","pos","pos2","unknown_1","strand","unknown_2","SNP")
head(v2)
v2$Chromo<-sapply(strsplit(v2$Chromo,"r"), "[", 2)
#v2$Chromo<-sapply(strsplit(v2$Chromo,"H"), "[",1)


v2$SNP<-sapply(strsplit(v2$SNP,"D="), "[", 2)
head(v2)
head(v2)
str(v2)
#dont run this recursively
v2[nrow(v2)+1,] = c("5H","KASP_marker","KASP",442160000,442160000,NA,"+",NA,"Qsd1")
v2[nrow(v2),]
v2<-as.data.frame(v2[with(v2, order(Chromo,pos)),])
v2$Order<-1:nrow(v2)
#need to find closest segregating marker to Qsd1(alat) in winter DH population
rownames(v2)[26096]
#find the position for the springs
#MKK3 5H position JHI-Hv50k-2016-367342
#Qsd1 5H position "4421600" JHI-Hv50k-2016-308584


v2$SNP
v2[v2$SNP=="JHI-Hv50k-2016-308584",] #26096 this marker is really close to significant marker 309659, which is a significant hit in our population
v2[v2$SNP=="JHI-Hv50k-2016-308659",]

#5H 442160000

442160000- 442158883

446690041-442160000
v2$SNP
v2[1:3,1:3]
v2[c(26096),c(9)]
v2[c(26097),c(4,9)]


which(v2[c(26096:c(26096+16)),9] %in% colnames(wdh))#17 above 26096
which(v2[c((26096-12):c(26096)),9] %in% colnames(wdh))#12 below 26096
v2[c(26096:c(26096+16)),9][17]#this marker is significant
#as a result we will put this marker in the dataframe before #JHI-Hv50k-2016-308659 at 44216000



#need 
myGM20=data.frame(SNP=colnames(wdh))


myGM20$Chromosome = v2$Chromo[match(myGM20$SNP, v2$SNP)]; myGM20$Chromosome = substr(myGM20$Chromosome ,1,1)
myGM20$Chromosome
  str(myGM20)

myGM20$Chromosome[which(myGM20$Chromosome=="U")]="8"

myGM20$Position = as.character(v2$pos[match(myGM20$SNP, v2$SNP)])

myGM20[myGM20$SNP=="JHI-Hv50k-2016-308659",]#8889 -inserting before 8889


myGD20=as.data.frame(cbind(rownames(wdh), wdh)); colnames(myGD20)[1]="taxa"
str(myGD20)
str(myGM20)

myGM20<-myGM20[with(myGM20, order(Chromosome,Position)),]
myGM20[myGM20$SNP=="Qsd1",]
```

```{r}
library(tidyverse)

DH_Entry<-WDH20_pheno[,c(1,2,3,7,10)]
DH_Entry<-DH_Entry[!duplicated(DH_Entry['PLOT']),]


library(gdsfmt)
library(SNPRelate)
myGM20$Chromosome
chr=as.integer(myGM20$Chromosome)
getwd()

str(wdh)
snpgdsCreateGeno("test1.gds", genmat = as.matrix(round(wdh,0)), sample.id=row.names(wdh), snp.id = colnames(wdh), snp.chromosome = chr,
                 snp.position = as.integer(myGM20$Position), snpfirstdim = F)

gdsfile <- snpgdsOpen("test1.gds")

pca <- snpgdsPCA(gdsfile)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))
plot(1:10,pc.percent[1:10])

library(ggplot2)
test=data.frame(pca$eigenvect)
DH_Entry$Pedigree
test$pedigree=DH_Entry$Pedigree[match(pca$sample.id, DH_Entry$Entry)]




ggplot(test, aes(x=X1, y=X2, color=pedigree))+
  geom_point(size=2) +
  theme_bw() +
  ggtitle("PCA by Pedigree")+
  xlab("PC1: 17.26%") +
  ylab("PC2: 5.08%")

getwd()
save(myGD20, file="myGD20.Rdata")
save(myGM20, file="myGM20.Rdata")
save(AlaT_markers, file="AlaT_markers.Rdata")
snpgdsClose(gdsfile)


```

LD Pruning

```{r}

chr=as.integer(myGM20$Chromosome)
snpgdsCreateGeno("test2.gds", genmat = as.matrix(round(myGD20[,-1],0)), sample.id=myGD20$taxa, snp.id = colnames(myGD20)[-1], snp.chromosome = chr,
                 snp.position = as.integer(myGM20$Position), snpfirstdim = F)

(gdsfile <- snpgdsOpen("test2.gds"))

set.seed(1000)

snpset <- snpgdsLDpruning(gdsfile, method="corr", ld.threshold=0.95, slide.max.bp = 7000)

snpset_names=unlist(snpset)
myGD20_prune=myGD20[,c(1,which(colnames(myGD20) %in% snpset_names))]
myGM20_prune=myGM20[which(myGM20$SNP %in% colnames(myGD20_prune)),]
str(myGM20_prune)

myGM20_prune<-myGM20_prune[with(myGM20_prune, order(Chromosome, Position)),]
str(myGM20_prune)
table(myGM20_prune$Chromosome)
save(myGD20_prune, file="myGD20_LDprune.Rdata")
save(myGM20_prune, file="myGM20_LDprune.Rdata")

snpgdsClose(gdsfile)

```