---
title: "R Notebook"
output: html_notebook
editor_options: 
  chunk_output_type: console
---

Workflow:
Winter_DH_data_consolidation.Rmd ->analysis_script.R->WDH_GenotypeData_Processing.Rmd->GWA_results_interpretation.Rmd

Filling in Missing Chr positions
Update: This is for the wrong alignment, use the .gff file for correct alignment if needed, as the most recent file from tassel should be correct.


Correct Morex v2 2019 upload
```{r}
library(stringr)
Morex2019<-read.delim("data/Genotype_data/barley50kMarkerPositions_Morex2019Assembly.gff",skip=1,header = FALSE)
#should be the correct alignment
Morex2019=Morex2019[,c(1,2,4,7,9)]
Morex2019$V1<-str_split_fixed(Morex2019$V1,"r" , 2)[,2]
Morex2019$V9<-str_split_fixed(Morex2019$V9,"=" , 2)[,2]
colnames(Morex2019)
colnames(Morex2019)<-c("Chrom","Type","pos","strand","Marker")
#check for NAs
#table(Morex2019$Chrom,useNA = "ifany")
Morex2019<-Morex2019[with(Morex2019, order(Chrom, pos)),]
#Alignment_list<-Morex2019[Morex2019$Marker%in%wdh_raw$`rs-`,]
```

Winter DH 50K GENOTYPING PROCESSING

```{r}

getwd()

library(here)
load(here("WMB DH 2020 all data for analysis.RData"))
WDH20_pheno<-All_pheno_TPall

#changed to read new table
wdh<-read.table("data/Genotype_data/Cornell2020-50K_Project_WDH_WMB_lines_numeric.hmp.txt",skip = 1, header=T,sep ='\t')
wdh[1:15,1:12]
wdh[15:20,15:20]
###
# This is the area where we add the list of Entry names to remove
# First is "BS911-50", coverage is too low, seen at PCA plot that it is an outlier, "BS911-67", we will do this when the parents are removed
table(wdh[wdh['X.Marker.'] == "BS911.50"],useNA = "ifany")




###
#wdh0<-read.table("data/Genotype_data/Cornell2020-50K_Project_WDH_WMB_lines.hmp.txt",skip = 1, header=T,sep ='\t')
rm(wdh0)
list.files("data/Genotype_data")

```
Check Entry names
```{r}
#Check marker names
#txt tab delimited files replace "-" character with ".", need to replace "." with "-"
colnames(wdh)
colnames(wdh)=gsub("\\.","-",colnames(wdh))
colnames(wdh)
#which(colnames(wdh)%in%allsnps$Name)
ncol(wdh)
colnames(wdh)[1]

colnames(wdh)[1]<-"Entry"

#13142 entries match
#allsnps retained 39,621 but we only want the 13,143 in the genotype file beacause those are the filtered ones

#remove markers in allsnps(paired sample file) not in the genotype file

#allsnps<-allsnps[which(allsnps$Name%in%colnames(wdh)),]
#this paired sample dataframe is redundant, I initally used allsnps from a paired sample frame, the purpose was to have a SNP from the orignal all 50K marker positions to only include markers that are part of the genotype file
#parent genotype names


```

Add AlAT marker dat
```{r}

AlaT_markers<-read_excel("data/Marker_data/WMB_DH_AlaAT_KASP_rawdata.xlsx",sheet = 'Results')
AlaT_markers
AlaT_markers<-AlaT_markers[,c(1:12)]
colnames(AlaT_markers)[12]
colnames(AlaT_markers)[12]<-"AlaT_Allele"
hist(AlaT_markers$AlaT_Allele)
AlaT_markers<-AlaT_markers[!AlaT_markers$AlaT_Allele==1,]
AlaT_markers
#remove het called markers, most likely a mistake
#AlaT_markers$AlaT_Allele<-as.factor(AlaT_markers$AlaT_Allele)
#cant be a factor once its in the genotype table
AlaT_markers[,c(4,12)]
colnames(AlaT_markers)[4]
colnames(AlaT_markers)[4]<-"Entry"
#in 0 and 2 coding
```
retain relevant wdh_parents for analysis
```{r}
#we want to keep DH1301910, SY Tepee, KWS Scala, Flavia, Wintmalt, Charles, and Endeavor(will be for 20-21)

#THIS IS Now 

wdh[c(3,489,490),]$Entry
wdh[c(3,489,490),]$Entry<-c("Flavia","Scala","SY Tepee")
wdh[c(1:2,484,487,488,491),]$Entry
#remove these rows
wdh<-wdh[-c(1:2,484,487,488,491),]
dim(wdh)
#removing problem genotype lines
wdh<-wdh[!(wdh$Entry=="BS911.50" | wdh$Entry=="BS911.67"),]
#DONT RUN RECURSIVELY

```
Check Entry Names

```{r}
wdh[1:8,1:8]

rownames(wdh)<-wdh[,1]
rownames(wdh)
wdh[,1]
wdh<-wdh[,-1]
#added ALaT marker to genotype matrix
colnames(wdh)[ncol(wdh)-1]

rownames(wdh)=gsub("\\.","-",rownames(wdh))

ncol(wdh)
wdh[1:4,c(15464:ncol(wdh))]


library(tidyverse)
#wdh<-as.data.frame(merge(wdh,AlaT_markers[,c(4,12)],by="Entry",all.x = TRUE))
which( colnames(wdh)=="JHI-Hv50k-2016-308659")#8890, now 9583-1
#determined that this is closet marker upstream of QsD1 position
AlaT_markers$AlaT_Allele[match( rownames(wdh), AlaT_markers$Entry)]
AlaT_markers[498,]$Entry<-"SY Tepee"
wdh=add_column(wdh,AlaT_markers$AlaT_Allele[match( rownames(wdh), AlaT_markers$Entry)],.before="JHI-Hv50k-2016-308659");colnames(wdh)[9583]
#check for changing column name





colnames(wdh)[9583]="Qsd1"

wdh$Qsd1

wdh[c(1:5),9580:9584]





#Entry names appear unchanged in genotype file from phenotype entry list(maybe
#every line we use should have a non name entry)

```

Change coding
This step should be modified if using GAPIT or rrBLUP
```{r}
#convert to 0/1/2 coding
nrow(wdh)
#QsD1 is already in 0/2 coding

str(wdh$BK_01)
wdh=as.data.frame(apply(wdh,1, function(x) { 
  x[which(x == 2)] = 2
  x[which(x == 0)] = 0
  x[which(x == 1)] = 2
  x[which(x == 0.5)] <- NA
  
  
  x
  }))
#be careful not to run this recursively(i e more than once)
#this aparently inverted the matrix, which you want

#check for entries as rows, markers as columns
colnames(wdh)[1:10]
rownames(wdh)[1:10]
head(wdh)
wdh=as.data.frame(t(wdh))
wdh[c(1:5),c(1:5)]
wdh$Qsd1

```

```{r}
library(rrBLUP)

wdh=as.data.frame(A.mat(wdh-1, return.imputed = T)$imputed+1)
str(wdh)
v2<-read.delim("data/Genotype_data/barley50kMarkerPositions_Morex2019Assembly.gff",header = F,skip = 1)


colnames(v2)<-c("Chromo","Platform","Type","pos","pos2","unknown_1","strand","unknown_2","SNP")
head(v2)
v2$Chromo<-sapply(strsplit(v2$Chromo,"r"), "[", 2)
#v2$Chromo<-sapply(strsplit(v2$Chromo,"H"), "[",1)


v2$SNP<-sapply(strsplit(v2$SNP,"D="), "[", 2)
head(v2)
head(v2)
str(v2)
#dont run this recursively
v2[nrow(v2)+1,] = c("5H","KASP_marker","KASP",442160000,442160000,NA,"+",NA,"Qsd1")
v2[nrow(v2),]
v2<-as.data.frame(v2[with(v2, order(Chromo,pos)),])
v2$Order<-1:nrow(v2)
#need to find closest segregating marker to Qsd1(alat) in winter DH population
rownames(v2)[26096]
#find the position for the springs
#MKK3 5H position JHI-Hv50k-2016-367342
#Qsd1 5H position "4421600" JHI-Hv50k-2016-308584


v2$SNP
v2[v2$SNP=="JHI-Hv50k-2016-308584",] #26096 this marker is really close to significant marker 309659, which is a significant hit in our population
v2[v2$SNP=="JHI-Hv50k-2016-308659",]

#5H 442160000

442160000- 442158883

446690041-442160000
v2$SNP
v2[1:3,1:3]
v2[c(26096),c(9)]
v2[c(26097),c(4,9)]


which(v2[c(26096:c(26096+16)),9] %in% colnames(wdh))#17 above 26096
which(v2[c((26096-12):c(26096)),9] %in% colnames(wdh))#12 below 26096
v2[c(26096:c(26096+16)),9][17]#this marker is significant
#as a result we will put this marker in the dataframe before #JHI-Hv50k-2016-308659 at 44216000



#need 
myGM20=data.frame(SNP=colnames(wdh))


myGM20$Chromosome = v2$Chromo[match(myGM20$SNP, v2$SNP)]; myGM20$Chromosome = substr(myGM20$Chromosome ,1,1)
myGM20$Chromosome
  str(myGM20)

myGM20$Chromosome[which(myGM20$Chromosome=="U")]="8"

myGM20$Position = as.character(v2$pos[match(myGM20$SNP, v2$SNP)])

myGM20[myGM20$SNP=="JHI-Hv50k-2016-308659",]#8889 -inserting before 8889


myGD20=as.data.frame(cbind(rownames(wdh), wdh)); colnames(myGD20)[1]="taxa"
str(myGD20)
str(myGM20)

myGM20<-myGM20[with(myGM20, order(Chromosome,Position)),]
myGM20[myGM20$SNP=="Qsd1",]
```

```{r}
library(tidyverse)
WDH20_pheno[1:3,c(1,2,3,7,10)]

DH_Entry<-WDH20_pheno[,c(1,2,3,7,10,30)]
DH_Entry<-DH_Entry[!duplicated(DH_Entry['PLOT']),]
DH_Entry<-DH_Entry[!is.na(DH_Entry),]

DH_Entry
library(gdsfmt)
library(SNPRelate)
myGM20$Chromosome
chr=as.integer(myGM20$Chromosome)
getwd()


snpgdsCreateGeno("test1.gds", genmat = as.matrix(round(wdh,0)), sample.id=row.names(wdh), snp.id = colnames(wdh), snp.chromosome = chr,
                 snp.position = as.integer(myGM20$Position), snpfirstdim = F)

gdsfile <- snpgdsOpen("test1.gds")
snpgdsPCA(gdsfile)
pca <- snpgdsPCA(gdsfile)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))
plot(1:10,pc.percent[1:10])
DH_Entry<-DH_Entry[!is.na(DH_Entry$Pedigree),]
DH_Entry<-droplevels(DH_Entry)

DH_Entry[DH_Entry$Entry=="Check 1-Flavia",]$Entry<-"Flavia"

DH_Entry[DH_Entry$Entry=="Check 2-Scala",]$Entry<-"Scala"
DH_Entry[DH_Entry$Entry=="Check 3-DH130910",]$Entry<-"DH130910"
DH_Entry[DH_Entry$Entry=="Check 4-SY Tepee",]$Entry<-"SY Tepee"
DH_Entry[DH_Entry$Entry=="Check 5-Wintmalt",]$Entry<-"Wintmalt"
DH_Entry[DH_Entry$Entry=="Check 6-Charles",]$Entry<-"Charles"

library(ggplot2)
pca$eigenvect
pca

test=data.frame(pca$eigenvect)

DH_Entry$Pedigree
pca$sample.id
DH_Entry$Pedigree[match(pca$sample.id, DH_Entry$Entry)]
Check=DH_Entry$Check[match(pca$sample.id, DH_Entry$Entry)]

test$pedigree=DH_Entry$Pedigree[match(pca$sample.id, DH_Entry$Entry)]
test$Check=Check=DH_Entry$Check[match(pca$sample.id, DH_Entry$Entry)]
DH_Entry$Pedigree[match(pca$sample.id, DH_Entry$Entry)]


test<-test[!is.na(test$pedigree),]
test[is.na(test$Check),]
ggplot(test, aes(x=X1, y=X2, color=pedigree))+
  geom_point(size=1.5) +
  theme_bw() +
 theme(plot.title = element_text(hjust=0.5))+
  ggtitle("PCA by Pedigree")+
  labs(color="Pedigree")+
  xlab("PC1: 16.30%") +
  ylab("PC2: 4.80%") 
  

test.R<-as.data.frame(test)

test.W<-test.R[test.R$pedigree=="Wintmalt/DH130910",]

test.W<-test.W[with(test.W, order(X1)),]
tail(test.W[,c(1:3)])
```


```{r}
getwd()

save(myGD20, file="myGD20.Rdata")
save(myGM20, file="myGM20.Rdata")
save(AlaT_markers, file="AlaT_markers.Rdata")
snpgdsClose(gdsfile)


```
LD heatmap
```{r}

library(LDcorSV)
library(ggplot2)
library(reshape2)
grep("JHI-Hv50k-2016-312217", colnames(myGD20))
grep("JHI-Hv50k-2016-312121", colnames(myGD20))
marker<-as.data.frame(myGD20[,9052])
marker$Entry<-NA
marker$Entry<-rownames(myGD20)
marker
rownames(marker)<-rownames(myGD20)

grep("JHI-Hv50k-2016-312094", colnames(myGD20))


grep("JHI-Hv50k-2016-316817", colnames(myGD20))
grep("JHI-Hv50k-2016-316934",colnames(myGD20))

grep("Qsd1", colnames(myGD20))




colnames(myGD20)
ld_heatmap=function(df){
  ld <- as.matrix(round(df,0))
  
  if(c(-1,3,4) %in% ld){
    ld[which(ld==3)]=2
    ld[which(ld==4)]=2
    ld[which(ld== -1)]=0
  }
  
  LD <- LD.Measures(donnees=ld,  na.presence=F)
  #LD$loc1=as.character(LD$loc1); LD$loc2=as.character(LD$loc2)
  r2 <- matrix(0, nrow=ncol(df), ncol=ncol(df))
  r2[lower.tri(r2, diag=FALSE)] <- LD$r2
  r2 <- t(r2)
  r2 <- as.data.frame(round(r2, 5))
  diag(r2) <- 1
  r2[lower.tri(r2)] = NA
  rownames(r2)=colnames(df); colnames(r2)=rownames(r2)
  r_2=melt(as.matrix(r2), na.rm=T)
  
  graphic = ggplot(r_2, aes(Var2, Var1, fill = value))+
    geom_tile(color = "white")+
    scale_fill_gradient2(low = "blue", high = "red", mid = "white", 
                       midpoint = 0.5, limit = c(0,1), space = "Lab", name="r2") +
    theme_classic() #+    ggtitle(paste("LD r2 from",colnames(r2)[1],"-", colnames(r2)[length(colnames(r2))], sep=" " ))
  return(graphic)
}

load(here("myGD20.Rdata"))
load(here("myGM20.Rdata"))
#to remove the marker labels on each axis, use theme_void() instead of theme_classic()
#test of some random markers
ld_heatmap(myGD20[,9043:9052])
ld_heatmap(myGD20[,9292:9302])
#SD1 and SD2
ld_heatmap(myGD[,c(10780:10800, 13000:13075)])

ld_heatmap(myGD20[,c(12990:13010,13050:13075)]) + ggtitle('LD between significant marker regions at 5H') +
  scale_x_discrete(label = c('5H near AlaT','Other region'), breaks = c('JHI-Hv50k-2016-312121','JHI-Hv50k-2016-316934'))+
  scale_y_discrete(label = c('AlaT','Unknown'), breaks = c('JHI-Hv50k-2016-312121','JHI-Hv50k-2016-316934'))+ 
  xlab(NULL)+ylab(NULL)

ld_heatmap(myGD20[,c(10785:10795,13000:13010,13055:13070)]) + ggtitle('LD between AlaAt, GA20Ox, and MKK3 markers') +
  scale_x_discrete(label = c('GA20Ox','MKK3','AlaAt'), 
                   breaks = c('SCRI_RS_121526','JHI-Hv50k-2016-367253','JHI-Hv50k-2016-308584'))+
  scale_y_discrete(label = c('GA20Ox','MKK3','AlaAt'), 
                   breaks = c('SCRI_RS_121526','JHI-Hv50k-2016-367253','JHI-Hv50k-2016-308584'))+ 
  xlab('470-670 Mb, selected markers')+ylab(NULL)
```

LD Pruning

```{r}

chr=as.integer(myGM20$Chromosome)
snpgdsCreateGeno("test2.gds", genmat = as.matrix(round(myGD20[,-1],0)), sample.id=myGD20$taxa, snp.id = colnames(myGD20)[-1], snp.chromosome = chr,
                 snp.position = as.integer(myGM20$Position), snpfirstdim = F)

(gdsfile <- snpgdsOpen("test2.gds"))

set.seed(1000)

snpset <- snpgdsLDpruning(gdsfile, method="corr", ld.threshold=0.95, slide.max.bp = 7000)

snpset_names=unlist(snpset)
myGD20_prune=myGD20[,c(1,which(colnames(myGD20) %in% snpset_names))]
myGM20_prune=myGM20[which(myGM20$SNP %in% colnames(myGD20_prune)),]
str(myGM20_prune)

myGM20_prune<-myGM20_prune[with(myGM20_prune, order(Chromosome, Position)),]
str(myGM20_prune)
table(myGM20_prune$Chromosome)
save(myGD20_prune, file="myGD20_LDprune.Rdata")
save(myGM20_prune, file="myGM20_LDprune.Rdata")

snpgdsClose(gdsfile)

```