---
title: "The SD1 seed dormancy locus influence of pre-harvest sprouting resistance and seed dormancy loss over time"
author:
  - name: Karl Kunze
    orcid: 0000-0003-4548-1808
    email: khk44@cornell.edu
    affiliations:
      - name: Cornell University
        address: 422 Bradfield Hall
        city: Ithaca
        state: NY
        postal-code: 14853
  - name: Mark Sorrells
    email: khk44@cornell.edu
    affiliations:
      - name: Cornell University
        address: 240 Emerson Hall
        city: Ithaca
        state: NY
        postal-code: 14853
address:
  - code: Cornell University
    address: Plant Breeding and Genetics, Bradfield Hall, Ithaca, NY 14853
editor: visual
format: html
 # docx:
  #  reference-doc: custom-reference-doc.docx
link-citations: TRUE
#execute:
#  eval: false
#  echo: false
editor_options: 
  chunk_output_type: console
crossref:
  fig-title: fig   # (default is "Figure")
  tbl-title: tbl     # (default is "Table")
  title-delim: â€”     
bibliography: references.bib
---

```{r, include=FALSE}
library(flextable)
library(officer)
library(officedown)
library(dplyr)
library(tidyr)
library(knitr)
library(png)

library(Hmisc)
sysname <- Sys.info()["sysname"]
sysname
if(sysname == "Darwin") {
    path<-("~/Google Drive/projectX") # example on mac machine
} else if(sysname == "Linux") {
    path<-("/home/karl/git/BarleyGermination-KK") # example on linux machine
} else if(sysname == "Windows") {
    path<-("C:/Users/kars8/git/BarleyGermination-KK") # example on win machine
}
path
quarto_path<-paste0(path,"/ms")
setwd(quarto_path)
#knitr::opts_chunk$set(echo = FALSE)
#knitr::opts_chunk$set(warning = FALSE, message = FALSE)

```

```{r}
load(paste0(path,"/","data/WMB_DH_Germination_data2020_2021.RData"))#DH2020_2021
load(paste0(path,"/data/Genotype_data/wmb_file.Rdata"))

load(paste0(path,"/data/Genotype_data/numeric_raw.Rdata"))
WinterGM=wmbGAPIT$GM;WinterGD=as.data.frame(wmbGAPIT$GD)

DH2020_2021<-DH2020_2021%>%filter(!Location=="Spring")
marker_ef<-WinterGD%>%as.data.frame()%>%dplyr::select("Qsd1","JHI_Hv50k_2016_308762","JHI_Hv50k_2016_311435")%>%tibble::rownames_to_column()%>%rename(GID=rowname)%>%mutate(Qsd1=as.factor(Qsd1),JHI_Hv50k_2016_308762=as.factor(JHI_Hv50k_2016_308762),"JHI_Hv50k_2016_311435"=as.factor("JHI_Hv50k_2016_311435"))

df_begin<-DH2020_2021%>%mutate(GID=toupper(taxa))%>%filter(!Location=="Spring")%>%left_join(marker_ef,by="GID")%>%mutate(type="base_data_single_step")

```

# Abstract

# Introduction

Barley(*Hordeum vulagare L*) is grown worldwide for food consumption, animal feed and malt for brewing and distilling. A significant portion of barley acreage in United States and Canada is grown for malting and distilling purposes due to the high premiums compared to animal feed and food consumption. Winter malting barley, where the barley is sown in the fall, is becoming of greater interest to growers, malsters, and brewers. Changes in climate, shifting cropping systems and demand for locally farmed products are pushing the adoption of winter malting barley to be grown in non traditional environments. Growing malting barley in novel environments is challenging, particularly if there is wet and humid conditions during grain fill or at harvest. Selection of malting barley for non traditional enviroments requires a delicate balance to select for grain that does not sprout at harvest but can also provide high quality malt within minimal storage time.

Seed dormancy and pre harvest sprouting(PHS) are critical traits for the production of malting barley. Seed dormancy is defined as the inability of a seed to germinate under favorable conditions [@bewley2013] and developed as an evolutionary mechanism to avoid stress environments. Dormancy in barley is induced during seed development and is influenced by genetic and environmental factors such as temperature, water context, oxygen availability and light[@gubler2005]. Dormancy peaks at physiological maturity(PM) where the last photosynate leaves the grain, and degrades over a period of time after harvest. The rate of dormancy loss varies widely from days to months based on genotype and environmental conditions during grainfill. Prolonged seed dormancy is undesirable for malting as germination is uneven, has low vigor and and is expensive to store for long periods of time. Barley that has little to no dormancy at physiological maturity is prone to pre-harvest sprouting(PHS), where wet and humid conditions induce starch degredation and germination in the field before harvest. PHS can be visible in the form of visible radicle and collectible emergence or in pre germination where starch in the grain begins to degrade. Pregerminated grain is most often usuable for malt or food purposes. Acheiving the balance between PHS resistance and high quality malt is a key breed target for barley breeding.

Genetic sources of PHS resistance and seed dormancy have been well characterized in barley. The large effect QTL loci of SD1 and SD2 on chromosome 5H have been mapped extensively in a wide range of barley germplasm. [@han1996; @ullrich1992quantitative; @linQTLMappingDormancy2009]. Genes have been cloned at each locus: alanine aminotransferase at SD1[*HvAlaT1* @satoAlanineAminotransferaseControls2016] and a mitogen-actived protein kinase kinase 3 at SD2[*HvMKK3* @nakamura2016]. Alleles for both AlaAT and MKK3 have been identified by @vetchMutationsHvMKK3HvAlaAT12020 @sweeneyInteractionsBarleySD12022 @sweeney2022 for spring and winter barley germplasm. The mutation L214F in AlaAT results in a loss of dormancy. ALaAT has been one of the first targets of Cas9-induced mutagenesis in barley @hisano2022 prolonging dormancy in mutant types.

Genome wide associations studies(GWAS) have become the standard practice to identify traits of interest in various types of breeding populations. Numerous GWAS methods are available, and improvement to models that account for population structure @yuGeneticAssociationMapping2006 using principal component analysis @PricePrincipalComponents and the inclusion of kinship and Q matrix allow for mapping association studies to be conducted across structed populations

Advances in barley marker platforms such as the Illumina barley 50K SNP chip @bayerDevelopmentEvaluationBarley2017 and the development of improved barley reference genomes [@mascherLongreadSequenceAssembly2021] have made for higher resolution mapping of relevant malting barley traits. Increased efficiency of models and computation power have allowed not only for the association of high density marker platforms with large phenotypes, but also the use of single step GWAS modeling. Single step GWAS offers more flexibility with the capacity for the integration of more phenotypic information, inclusion of multiple fixed and random effects, as well as retaining more information that can be lost in two step analyses.

The objectives of this study aimed to genetic characterize pre-harvest sprouting and seed dormancy loss over time in a connected half sib winter malting barley breeding population. Our goals for this research were to a) Identifiy QTL for PHS, seed dormancy and seed dormacy loss over time using GWAS and b) model seed dormancy loss over time to assess optimal selection for seed dormancy. Understanding the genetic basis for PHS, seed dormancy, and seed dormancy loss over time will be invaluable information for breeders to select highly quality malting barley that have PHS resistance while also maintain a short seed dormancy period to ensure high malting quality.

# Methods

## Plant populations

The populations used in study were part of the winter malting barley breeding population at Cornell University. Four bi-parental half sibling populations were developed by crossing a common parent 'Lightning' @hayesRegistrationLightningBarley2021 , a faculative type barley, to four winter malting barley cultivars; 'KWS Scala', 'Flavia', 'SY Tepee' and 'Wintmalt'. Double haploids were developed from the crosses at Oregon state University using anther culture of $F_1$ seed from each cross. After sufficient seed increase, lines were planted in fall 2019 and fall 2020 in two fields each season in Ithaca, NY. Fields in 2019 for the 2020 harvest season,named Snyder and Ketola 5, were planted in a modified augmented design of single 1m rows with all parent lines(5), and check 'Charles' replicated in blocks across the field. The total number of plots for each locations was 544. The ratio of checks to experimental lines was approximately 10%. Fields sown in 2020 for the 2021 harvest season, named Ketola 3 and McGowan , were planted according to a randomized 480 plot augmented block design of 3 x 1 meter trimmed plots with parent lines "Lightning" and "KWS Scala" and additional line "Endeavor" used as checks. The ratio of checks to experimental lines was approximately 11%.

## Field Phenotyping and sampling

Physiological maturity(PM) was recorded as the date when 50% of the of plot lost green color from the peduncle and spike. Two days after PM, bundles of 15-20 selected mature spikes were harvested, dried for 2 days at 38 C, hand threshed, and stored at -20C to pause after-ripening. @nagelNovelLociRole2019 The standardization of harvesting and freezing spikes 2 days after the PM of the plot enabled observation of traits at the same number of after-ripening for all lines.

For seed dormancy in 2020, a total of 450 lines were sampled from each location. Lines with poor winter survival and poor agromic quality were excluded. For seed dormancy 2021 the complete trial at ketola of 435 lines(480 plots) was sampled, and approximatley 25% of lines at the McGowan location were sampled.

PHS was measured by harvesting 5 spikes per headrow at physiological maturity (PM), after-ripening for 3 days, and then misting in a greenhouse for 3 days, after which the spikes were assessed for PHS on a 0 to 9 scale. @andersonRFLPAnalysisGenomic1993 Due to labor constraints brought on by the pandemic in the 2020 field season, phenotyping capacity was limited and seed dormancy sampling was prioritized for that year. PHS was phenotyped on a sub-sample of 100 lines from the Ketola 5 location. The sub-sample consisted of all facultative types across all four families, parental checks, and Charles. For 2021, all plots from both locations were sampled.

In addition to dormancy sampling in 2021, 120 lines were selected for micromalting analysis, with 60 lines overlapping between the two locations and an additional 60 lines selected based on seed avaialability after appropriated amounts for the seed dormancy tests. Lines were processed in an identical manner to the seed dormancy sampling method

## Post harvest germination assay

To measure dormancy loss, all samples were removed the freezer at the same time, beginning the after-ripening process at the same physiological state for all lines. Samples were stored at ambient room temperature for the duration of the experiment. Germination assays were measured with petri plate assay tests that followed the American Society of Brewing Chemists (ASBC) [@kuesterSimultaneousDeterminationGermination1997] with modifications and subsequent steps followed in @sweeneyInteractionsBarleySD12021. The first modification was the use of 30 kernels instead of 100 kernels. The second modification was an extended germination count from 3 days to 5 days in lieu of counting for 3 days and using H~2~O~2~ to break dormancy. Germination ernergy(GE) as a measure of seed dormancy was determined as

$GE = \frac{n_{24} + n_{48} + n_{72}}{n_{total}}$

$GE_{5 day} = \frac{n_{24} + n_{48} + n_{72}+n_{96}+n_{120}}{n_{total}}$

Where $n$ corresponds to the the number of germinated kernels at 24, 48, 72, 96 and 120 hours after the start of the assay and $n_{total}$ is the total number of germinated and ungerminated kernels.

Germination Index was calculated as a germination rate following: @francakova2012

\$$GI_{base_{3 day}}=\frac{10 *(n_{24}+n_{48} +n_{72})} {1n_{24}+2n_{48}+3n_{72}}$\$

$GI_{base_{5 day}}=\frac{10 *(n_{24}+n_{48} +n_{72}+n_{96}+n_{120})} {1n_{24}+2n_{48}+3n_{72}+4n_{96}+5n_{120}}$

where $n_{24}$, $n_{48}$, $n_{72}$, $n_{48}$ and $n_{120}$ were the number of germinated kernels at 24, 48, 72, 96 and 120 hours after the start of the assay. GI was scaled by GE as $GI=GI_{0}*GE$ to account for low germination at earlier timepoints. At later timepoints, $GI_{0}$ was used instead of $GI$ for the inverse reason of using scaling at earlier timepoints. GE and GI were measured at five time points for 2020: 5 (TP1), 19 (TP2), 47 (TP3), 96 (TP4), and 152 (TP5) days post PM. GE and GI measured at eight time points for 2021: 5(TP1), 12(TP1.5), 19 (TP2), 33(TP2.5), 47(TP3), 68 (TP3.5), 96 (TP4) and 152(TP5) days post PM.

```{r}
#| label: tbl-H2_all
#| tbl-cap: heritability
#| fig-cap: heritability
getwd()
# Table 1
load("Rdata/genetic_variance_h2.Rdata")
load("Rdata/genetic_variance_h2.Rdata")

phenoS1<-flextable(sum_table,cwidth = 0.1,cheight = 0.1)%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")


phenoS1<-phenoS1%>%align(align = "justify",part = "header")%>%align(align = "justify",part = "body")%>%autofit(part="body",hspans = "divided")%>%fontsize(size = 9,part = "body")%>%fontsize(size = 9,part = "header")%>%
bold(bold = TRUE, part = "header")%>%set_caption(caption=as_paragraph(as_chunk("Summary statistics single and combined time point analysis ",props = fp_text_default(font.family = "Times New Roman", bold = FALSE))),fp_par(text.align="justify"),align_with_table = FALSE)%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")


phenoS1

```

## Genotyping

The entire double haploid population was genotyped with the 50K Barley SNP array @bayerDevelopmentEvaluationBarley2017 at the USDA small grains research laborartory in Fargo, ND. Marker positions were based on the Morex version 3 assembly [@mascherLongreadSequenceAssembly2021]. Markers were filtered a maximum heterozygosity level of 0.01 and a minumum minor allele frequency of 0.05 using rTASSEL @monierRTASSELInterfaceTASSEL2022, resulting in a total of 13,452 markers. The R package \@ZhengSNP2012 used to Linkage Disequlibrium(LD) prune markers using a sliding base pair window of 2000 markers and a maximum LD threshold of 0.9. PCA analysis was also conducted with the SNPRelate package. After LD pruning, 9628 markers remained for analysis. KASP markers were used to genotype *HVAlaAT* and *HvMKK3* folllowing casual mutations discovered in @satoAlanineAminotransferaseControls2016 and @nakamura2016 respectively. Details of KASP marker development can be found in @sweeney2021_QTL.

## Statisical Analysis

Given the advancement in computational efficiency, per time point GWAS was analzyed using a single step approach. The package @ASRgwas was used to develop the K and Q matricies, filter out missing information, and conduct single step GWAS using the raw data as input for each timepoint and trait ASRgwas model flexibility allowed for the integration of fixed, random, covariate and residual factor integration. The base model for GWAS per timepoint was as follows:

$$y_{} \sim 1\mu + X\beta+Z_{1}u + Q\delta + \epsilon$$

Where $y$ is the response variable for GI, $GI_{base}$ or GE, $\mu$ is the overall mean, $\beta$ is a vector of fixed effects(i.e. Location, replication), $u$ is the vector of addivitive genotype effects assoicated with the genomic additive relationship matrix with $u \sim N(0,G_{A}\sigma^2_{A})$, $Q$ is a matrix of vectors describing population structure and $\delta$ corresponds to the number of vectors(PCAs) of the Q matrix to be included to account for population structure and $\epsilon ~N(0,E\sigma^2)$ where $E$ is either a incidence matrix corresponding independently and identically distributed residuals or a heteroscadistic error structure. A summary of model terms included for each timepoint is presented in supplementary (table 2). the package @ASRgwas

```{r}
model_2020_GI<-data.frame(year=rep("2020",5),trait=rep("GI",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Location + Rep","Location",NA,NA,NA),Random=rep(NA,5),Residual=c(NA,"Location",NA,NA,NA),Converged=rep("yes",5))

model_2020_GE<-data.frame(year=rep("2020",5),trait=rep("GE",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Location","Location","Location:Rep + Location + Rep","Location","Location"),Random=rep(NA,5),Residual=c(NA,"Location",NA,NA,NA),Converged=c("yes","yes","yes","no","yes"))

models2020=rbind(model_2020_GI,model_2020_GE)
##
model_2021_GI<-data.frame(year=rep("2021",8),trait=c(rep("GI",6),rep("GI_base",2)),PM_date=c(5,22,19,33,47,68,96,152),Q=rep("yes",8),npc=rep("2",8),Fixed=c(NA,NA,NA,"Rep",NA,"Location","Location:Row + replication","Location"),Random=rep(NA,8),Residual=c("Location",NA,"Rep",NA,NA,NA,"Row",NA),Converged=c("yes","yes","yes","yes","yes","yes","yes","yes"))


model_2021_GE<-data.frame(year=rep("2021",8),trait=c(rep("GE",8)),PM_date=c(5,22,19,33,47,68,96,152),Q=rep("yes",8),npc=rep("2",8),Fixed=c(NA,NA,"Qsd1","replication:Row",NA,"Location:Row + Row","Location:Row","Location:Row"),Random=rep(NA,8),Residual=c("Location",NA,"Rep",NA,NA,NA,"Location",NA),Converged=c("no","yes","yes","yes","yes","yes","no","no"))
models2021=rbind(model_2021_GI,model_2021_GE)

#combined
model_combined_GI<-data.frame(year=rep("2020/2021",5),trait=rep("GI",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Year:Location","Location:Year","Year:Rep","Year + Year:Rep","Year + Year:replication"),Random=rep(NA,5),Residual=c("Year",NA,"Year",NA,NA),Converged=rep("yes",5))
model_combined_GI

model_combined_GE<-data.frame(year=rep("2020/2021",5),trait=rep("GE",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Year+ Year:Location",NA,"Year:Location + Location","Year:Location",NA),Random=rep(NA,5),Residual=c(NA,"Year",NA,NA,NA),Converged=c("no",rep("yes",4)))

phs_sum<-data.frame(year=rep("2020/2021"),trait="PHS",PM_date=c(7),Q=rep("yes",1),npc="2",Fixed=c("Year+ Year:Location"),Random=rep(NA,1),Residual=c(NA),Converged=c("yes"))

model_table1<-rbind(models2020,models2021,model_combined_GI,model_combined_GE,phs_sum)%>%relocate(trait,.before = 1)

```

```{r}
#| label: tbl-model_components
#| tbl-cap: components
#| fig-cap: components

# Table of model terms used for GWAS
#Table S1
E1_ft<-flextable(model_table1,cwidth = 0.1,cheight = 0.1)%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")
E1_ft<-E1_ft%>%align(align = "justify",part = "header")%>%align(align = "justify",part = "body")%>%autofit(part="body",hspans = "divided")%>%fontsize(size = 9,part = "body")%>%fontsize(size = 9,part = "header")%>%
bold(bold = TRUE, part = "header")%>%set_caption(caption=as_paragraph(as_chunk("Summary of the fixed, random and residual model terms fitted each timepoint across germination energy and index for genome wide assocation modeling ",props = fp_text_default(font.family = "Times New Roman", bold = FALSE))),fp_par(text.align="justify"),align_with_table = FALSE)%>%
#E1_ft<-E1_ft%>%add_footer_lines(E1_ft, values = "")
footnote(i = c(1), j = c(1),
            value = as_paragraph(
              c("GI: Germination index, GE: Germination Energy, GI_base:unadjusted germination index, PM Date: Post physiological maturity date and timepoint,Q: fixed effect population structure matrix, npc: number of principal components included, Rep: Replication")),
            ref_symbols = c("a"),
            part = "header")

E1_ft_2<-E1_ft%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")

# 
# footnote(i = c(1), j = c(2),
#             value = as_paragraph(
#               c("BSR-Barley Stripe Rust;LR-Leaf Rust;SB-Spot Blotch;SLD-Scald")),
#             ref_symbols = c("b"),
#             part = "header")

  
E1_ft_2
phenoS1
```

We previously reported development of a high-throughput Kompetitive Allele Specific Primer (KASP) marker AlaAT1_L214F for the causal mutation in *HvAlaAT1* discovered by Sato et al. (2016). We also developed KASP assays for a SNP in the 5' UTR of *HvGA20ox1* (GA20ox1_331_5UTR), the E165Q mutation in *HvMKK3* identified in Vetch et al. (2020) for both spring barley and winter barley. A summary of the winter barley haplotypes for the parent lines and checks are found in QTL x environment modeling of malting barley preharvest sprouting (Sweeney et al. 2021). All experimental lines and parent genotypes were monomorphic for dormant HvMKK3. The check lines Charles and Endeavor used as a phs susceptibility check in 2020 and 2021 respectively, have the highly non-dormant (N\*) allele for MKK3 and the dormant AlaAT(D) allele, however we cannot make specific conclusions about this haplotype given the low representation in the winter barley population.

All lines were genotyped with the 50Fk Illumina Infinium iSelect SNP array at the USDA Small Grains Genotyping Lab in Fargo, ND. After filtering poor quality markers, minor allele frequency (MAF) below 0.05, and monomorphic sites, 15,467 polymorphic markers remained and were used for genome-wide association (GWA). After conducting linkage disequilibrium (LD) pruning to reduce high LD blocks that exist in double haploid populations, we retained 8,384 markers for analysis. A summary of the GWA results is presented in Table 1. Models were run for all trait/time point combinations

```{r}

df_begin$Family<-gsub("DH130910","Lightning",df_begin$Family)

df_begin_long<-df_begin%>%mutate(GI_0=GI/GE)%>%mutate(GI_0=ifelse(is.nan(GI_0),NA,GI_0))%>%pivot_longer(names_to="trait",cols=c("GE","GI"))%>%rename(year=Year)
df2<-df_begin_long%>%filter(PM_date%in%c("5","19","47","96","152"))


df2$year<-"2020/2021"
library(arm)
gd_Q<-as.data.frame(WinterGD[,c("Qsd1","JHI_Hv50k_2016_8507")])

gd_Q$taxa<-rownames(gd_Q)

###




```

```{r}
#| label: fig-PHS_dist
#| fig-cap: "PHS"

#PHS
#Figure 1
df_begin_long
library(Hmisc)
phs_setup<-df_begin_long%>%rbind(df2) %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(gd_Q[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(AlaAT= ifelse(Qsd1==2,'Dormant','Non Dormant')) %>% plyr::join(gd_Q[,c('taxa','JHI_Hv50k_2016_8507')])%>%
  filter(Family %nin% c('Lightning'))  %>%dplyr::select(phs,Qsd1,AlaAT,taxa,SourcePLOT,Location,year)%>%unique()
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
# %>% mutate(JHI_Hv50k_2016_8507= ifelse(JHI_Hv50k_2016_8507==2,'minor','ref'))%>%
phs_setup
library(ggplot2)
phs_dist<-phs_setup%>%group_by(taxa)%>%
  ggplot(aes(phs, fill = AlaAT))+
 
  geom_histogram(binwidth = 0.2,col=I("black"))+
  labs(x="PHS score(0-9)",y="Frequecny")+ggtitle(label = paste0("Hisogram of Pre-harvest sprouting values for 2020/2021"))+
theme_bw()+theme(axis.text.x = element_text(angle = 0),plot.title = element_text(hjust=0.5))+
    annotate('rect', xmin=-0.1, xmax=2, ymin=0, ymax=1000, alpha=.1, fill='green')+
    annotate('rect', xmin=2, xmax=4, ymin=0, ymax=1000, alpha=.1, fill='yellow')+
     annotate('rect', xmin=4, xmax=7, ymin=0, ymax=1000, alpha=.1, fill='red')
 

phs_dist
 # stat_bin(aes(y=(phs)/sum(phs), 
 #    label=round((phs)/sum(phs),2)),
 #    geom="text", size=4, binwidth = 0.1, vjust=-1.5)
phenoS1
```
```{r}
load("Rdata/geno_corr.Rdata")
load("Rdata/p_corr.Rdata")
new
vb<-new%>%round(2)%>%as.data.frame(row.names(new))%>% tibble::rownames_to_column("Variable")%>%flextable()%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")
std_border = fp_border(color="black")
vb
vb<-vb%>%align(align = "justify",part = "header")%>%align(align = "justify",part = "body")%>%autofit(part="body",hspans = "divided")%>%fontsize(size = 9,part = "body")%>%fontsize(size = 9,part = "header")%>%
bold(bold = TRUE, part = "header")%>%bold(bold = TRUE, part = "body",j=1)%>%
  set_caption(caption=as_paragraph(as_chunk("Phenotypic and Genotypic correlations between preharvest sprouting(PHS),germination energy(GE) and germination index(GI) at post physiological maturity(PM) timepoints. The lower diagnol corresponds to phenotypic correlation and the upper diagnial corresponds to genotypic correlations",props = fp_text_default(font.family = "Times New Roman", bold = FALSE))),fp_par(text.align="justify"),align_with_table = FALSE)%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")%>%hline(part="body",i=c(1,8) ,border = std_border )%>%vline(part="body",j=c(2,10) ,border = std_border )


corr_table<-vb%>% add_footer_lines(values = "")%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")%>%compose(value= as_paragraph(as_i("Note:"),"phs, preharvest sprouting; GE Germination Energy; GI Germination Index; PM Physiological maturity date"), part = "footer")%>%align(align = c("left"),part = "footer")%>%fontsize(size = 8, part = "footer")%>%
  footnote(i = c(1), j = c(4,6,8,12,14,16),
            value = as_paragraph(
              c("Timepoints were only measured in 2021")),
            ref_symbols = c("a"),
            part = "header")


corr_table


#upper is geno, lower is pheno

```

# Results

## Phenotypic Distribution

Preharvest sprouting scores presented were averaged across years as there was a lack of sufficent phs data in the 2020 year alone. PHS score distribution was skewed towards resistance(Figure 1). 95% of genotypes were classified as resistant(0 to 2 score), 4.4% were moderately resistant and 1.6% classified as PHS susceptible. . A geom smooth plot using a GLM loess method was used to visualize the timepoint data for GE and GI within and across years.(Figure loess) Mean GE increased at early timepoints and and reached near 100% GE at later timepoints. Mean GE increased at earlier timepoints in 2020 compared to 2021 Mean GI was initially low at early timepoints and platued near 5.5 d. Mean GI increased earlier in 2020 compared to 2021. Experimental checks Endeavor and Charles had consistent mean GE values of 0.8 and mean GI values of 7 across all timepoints and years. GI values for winter experimental lines platued near 5.5-6, compared to experimental checks Endeavor and Charles, as well as spring barley lines observed in 2020 and spring lines observed in the same experimental conditions as the winters in 2021. Dormancy loss occured at approximately $PM_{47}$ to $PM_{50}$ in 2020 and between $PM_{70}$ to $PM_90$ for 2021.

##Correlations and heritaibliy 
Phenotypic correlation between GE and GI was high between 
Moderate narrow sense heritability was observed for pre-harvest sprouting in the 2020 and 2021 years(h2 = 0.37) (Table S1). Heritability for GE was high in early timepoints and peaked at PM_19 in 2020 and PM_33 in 2021. Across years, heritability was highest at the earliest timepoint PM_5.

```{r}
# Figure 3
library(tidyr)
library(ggh4x)
library(Hmisc)


op<-df_begin_long%>%rbind(df2)%>% mutate(headerFacet = 'Dataset Year') %>%filter(!trait%in%c("GI_0"))%>%
  ggplot(aes(x = PM_date, y = value, group = GID))+
  geom_smooth(na.rm = TRUE,se=FALSE,linewidth=0.2,colour="black")+ggh4x::facet_nested(trait~headerFacet+year, scales = 'free')+
  geom_vline(xintercept = c(12,33,68), color = 'red')+
  labs(title = 'Germination Energy and Index over time by individual genotype')+theme_bw()
op
# distribution of of timepoint data
```

## Genetics

### KASP marker results

THe population for this experiment used a modified NAM with a common parent Lightning crossed to four malting barley parents. KASP marker analysis revealed that the common parent Lightning and parent Flavia contained the dormant "C" allele for AlaAT and parents KWS Scala, SY Tepee and Wintmalt containing the non dormant "G" allele, making Lightning the donor for the dormant AlaAT. Experimental lines and the parents were monomorhpic for the dormant MKK3 allele, a loci identified in spring malting barley to be indicative of high malting quality but also high PHS susceptiblity. Check varities Charles and Endeavor have a super non dormant allele for MKK3, which provided a consistent check for PHS and germination, however given the low frequency of the haplotype in the experimental population, we cannot make further assertions about the role of HvMKK3 here.

### Population structure

Principal component analysis Principal component analysis revealed variation attributed to the first, second and third PCs at 14.55% and 5.69 and 4.9 % respectively. The PCA plot displays the first two PCs on the x and y axis. The first component separated SY Tepee populations from the wintmalt, KWS Scala and Flavia populations, with Lightning being a common intermediate between the two parents. Position of genotypes remaineds largely unchanged when separate recombinant inbred lines of crosses not including Lightning were included(data not shown). Wintmalt, Scala and Flavia some some degree of relatedness.

```{r}
#| label: fig-PCA_DH
#| fig-cap: "Principal components analysis plot"



# Figure 2 PCA plot of double haploid population
library(ggplot2)
load("Rdata/WinterRelationship.Rdata")


WinterPCA = eigen(WinterRelationship)


WinterPVEPCA = WinterPCA$values/sum(WinterPCA$values)
#data.frame(ordinal = 1:10, PVE = WinterPVEPCA[1:10]) %>%plot(., xlab = 'PC', col = 'red') 
WinterPVEPCA[1:5]
winterlinePCAvalues = WinterPCA$vectors %>% data.frame()%>% 
  
  mutate(family = plyr::mapvalues(substr(rownames(WinterGD),1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
                            to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
      
                                                                'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")),
         taxa = rownames(WinterGD),
         shapes = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), taxa, 'Lines'),
         size = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), 3, 2))

PCA_DH<-winterlinePCAvalues %>% mutate(shapes=gsub("(^|[[:space:]])([[:alpha:]])", "\\1\\U\\2", 
     tolower(winterlinePCAvalues$shapes),
     perl = TRUE),size=ifelse(size==2,3,4))%>%filter(family !=c('Check')) %>%
  ggplot(aes(x = X1, y = X2, color = family, shape = shapes)) + geom_point(aes(size = size))+theme_bw() +guides(shape=guide_legend(title = "Variety/Experimental Line"),size="none",color=guide_legend(title="Family/Population"))+xlab(as.character(paste0("PC1: ",round(WinterPVEPCA[1]*100,2),"%")))+ylab(as.character(paste0("PC2: ",round(WinterPVEPCA[2]*100,2),"%")))+ggtitle(lab="Principal component analysis plot of the \n NY winter malting barley double haploid population")+theme(plot.title = element_text(hjust = 0.5))+scale_x_continuous(expand = c(0, 0), limits = c(-0.125, 0.125))+scale_y_continuous(expand = c(0, 0), limits = c(-0.125, 0.125))
PCA_DH
```

### Model fitting using asrGWAS

Single step genome wide association was conducted using the package ASRgwas as fitting single step models reduce the amount of information loss associated when using fixed effect responses without weights, or using random effects models where BLUPS are either not deregressed and shrunk twice or degressed but partial shrinkage occurs regardless. Fitting single step models for each timepoint produced challenges however, as models did not converge for every timepoint trait combination. Additional challenges included variation was low, or non genetic terms such as row, location, or user phenotyping error from the replication term.

### Genome wide association analysis

Usining a false discovery rate of alpha=0.05, 75 marker trait associations were identifed across the genome. P-values within each timepoint were adjusted for false discovery using the p-adjust function. The highest raw p-value below the FDR adjusted p value threshold of 0.05 was 0.0003(). Given the highly structured breeding populations, the LD in this population was high and markers, in LD were contacted to a single row if r2=0.9 between markers. Markers are presented as a range in base pair positions of signficant MTA 2, as we do not have enough resolution to determine where specificially a casual variant would be within a specific linkage block. After accounting for markers in high LD, 36 markers were identified. The AlaAT KASP marker and markers associated in the same LD block contributed between 8-30 % PVE. In addition to the LD block associated with AlaAT, another region located in the 480 MB region was identified as significant, with PVE ranging between 8-40%. This LD block was also correlated with

```{r}
#| label: fig-AlaAT Distributions
#| fig-cap: "Qsd1 distribution"


#Figure 4
library(Hmisc)
#JHI_Hv50k_2016_314355
WinterGD$taxa<-rownames(WinterGD)
WinterGD<-WinterGD%>%relocate(taxa,.before = 1)
df_begin_long%>%rbind(df2) %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Nondormant','Dormant')) %>% plyr::join(WinterGD[,c('taxa','JHI_Hv50k_2016_310278')]) %>% filter(JHI_Hv50k_2016_310278!= 1) %>% mutate(JHI_Hv50k_2016_310278= ifelse(JHI_Hv50k_2016_310278==2,'minor','ref'))%>%
  filter(Family %nin% c('Lightning'))   %>%
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x=PM_date,y=value,fill = JHI_Hv50k_2016_310278))+
geom_boxplot(outlier.shape =NA)+ggh4x::facet_nested(trait~Family+year, scales = 'free')+

  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination rates by grouping of AlaAT"))+
theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))+guides(fill=guide_legend(title="JHI_Hv50k_2016_310278"))
```

#manhattan plot

```{r}
#Figure 5
load(paste0(path,"/data/GWA_results/STP_all.Rdata"))



ord_list<-STP_all[STP_all$ID=="GI-2020/2021-PM_152",c("chrom","pos","marker","ordinal")]
ord_list<-ord_list%>%arrange(chrom,pos)%>%mutate(ordinal=1:n())
table_chr<-STP_all%>%ungroup()%>%filter()%>%dplyr::select(marker,chrom,pos,ordinal)%>%arrange(chrom,pos)%>%mutate(ordinal=plyr::mapvalues(marker,from=ord_list$marker,to=ord_list$ordinal),ordinal=as.numeric(ordinal))%>%unique()

table_chr_min<-table_chr%>%group_by(chrom)%>%summarise(min=min(pos),min_ord=min(ordinal))
table_chr_max<-table_chr%>%group_by(chrom)%>%summarise(max=max(pos),max_ord=max(ordinal))
table_chr<-merge(table_chr_min,table_chr_max,by="chrom")%>%tidyr::drop_na(chrom)


STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$notes<-"GE-2021-PM_96: did_not_converge"
STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$ID<-"GE-2021-PM_96"
b

STP_all$ID<-gsub("GI_0","GI0",STP_all$ID)
library(stringr)
STP_all<-STP_all%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),year=str_split_i(ID,"-",2))%>%mutate(log10PVal=-log10(p.value))%>%arrange(year,trait,PM)
WinterPerTPGWAS2<-STP_all
###



Chr_breaks=table_chr$max_ord-table_chr$min_ord
ChrTable<-Chr_breaks

chrLabel = c(1:7, 'UN')
winterOrdinalBreaks = c(ChrTable[1]/2)

winterChrLines = ChrTable[1]#for the first position

for (i in 2:8){
 winterOrdinalBreaks[i] = sum(ChrTable[1:i-1])+ChrTable[i]/2
 winterChrLines[i] = sum(ChrTable[1:i])
}


WinterPerTPGWAS2<-STP_all%>%ungroup()%>%mutate(ordinal=plyr::mapvalues(marker,from=ord_list$marker,to=ord_list$ordinal),ordinal=as.numeric(ordinal))

WinterPerTPGWAS2[WinterPerTPGWAS2$ID=="GE-2021-PM_96"&WinterPerTPGWAS2$log10PVal>3.5&!is.na(WinterPerTPGWAS2$log10PVal),]$log10PVal<-0
#this GWAS TP is giving weird assocations, so I just removed it
```

```{r}



#False discovery rate
#all P values are FDR adjusted, to find the p-value of the FDR rate, I found the max p value for which the FDR p value is less than 0.05, code is below
FDR_rate<-max(WinterPerTPGWAS2[WinterPerTPGWAS2$fdr.pvalues<0.051,]$p.value,na.rm=TRUE)

#even the line is slightly more conservative than the true FDR rate since the nearest FDR value below 0.05 is 0.03. This is important because while we do not want false positive(QQ plots should be used for this), we also shouldn't be over stringent with Bonferonni cutoff, we elimate
#ALaAT_ord<-unique(WinterPerTPGWAS2[WinterPerTPGWAS2$marker=="Qsd1",]$ordinal)
ALaAT_ord=WinterPerTPGWAS2[WinterPerTPGWAS2$marker=="Qsd1",]$ordinal%>%unique()

WinterPerTPGWAS2[WinterPerTPGWAS2$trait=="PHS",]$PM<-4
#All chromosomes
ALL_MM<-WinterPerTPGWAS2%>%ungroup()%>%mutate(trait=ifelse(trait=="GI0","GI",trait))%>% ggplot(aes(ordinal, log10PVal, color = as.factor(PM), shape = year))+geom_point()+ggtitle(label="Genome wide association for Germination Energy and Index \n across timepoints and years")+
  geom_vline(xintercept = winterChrLines, color = 'black')+
 geom_vline(xintercept = ALaAT_ord, color = 'red',linetype='longdash')+
  annotate(geom= 'text', x = ALaAT_ord, y = 12, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+
  scale_x_continuous(label = c("1H","2H", "3H", "4H", "5H", "6H", "7H", "UN"),
                     breaks = winterOrdinalBreaks)+
  ylab('-log(p-value)')+xlab('Chromosome')+ geom_hline(yintercept = -log10(FDR_rate),linetype="longdash")+geom_hline(yintercept = -log10(0.05/9658) ,alpha=0.2,color="red")+theme_bw()+
 facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()

CH5_MM<-WinterPerTPGWAS2%>%ungroup()%>%mutate(trait=ifelse(trait=="GI0","GI",trait))%>% ggplot(aes(ordinal, log10PVal, color = as.factor(PM), shape = year))+geom_point()+ggtitle(label="Genome wide association for Germination Energy, Index and PHS on CH 5")+
  geom_vline(xintercept = winterChrLines, color = 'black')+
 geom_vline(xintercept = 5760, color = 'red',linetype='longdash')+
  annotate(geom= 'text', x = 5760, y = 12, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+

  ylab('-log(p-value)')+xlab('Chromosome 5')+ geom_hline(yintercept = -log10(FDR_rate),linetype="longdash")+geom_hline(yintercept = -log10(0.05/9658) ,alpha=0.2,color="red")+theme_bw()+
 facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()+scale_x_continuous(limits = c(5000,7000)) 
#Ch 5


#print(paste("FDR p value at alpha 0.05 :",round(FDR_rate,8)))
#print(paste("Bonferroni cutoff at 0.05 :",round(0.05/9658,8)))
```

Genome wide association results

If Qsd1 and associated MTAs on 5H are included as fixed effects and removed, markers JHI_Hv50k_2016_275586 JHI_Hv50k_2016_275688 JHI_Hv50k_2016_275904 on 4H reached borderline significant at Timepoints 12 and 19 in 2021 for GI. "JHI_Hv50k_2016_275586" "JHI_Hv50k_2016_275688" "JHI_Hv50k_2016_275904" reached significance in TP 68 2021 GI

There is also a marker "JHI_Hv50k_2016_64721" on 2H that reaches borderline signifcnce in 19 GI 2021

For single timepoint GWAS, we found a total of 37 significant marker trait associations associated with GE and GI across all timepoints and years. One marker per LD group of the significant single time point GWA was selected in table 1 to prevent redundancy. . The most significant association was the KASP marker for AlaAT_L214F(Qsd1) and the closely associated 50K_JHI-Hv50k-2016-276836(r=0.91) marker. Other potential hits detected in the study include Isoamylase (*HvISA3*,HORVU.MOREX.r2.5HG0404420) which is a starch-debranching enzyme located at 475796690-475807295 bp, 705025 bp distal to marker JHI-Hv50k-2016-311435. Isoamylases hydrolyze Î±-(1,6) glycosidic linkages and debranch amylopectin during grain filling (Gous and Fox, 2017). Shu and Rasmussen (2014) identified a MTA for amylose content highly associated with the contig MLOC_10776, which includes *HvISA3* (https://ics.hutton.ac.uk/morexGenes/). Starch with higher amylose content is hydrolyzed more slowly by amylolytic enzymes and higher amylose content has been hypothesized to be a contributing factor to grain dormancy (Chu et al., 2014). Although the role of amylose content in seed dormancy has been hypothesized, prior evidence for the role of *HvISA3* is limited. This locus may be useful for PHS resistance but a consistent decrease in GI is not desirable. Negative impacts on starch related malting quality traits like malt extract may also be present.Another gene of interest in this region is an abscisic acid responsive protein (Liu et al. 2013). Abscisic acid (ABA) is an important regulator of seed dormancy in barley: increases in ABA maintain seed dormancy and decreases in ABA reduce seed dormancy in barley (GÃ³mez-Cadenas et. al 1999). Other potential novel loci include HvVP1 (Viviparous-1) which may be associated with JHI-Hv50k-2016-165725 on chromosome 3H and segregating in the SY Tepee family. HvVP1 is a master transcription factor regulator that controls switching between seed maturation and germination in barley (Abraham et al. 2016). \# Marker table

```{r}
#Table 2

load("Rdata/Significant_marker_table.Rdata")



sig_markersAll<-flextable(marker_tableR,cwidth = 0.1,cheight = 0.1)%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")
sig_markersAll
```

### Role of alanine amino transferase

Alanine amino transferase played a significant role in dormancy break in our winter malting barley population. Figures 3a-c show germination rate and energy for each family of the population grouped by haplotype for AlaAT over years 2020, 2021, and combined analysis 2020/2021. A check "family" was included for comparison to Charles for 2020 and Endeavor for 2021. Parents Scala, SY Tepee and Wintmalt contained the non-dormant AlaAT allele(N) and the common parent DH130910 and Flavia contained the dormant allele(D). In both years, GE showed clear differentiation over time based on AlaAT haplotype, particularly at time point 2(19 days post PM) for 2020 and timepoints 1.5-3.5(19 days post PM through 68 days post PM). GI showed clear differentiation both across timepoints and within timepoints based on the AlaAT haplotype. The differences in GE and GI for both years based on each haplotype demonstrate that AlaAT significantly affects germination rate, energy and subsequently the rate of dormancy breakage. We also observed significant differences between GE and GI depending on the year. In 2020, dormancy break occurred around time point 3(47 days post PM). In 2021, dormancy break was substantially higher and did not occur until timepoints 3.5(68 days post PM) and 4(96 days post PM). The significant effect of the year environment on dormancy demonstrates the need to test seed germination traits across multiple years.

The observed dormancy for most non dormant haplotypes at earlier post physiological maturity (PM) days is encouraging for balancing the selection of high germination rate and maintaining initial seed dormancy for PHS. If we assume that a malster, at the earliest, would start malting barley approximately 47 days after harvest from the field, we will want to select lines that maintain dormancy until 50-60 days post PM. Most non-dormant AlaAT haplotypes fit this profile for the 2020 and 2021 crop year. However, given the substantial dormancy we observed in 2021, we must be careful in considering how it relates to our PHS values observed for that same year. PHS will need to be tested in multiple years where conditions would develop less dormant barley. If we continue to have low PHS mean and variation across multiple years with potentially differing levels of dormancy per year, we can then focus the selection pressure to increase germination rate with less concern of reducing our PHS resistance for the given AlaAT haplotypes.

# Discussion

##phenotypes

For the 2020 year, we only did a subsample of phs scoring. While the 2020 data did follow the trend explained for the 2021 results, we did not have enough observations to determine whether all experimental lines in the 2020 year were resistant or susceptible to phs in the 2020 environment mainly due to low variation and sample size. In our first complete year observation of the 435 unique lines in our trials, (above 4). The low mean of phs for most of the lines is encouraging, however phs needs to be tested in at least more than one year to account for different environmental effects. This is particularly important given the significant increased dormancy we observed for the 2021 year. Given the low variation of phs scores, correlations were low to most of the GE and GI timepoints. PHS was only moderately correlated with the first timepoint (PM 5) for GE(0.633) and GI(0.683). Even with a week of after ripening, correlation with phs scores dropped significantly at time point 1.5(12 days post PM) for GE(0.367) and GI(0.415). This suggests that for our winter barley population, there is potential to select for increased dormancy break while maintaining PHS resistance. Broad sense heritability for all germination traits, timepoints, years and combinations were very high(0.9). Heritability dropped slight for GE at later time points due to reduced variation but still retained heritability values a minimum\~0.75. \## Distrbutions

Important point- cooler temperatures during grainfill for winter barley could be a pereptual inducment of longer dormancy

### How can germination assays be improved

Dormancy assays current limited throughput and many considerations were made to balance the statistical power effectiveness and allocation of time to maximize observation of dormancy loss over time. With the advent of imaging technologies, higher throughput methods using image analysis could not only classify germinated kernels to non germination kernels, but also track the vigor of kernels over the course of the germination experiment, as germination vigor is an important factor in malting. GI scores and day 1 germination scores were consistently lower in experimental winter malting barley populations, even after dormancy was broken. This offers

# Conclusion

# 
