---
title: "The SD1 seed dormancy locus influence of pre-harvest sprouting resistance and seed dormancy loss over time"
author:
  - name: Karl Kunze
    orcid: 0000-0003-4548-1808
    email: khk44@cornell.edu
    affiliations:
      - name: Cornell University
        address: 422 Bradfield Hall
        city: Ithaca
        state: NY
        postal-code: 14853
  - name: Mark Sorrells
    email: khk44@cornell.edu
    affiliations:
      - name: Cornell University
        address: 240 Emerson Hall
        city: Ithaca
        state: NY
        postal-code: 14853
address:
  - code: Cornell University
    address: Plant Breeding and Genetics, Bradfield Hall, Ithaca, NY 14853
editor: visual
format: html
 # docx:
  #  reference-doc: custom-reference-doc.docx
link-citations: TRUE
#execute:
#  eval: false
#  echo: false
editor_options: 
  chunk_output_type: console
crossref:
  fig-title: fig   # (default is "Figure")
  tbl-title: tbl     # (default is "Table")
  title-delim: â€”     
bibliography: references.bib
---

```{r, include=FALSE}
library(flextable)
library(officer)
library(officedown)
library(dplyr)
library(tidyr)
library(knitr)
library(png)

library(Hmisc)
sysname <- Sys.info()["sysname"]
sysname
if(sysname == "Darwin") {
    path<-("~/Google Drive/projectX") # example on mac machine
} else if(sysname == "Linux") {
    path<-("/home/karl/git/BarleyGermination-KK") # example on linux machine
} else if(sysname == "Windows") {
    path<-("C:/Users/kars8/git/BarleyGermination-KK") # example on win machine
}
path
quarto_path<-paste0(path,"/ms")
setwd(quarto_path)
#knitr::opts_chunk$set(echo = FALSE)
#knitr::opts_chunk$set(warning = FALSE, message = FALSE)

```

```{r}
load(paste0(path,"/","data/WMB_DH_Germination_data2020_2021.RData"))#DH2020_2021
load(paste0(path,"/data/Genotype_data/wmb_file.Rdata"))

load(paste0(path,"/data/Genotype_data/numeric_raw.Rdata"))
WinterGM=wmbGAPIT$GM;WinterGD=as.data.frame(wmbGAPIT$GD)

DH2020_2021<-DH2020_2021%>%filter(!Location=="Spring")
marker_ef<-WinterGD%>%as.data.frame()%>%dplyr::select("Qsd1","JHI_Hv50k_2016_308762","JHI_Hv50k_2016_311435")%>%tibble::rownames_to_column()%>%rename(GID=rowname)%>%mutate(Qsd1=as.factor(Qsd1),JHI_Hv50k_2016_308762=as.factor(JHI_Hv50k_2016_308762),"JHI_Hv50k_2016_311435"=as.factor("JHI_Hv50k_2016_311435"))

df_begin<-DH2020_2021%>%mutate(GID=toupper(taxa))%>%filter(!Location=="Spring")%>%left_join(marker_ef,by="GID")%>%mutate(type="base_data_single_step")

```

# Abstract

# Introduction

Barley(*Hordeum vulagare L*) is grown worldwide for food consumption, animal feed and malt for brewing and distilling. A significant portion of barley acreage in United States and Canada is grown for malting and distilling purposes due to the high premiums compared to animal feed and food consumption. Winter malting barley, where the barley is sown in the fall, is becoming of greater interest to growers, malsters, and brewers. Changes in climate, shifting cropping systems and demand for locally farmed products are pushing the adoption of winter malting barley to be grown in non traditional environments. Growing malting barley in novel environments is challenging, particularly if there is wet and humid conditions during grain fill or at harvest. Selection of malting barley for non traditional enviroments requires a delicate balance to select for grain that does not sprout at harvest but can also provide high quality malt within minimal storage time.

Seed dormancy and pre harvest sprouting(PHS) are critical traits for the production of malting barley. Seed dormancy is defined as the inability of a seed to germinate under favorable conditions [@bewley2013] and developed as an evolutionary mechanism to avoid stress environments. Dormancy in barley is induced during seed development and is influenced by genetic and environmental factors such as temperature, water context, oxygen availability and light[@gubler2005]. Seed dormancy is regulated by the ratio of Abscisic acid (ABA) to Giberillic acid(GA) as a high ratio of ABA/GA maintains seed dormancy and decreases in the ABA/GA ratio reduce seed dormancy in barley @gomez-cadenasAbscisicAcidinducedProtein1999

Dormancy peaks at physiological maturity(PM) where the last photosynate leaves the grain, and degrades over a period of time after harvest. The rate of dormancy loss varies widely from days to months based on genotype and environmental conditions during grainfill. Prolonged seed dormancy is undesirable for malting as germination is uneven, has low vigor and and is expensive to store for long periods of time. Barley that has little to no dormancy at physiological maturity is prone to pre-harvest sprouting(PHS), where wet and humid conditions induce starch degredation and germination in the field before harvest. PHS can be visible in the form of visible radicle and collectible emergence or in pre germination where starch in the grain begins to degrade. Pregerminated grain is most often usuable for malt or food purposes. Acheiving the balance between PHS resistance and high quality malt is a key breed target for barley breeding.

Genetic sources of PHS resistance and seed dormancy have been well characterized in barley. The large effect QTL loci of SD1 and SD2 on chromosome 5H have been mapped extensively in a wide range of barley germplasm. [@han1996; @ullrich1992quantitative; @linQTLMappingDormancy2009]. Genes have been cloned at each locus: alanine aminotransferase at SD1[*HvAlaT1* @satoAlanineAminotransferaseControls2016] and a mitogen-actived protein kinase kinase 3 at SD2[*HvMKK3* @nakamura2016]. Alleles for both AlaAT and MKK3 have been identified by @vetchMutationsHvMKK3HvAlaAT12020 @sweeneyInteractionsBarleySD12022 @sweeney2022 for spring and winter barley germplasm. The mutation L214F in AlaAT results in a loss of dormancy. ALaAT has been one of the first targets of Cas9-induced mutagenesis in barley @hisano2022 prolonging dormancy in mutant types.

Genome wide associations studies(GWAS) have become the standard practice to identify traits of interest in various types of breeding populations. Numerous GWAS methods are available, and improvement to models that account for population structure @yuGeneticAssociationMapping2006 using principal component analysis @PricePrincipalComponents and the inclusion of kinship and Q matrix allow for mapping association studies to be conducted across structed populations

Advances in barley marker platforms such as the Illumina barley 50K SNP chip @bayerDevelopmentEvaluationBarley2017 and the development of improved barley reference genomes [@mascherLongreadSequenceAssembly2021] have made for higher resolution mapping of relevant malting barley traits. Increased efficiency of models and computation power have allowed not only for the association of high density marker platforms with large phenotypes, but also the use of single step GWAS modeling. Single step GWAS offers more flexibility with the capacity for the integration of more phenotypic information, inclusion of multiple fixed and random effects, as well as retaining more information that can be lost in two step analyses.

The objectives of this study aimed to genetic characterize pre-harvest sprouting and seed dormancy loss over time in a connected half sib winter malting barley breeding population. Our goals for this research were to a) Identifiy QTL for PHS, seed dormancy and seed dormacy loss over time using GWAS and b) model seed dormancy loss over time to assess optimal selection for seed dormancy. Understanding the genetic basis for PHS, seed dormancy, and seed dormancy loss over time will be invaluable information for breeders to select highly quality malting barley that have PHS resistance while also maintain a short seed dormancy period to ensure high malting quality.

# Methods

## Plant populations

The populations used in study were part of the winter malting barley breeding population at Cornell University. Four bi-parental half sibling populations were developed by crossing a common parent 'Lightning' @hayesRegistrationLightningBarley2021 , a faculative type barley, to four winter malting barley cultivars; 'KWS Scala', 'Flavia', 'SY Tepee' and 'Wintmalt'. Double haploids were developed from the crosses at Oregon state University using anther culture of $F_1$ seed from each cross. After sufficient seed increase, lines were planted in fall 2019 and fall 2020 in two fields each season in Ithaca, NY. Fields in 2019 for the 2020 harvest season,named Snyder and Ketola 5, were planted in a modified augmented design of single 1m rows with all parent lines(5), and check 'Charles' replicated in blocks across the field. The total number of plots for each locations was 544. The ratio of checks to experimental lines was approximately 10%. Fields sown in 2020 for the 2021 harvest season, named Ketola 3 and McGowan , were planted according to a randomized 480 plot augmented block design of 3 x 1 meter trimmed plots with parent lines "Lightning" and "KWS Scala" and additional line "Endeavor" used as checks. The ratio of checks to experimental lines was approximately 11%.

## Field Phenotyping and sampling

Physiological maturity(PM) was recorded as the date when 50% of the of plot lost green color from the peduncle and spike. Two days after PM, bundles of 15-20 selected mature spikes were harvested, dried for 2 days at 38 C, hand threshed, and stored at -20C to pause after-ripening. @nagelNovelLociRole2019 The standardization of harvesting and freezing spikes 2 days after the PM of the plot enabled observation of traits at the same number of after-ripening for all lines.

For seed dormancy in 2020, a total of 450 lines were sampled from each location. Lines with poor winter survival and poor agromic quality were excluded. For seed dormancy 2021 the complete trial at ketola of 435 lines(480 plots) was sampled, and approximatley 25% of lines at the McGowan location were sampled.

PHS was measured by harvesting 5 spikes per headrow at physiological maturity (PM), after-ripening for 3 days, and then misting in a greenhouse for 3 days, after which the spikes were assessed for PHS on a 0 to 9 scale. @andersonRFLPAnalysisGenomic1993 Due to labor constraints brought on by the pandemic in the 2020 field season, phenotyping capacity was limited and seed dormancy sampling was prioritized for that year. PHS was phenotyped on a sub-sample of 100 lines from the Ketola 5 location. The sub-sample consisted of all facultative types across all four families, parental checks, and Charles. For 2021, all plots from both locations were sampled.

In addition to dormancy sampling in 2021, 120 lines were selected for micromalting analysis, with 60 lines overlapping between the two locations and an additional 60 lines selected based on seed avaialability after appropriated amounts for the seed dormancy tests. Lines were processed in an identical manner to the seed dormancy sampling method

## Post harvest germination assay

To measure dormancy loss, all samples were removed the freezer at the same time, beginning the after-ripening process at the same physiological state for all lines. Samples were stored at ambient room temperature for the duration of the experiment. Germination assays were measured with petri plate assay tests that followed the American Society of Brewing Chemists (ASBC) [@kuesterSimultaneousDeterminationGermination1997] with modifications and subsequent steps followed in @sweeneyInteractionsBarleySD12021. The first modification was the use of 30 kernels instead of 100 kernels. The second modification was an extended germination count from 3 days to 5 days in lieu of counting for 3 days and using H~2~O~2~ to break dormancy. Germination ernergy(GE) as a measure of seed dormancy was determined as

$GE = \frac{n_{24} + n_{48} + n_{72}}{n_{total}}$

$GE_{5 day} = \frac{n_{24} + n_{48} + n_{72}+n_{96}+n_{120}}{n_{total}}$

Where $n$ corresponds to the the number of germinated kernels at 24, 48, 72, 96 and 120 hours after the start of the assay and $n_{total}$ is the total number of germinated and ungerminated kernels.

Germination Index was calculated as a germination rate following: @francakova2012

\$$GI_{base_{3 day}}=\frac{10 *(n_{24}+n_{48} +n_{72})} {1n_{24}+2n_{48}+3n_{72}}$\$

$GI_{base_{5 day}}=\frac{10 *(n_{24}+n_{48} +n_{72}+n_{96}+n_{120})} {1n_{24}+2n_{48}+3n_{72}+4n_{96}+5n_{120}}$

where $n_{24}$, $n_{48}$, $n_{72}$, $n_{48}$ and $n_{120}$ were the number of germinated kernels at 24, 48, 72, 96 and 120 hours after the start of the assay. GI was scaled by GE as $GI=GI_{0}*GE$ to account for low germination at earlier timepoints. At later timepoints, $GI_{0}$ was used instead of $GI$ for the inverse reason of using scaling at earlier timepoints. GE and GI were measured at five time points for 2020: 5 (TP1), 19 (TP2), 47 (TP3), 96 (TP4), and 152 (TP5) days post PM. GE and GI measured at eight time points for 2021: 5(TP1), 12(TP1.5), 19 (TP2), 33(TP2.5), 47(TP3), 68 (TP3.5), 96 (TP4) and 152(TP5) days post PM.

```{r}
#| label: tbl-H2_all
#| tbl-cap: heritability
#| fig-cap: heritability
getwd()
# Table 1
load("Rdata/genetic_variance_h2.Rdata")
load("Rdata/genetic_variance_h2.Rdata")
sum_table
phenoS1<-sum_table%>%mutate(o2g=round(o2g,3),h2=round(h2,3),h2_PEV=round(h2_PEV,3))%>%flextable(cwidth = 0.1,cheight = 0.1)%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")


phenoS1<-phenoS1%>%align(align = "justify",part = "header")%>%align(align = "justify",part = "body")%>%autofit(part="body",hspans = "divided")%>%fontsize(size = 9,part = "body")%>%fontsize(size = 9,part = "header")%>%
bold(bold = TRUE, part = "header")%>%set_caption(caption=as_paragraph(as_chunk("Summary statistics single and combined time point analysis ",props = fp_text_default(font.family = "Times New Roman", bold = FALSE))),fp_par(text.align="justify"),align_with_table = FALSE)%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")


phenoS1

```

## Genotyping

The entire double haploid population was genotyped with the 50K Barley SNP array @bayerDevelopmentEvaluationBarley2017 at the USDA small grains research laborartory in Fargo, ND. Marker positions were based on the Morex version 3 assembly [@mascherLongreadSequenceAssembly2021]. Markers were filtered a maximum heterozygosity level of 0.01 and a minumum minor allele frequency of 0.05 using rTASSEL @monierRTASSELInterfaceTASSEL2022, resulting in a total of 13,452 markers. The R package \@ZhengSNP2012 used to Linkage Disequlibrium(LD) prune markers using a sliding base pair window of 2000 markers and a maximum LD threshold of 0.9. PCA analysis was also conducted with the SNPRelate package. After LD pruning, 9628 markers remained for analysis. KASP markers were used to genotype *HVAlaAT* and *HvMKK3* folllowing casual mutations discovered in @satoAlanineAminotransferaseControls2016 and @nakamura2016 respectively. Details of KASP marker development can be found in @sweeney2021_QTL.

## Statisical Analysis

Given the advancement in computational efficiency, per time point GWAS was analzyed using a single step approach. The package @ASRgwas was used to develop the K and Q matricies, filter out missing information, and conduct single step GWAS using the raw data as input for each timepoint and trait ASRgwas model flexibility allowed for the integration of fixed, random, covariate and residual factor integration. The base model for GWAS per timepoint was as follows:

$$y_{} \sim 1\mu + X\beta+Z_{1}u + Q\delta + \epsilon$$

Where $y$ is the response variable for GI, $GI_{base}$ or GE, $\mu$ is the overall mean, $\beta$ is a vector of fixed effects(i.e. Location, replication), $u$ is the vector of addivitive genotype effects assoicated with the genomic additive relationship matrix with $u \sim N(0,G_{A}\sigma^2_{A})$, $Q$ is a matrix of vectors describing population structure and $\delta$ corresponds to the number of vectors(PCAs) of the Q matrix to be included to account for population structure and $\epsilon ~N(0,E\sigma^2)$ where $E$ is either a incidence matrix corresponding independently and identically distributed residuals or a heteroscadistic error structure. A summary of model terms included for each timepoint is presented in supplementary (table 2). the package @ASRgwas

```{r}
model_2020_GI<-data.frame(year=rep("2020",5),trait=rep("GI",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Location + Rep","Location",NA,NA,NA),Random=rep(NA,5),Residual=c(NA,"Location",NA,NA,NA),Converged=rep("yes",5))

model_2020_GE<-data.frame(year=rep("2020",5),trait=rep("GE",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Location","Location","Location:Rep + Location + Rep","Location","Location"),Random=rep(NA,5),Residual=c(NA,"Location",NA,NA,NA),Converged=c("yes","yes","yes","no","yes"))

models2020=rbind(model_2020_GI,model_2020_GE)
##
model_2021_GI<-data.frame(year=rep("2021",8),trait=c(rep("GI",6),rep("GI_base",2)),PM_date=c(5,22,19,33,47,68,96,152),Q=rep("yes",8),npc=rep("2",8),Fixed=c(NA,NA,NA,"Rep",NA,"Location","Location:Row + replication","Location"),Random=rep(NA,8),Residual=c("Location",NA,"Rep",NA,NA,NA,"Row",NA),Converged=c("yes","yes","yes","yes","yes","yes","yes","yes"))


model_2021_GE<-data.frame(year=rep("2021",8),trait=c(rep("GE",8)),PM_date=c(5,22,19,33,47,68,96,152),Q=rep("yes",8),npc=rep("2",8),Fixed=c(NA,NA,"Qsd1","replication:Row",NA,"Location:Row + Row","Location:Row","Location:Row"),Random=rep(NA,8),Residual=c("Location",NA,"Rep",NA,NA,NA,"Location",NA),Converged=c("no","yes","yes","yes","yes","yes","no","no"))
models2021=rbind(model_2021_GI,model_2021_GE)

#combined
model_combined_GI<-data.frame(year=rep("2020/2021",5),trait=rep("GI",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Year:Location","Location:Year","Year:Rep","Year + Year:Rep","Year + Year:replication"),Random=rep(NA,5),Residual=c("Year",NA,"Year",NA,NA),Converged=rep("yes",5))
model_combined_GI

model_combined_GE<-data.frame(year=rep("2020/2021",5),trait=rep("GE",5),PM_date=c(5,19,47,96,152),Q=rep("yes",5),npc=rep("2",5),Fixed=c("Year+ Year:Location",NA,"Year:Location + Location","Year:Location",NA),Random=rep(NA,5),Residual=c(NA,"Year",NA,NA,NA),Converged=c("no",rep("yes",4)))

phs_sum<-data.frame(year=rep("2020/2021"),trait="PHS",PM_date=c(7),Q=rep("yes",1),npc="2",Fixed=c("Year+ Year:Location"),Random=rep(NA,1),Residual=c(NA),Converged=c("yes"))

model_table1<-rbind(models2020,models2021,model_combined_GI,model_combined_GE,phs_sum)%>%relocate(trait,.before = 1)
length(model_table1)
dim(model_table1)
library(flextable)
flextable(model_table1)

library(dplyr)
```

```{r}
#| label: tbl-model_components
#| tbl-cap: components
#| fig-cap: components

# Table of model terms used for GWAS
#Table S1
E1_ft<-flextable(model_table1,cwidth = 0.1,cheight = 0.1)%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")
E1_ft<-E1_ft%>%align(align = "justify",part = "header")%>%align(align = "justify",part = "body")%>%autofit(part="body",hspans = "divided")%>%fontsize(size = 9,part = "body")%>%fontsize(size = 9,part = "header")%>%
bold(bold = TRUE, part = "header")%>%set_caption(caption=as_paragraph(as_chunk("Summary of the fixed, random and residual model terms fitted each timepoint across germination energy and index for genome wide assocation modeling ",props = fp_text_default(font.family = "Times New Roman", bold = FALSE))),fp_par(text.align="justify"),align_with_table = FALSE)%>%
#E1_ft<-E1_ft%>%add_footer_lines(E1_ft, values = "")
footnote(i = c(1), j = c(1),
            value = as_paragraph(
              c("GI: Germination index, GE: Germination Energy, GI_base:unadjusted germination index, PM Date: Post physiological maturity date and timepoint,Q: fixed effect population structure matrix, npc: number of principal components included, Rep: Replication")),
            ref_symbols = c("a"),
            part = "header")

E1_ft_2<-E1_ft%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")

# 
# footnote(i = c(1), j = c(2),
#             value = as_paragraph(
#               c("BSR-Barley Stripe Rust;LR-Leaf Rust;SB-Spot Blotch;SLD-Scald")),
#             ref_symbols = c("b"),
#             part = "header")

  
E1_ft_2
phenoS1
```

We previously reported development of a high-throughput Kompetitive Allele Specific Primer (KASP) marker AlaAT1_L214F for the causal mutation in *HvAlaAT1* discovered by Sato et al. (2016). We also developed KASP assays for a SNP in the 5' UTR of *HvGA20ox1* (GA20ox1_331_5UTR), the E165Q mutation in *HvMKK3* identified in Vetch et al. (2020) for both spring barley and winter barley. A summary of the winter barley haplotypes for the parent lines and checks are found in QTL x environment modeling of malting barley preharvest sprouting (Sweeney et al. 2021). All experimental lines and parent genotypes were monomorphic for dormant HvMKK3. The check lines Charles and Endeavor used as a phs susceptibility check in 2020 and 2021 respectively, have the highly non-dormant (N\*) allele for MKK3 and the dormant AlaAT(D) allele, however we cannot make specific conclusions about this haplotype given the low representation in the winter barley population.

All lines were genotyped with the 50Fk Illumina Infinium iSelect SNP array at the USDA Small Grains Genotyping Lab in Fargo, ND. After filtering poor quality markers, minor allele frequency (MAF) below 0.05, and monomorphic sites, 15,467 polymorphic markers remained and were used for genome-wide association (GWA). After conducting linkage disequilibrium (LD) pruning to reduce high LD blocks that exist in double haploid populations, we retained 8,384 markers for analysis. A summary of the GWA results is presented in Table 1. Models were run for all trait/time point combinations

```{r}

df_begin$Family<-gsub("DH130910","Lightning",df_begin$Family)

df_begin_long<-df_begin%>%mutate(GI_0=GI/GE)%>%mutate(GI_0=ifelse(is.nan(GI_0),NA,GI_0))%>%pivot_longer(names_to="trait",cols=c("GE","GI"))%>%rename(year=Year)
df2<-df_begin_long%>%filter(PM_date%in%c("5","19","47","96","152"))


df2$year<-"2020/2021"
library(arm)
gd_Q<-as.data.frame(WinterGD[,c("Qsd1","JHI_Hv50k_2016_8507")])
table(gd_Q$taxa,gd_Q$Qsd1)
gd_Q$taxa<-rownames(gd_Q)

###




```

```{r}
#| label: fig-PHS_dist
#| fig-cap: "PHS"

#PHS
#Figure 1
df_begin_long
library(Hmisc)
phs_setup<-df_begin_long%>%rbind(df2) %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(gd_Q[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(AlaAT= ifelse(Qsd1==2,'Non dormant','Dormant')) %>% plyr::join(gd_Q[,c('taxa','JHI_Hv50k_2016_8507')])%>%
  filter(Family %nin% c('Lightning'))  %>%dplyr::select(phs,Qsd1,AlaAT,taxa,SourcePLOT,Location,year)%>%unique()
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
# %>% mutate(JHI_Hv50k_2016_8507= ifelse(JHI_Hv50k_2016_8507==2,'minor','ref'))%>%
View(phs_setup)
library(ggplot2)
phs_dist<-phs_setup%>%group_by(taxa)%>%
  ggplot(aes(phs, fill = AlaAT))+
 
  geom_histogram(binwidth = 0.2,col=I("black"))+
  labs(x="PHS score(0-9)",y="Frequecny")+ggtitle(label = paste0("Hisogram of Pre-harvest sprouting values for 2020/2021"))+
theme_bw()+theme(axis.text.x = element_text(angle = 0),plot.title = element_text(hjust=0.5))+
    annotate('rect', xmin=-0.1, xmax=2, ymin=0, ymax=1000, alpha=.1, fill='green')+
    annotate('rect', xmin=2, xmax=4, ymin=0, ymax=1000, alpha=.1, fill='yellow')+
     annotate('rect', xmin=4, xmax=7, ymin=0, ymax=1000, alpha=.1, fill='red')
 

phs_dist
 # stat_bin(aes(y=(phs)/sum(phs), 
 #    label=round((phs)/sum(phs),2)),
 #    geom="text", size=4, binwidth = 0.1, vjust=-1.5)
phenoS1
```

```{r}
load("Rdata/geno_corr.Rdata")
load("Rdata/p_corr.Rdata")
new
vb<-new%>%round(2)%>%as.data.frame(row.names(new))%>% tibble::rownames_to_column("Variable")%>%flextable()%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")
std_border = fp_border(color="black")
vb
vb<-vb%>%align(align = "justify",part = "header")%>%align(align = "justify",part = "body")%>%autofit(part="body",hspans = "divided")%>%fontsize(size = 9,part = "body")%>%fontsize(size = 9,part = "header")%>%
bold(bold = TRUE, part = "header")%>%bold(bold = TRUE, part = "body",j=1)%>%
  set_caption(caption=as_paragraph(as_chunk("Phenotypic and Genotypic correlations between preharvest sprouting(PHS),germination energy(GE) and germination index(GI) at post physiological maturity(PM) timepoints. The lower diagnol corresponds to phenotypic correlation and the upper diagnial corresponds to genotypic correlations",props = fp_text_default(font.family = "Times New Roman", bold = FALSE))),fp_par(text.align="justify"),align_with_table = FALSE)%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")%>%hline(part="body",i=c(1,8) ,border = std_border )%>%vline(part="body",j=c(2,10) ,border = std_border )


corr_table<-vb%>% add_footer_lines(values = "")%>%align(align = c("left"),part = "footer")%>%fontsize(size=8,part="footer")%>%compose(value= as_paragraph(as_i("Note:"),"phs, preharvest sprouting; GE Germination Energy; GI Germination Index; PM Physiological maturity date"), part = "footer")%>%align(align = c("left"),part = "footer")%>%fontsize(size = 8, part = "footer")%>%
  footnote(i = c(1), j = c(4,6,8,12,14,16),
            value = as_paragraph(
              c("Timepoints were only measured in 2021")),
            ref_symbols = c("a"),
            part = "header")


corr_table


#upper is geno, lower is pheno
phenoS1
```

# Results

## Phenotypic Distribution

Preharvest sprouting scores presented were averaged across years as there was a lack of sufficent phs data in the 2020 year alone. PHS score distribution was skewed towards resistance(Figure 1). 95% of genotypes were classified as resistant(0 to 2 score), 4.4% were moderately resistant and 1.6% classified as PHS susceptible. Average GE had significant variation based on year as mean GE in 2021 was substaintally lower than mean GE at equivalent time points.Heritability for GE was high at early timepoints but decreased as primary dormancy loss occured. The slight increase of heritability of GE at later timepoints is most likely attributed to diseased kernels as seeds were not surface sterilized due to population size. GI was initally low at early timepoints and increased as primary dormancy loss occured. Unlike GE, GI variation remained after dormancy resulting in higher heritability at later timepoints. Similar to GE, there was significantly more dormancy in 2021 compared to 2020, as intial genetic variation and heritability was low for PM 5 in 2021. After observed dormancy loss, GI platued at a mean GI of 5.6

#Correlations

Phenotypic correlations between GE and GI were significant between all timepoints except for PM 152. PHS phenotypic correlations were not significant to GE and GI. Genotypic correlations were high for all traits, indicating shared genetic control.

```{r}
# Figure 3
library(tidyr)
library(ggh4x)
library(Hmisc)


op<-df_begin_long%>%rbind(df2)%>% mutate(headerFacet = 'Dataset Year') %>%filter(!trait%in%c("GI_0"))%>%
  ggplot(aes(x = PM_date, y = value, group = GID))+
  geom_smooth(na.rm = TRUE,se=FALSE,linewidth=0.2,colour="black")+ggh4x::facet_nested(trait~headerFacet+year, scales = 'free')+
  geom_vline(xintercept = c(12,33,68), color = 'red')+
  labs(title = 'Germination Energy and Index over time by individual genotype')+theme_bw()
op
# distribution of of timepoint data
```

## Genetics

### KASP marker results

THe population for this experiment used a modified NAM with a common parent Lightning crossed to four malting barley parents. KASP marker analysis revealed that the common parent Lightning and parent Flavia contained the dormant "C" allele for AlaAT and parents KWS Scala, SY Tepee and Wintmalt containing the non dormant "G" allele, making Lightning the donor for the dormant AlaAT. Experimental lines and the parents were monomorhpic for the dormant MKK3 allele, a loci identified in spring malting barley to be indicative of high malting quality but also high PHS susceptiblity. Check varities Charles and Endeavor have a super non dormant allele for MKK3, which provided a consistent check for PHS and germination, however given the low frequency of the haplotype in the experimental population, we cannot make further assertions about the role of HvMKK3 here.

### Population structure

Principal component analysis Principal component analysis revealed variation attributed to the first, second and third PCs at 14.55% and 5.69 and 4.9 % respectively. The PCA plot displays the first two PCs on the x and y axis. The first component separated SY Tepee populations from the wintmalt, KWS Scala and Flavia populations, with Lightning being a common intermediate between the two parents. Position of genotypes remaineds largely unchanged when separate recombinant inbred lines of crosses not including Lightning were included(data not shown). Wintmalt, Scala and Flavia families have closer degree of relatedness.

```{r}
#| label: fig-PCA_DH
#| fig-cap: "Principal components analysis plot"



# Figure 2 PCA plot of double haploid population
library(ggplot2)
load("Rdata/WinterRelationship.Rdata")


WinterPCA = eigen(WinterRelationship)


WinterPVEPCA = WinterPCA$values/sum(WinterPCA$values)
#data.frame(ordinal = 1:10, PVE = WinterPVEPCA[1:10]) %>%plot(., xlab = 'PC', col = 'red') 
WinterPVEPCA[1:5]
winterlinePCAvalues = WinterPCA$vectors %>% data.frame()%>% 
  
  mutate(family = plyr::mapvalues(substr(rownames(WinterGD),1,3), from = toupper(c('BS6','BS7','BS8','BS9','Lig','Fla','Tep','Sca','Win',"End","Cha","Che")), 
                            to = c('Flavia/Lightning','Scala/Lightning','SY_Tepee/Lightning','Wintmalt/Lightning',
      
                                                                'Lightning','Flavia/Lightning','SY_Tepee/Lightning','Scala/Lightning','Wintmalt/Lightning',"Check","Check","Check")),
         taxa = rownames(WinterGD),
         shapes = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), taxa, 'Lines'),
         size = ifelse(taxa %in% toupper(c('Lightning', 'Flavia','Tepee','Wintmalt','Scala')), 3, 2))

PCA_DH<-winterlinePCAvalues %>% mutate(shapes=gsub("(^|[[:space:]])([[:alpha:]])", "\\1\\U\\2", 
     tolower(winterlinePCAvalues$shapes),
     perl = TRUE),size=ifelse(size==2,3,4))%>%filter(family !=c('Check')) %>%
  ggplot(aes(x = X1, y = X2, color = family, shape = shapes)) + geom_point(aes(size = size))+theme_bw() +guides(shape=guide_legend(title = "Variety/Experimental Line"),size="none",color=guide_legend(title="Family/Population"))+xlab(as.character(paste0("PC1: ",round(WinterPVEPCA[1]*100,2),"%")))+ylab(as.character(paste0("PC2: ",round(WinterPVEPCA[2]*100,2),"%")))+ggtitle(lab="Principal component analysis plot of the \n NY winter malting barley double haploid population")+theme(plot.title = element_text(hjust = 0.5))+scale_x_continuous(expand = c(0, 0), limits = c(-0.125, 0.125))+scale_y_continuous(expand = c(0, 0), limits = c(-0.125, 0.125))
PCA_DH
```

```{r}
#| label: fig-AlaAT Distributions
#| fig-cap: "Qsd1 distribution"


#Figure 4
library(Hmisc)
#JHI_Hv50k_2016_314355
WinterGD$taxa<-rownames(WinterGD)
WinterGD<-WinterGD%>%relocate(taxa,.before = 1)
df_begin_long%>%rbind(df2) %>% mutate(year = factor(year, levels = c('2020','2021','2020/2021')))%>%
  plyr::join(WinterGD[,c('taxa','Qsd1')]) %>% filter(Qsd1!= 1) %>% mutate(Qsd1= ifelse(Qsd1==2,'Nondormant','Dormant')) %>% plyr::join(WinterGD[,c('taxa','JHI_Hv50k_2016_310278')]) %>% filter(JHI_Hv50k_2016_310278!= 1) %>% mutate(JHI_Hv50k_2016_310278= ifelse(JHI_Hv50k_2016_310278==2,'minor','ref'))%>%
  filter(Family %nin% c('Lightning'))   %>%
 # filter(!(year == '2020/2021' & TP %in% c('TP1.5','TP2.5','TP3.5'))) %>%
  filter(!(trait =='GE' &value>1.05)) %>%
  ggplot(aes(x=PM_date,y=value,fill = JHI_Hv50k_2016_310278))+
geom_boxplot(outlier.shape =NA)+ggh4x::facet_nested(trait~Family+year, scales = 'free')+

  labs(x="Time Point",y="Germination value")+ggtitle(label = paste0("Germination rates by grouping of AlaAT"))+
theme_bw()+theme(axis.text.x = element_text(angle = 90),plot.title = element_text(hjust=0.5))+guides(fill=guide_legend(title="JHI_Hv50k_2016_310278"))
```

#manhattan plot

```{r}
#Figure 5
load(paste0(path,"/data/GWA_results/STP_all.Rdata"))



ord_list<-STP_all[STP_all$ID=="GI-2020/2021-PM_152",c("chrom","pos","marker","ordinal")]
ord_list<-ord_list%>%arrange(chrom,pos)%>%mutate(ordinal=1:n())
table_chr<-STP_all%>%ungroup()%>%filter()%>%dplyr::select(marker,chrom,pos,ordinal)%>%arrange(chrom,pos)%>%mutate(ordinal=plyr::mapvalues(marker,from=ord_list$marker,to=ord_list$ordinal),ordinal=as.numeric(ordinal))%>%unique()

table_chr_min<-table_chr%>%group_by(chrom)%>%summarise(min=min(pos),min_ord=min(ordinal))
table_chr_max<-table_chr%>%group_by(chrom)%>%summarise(max=max(pos),max_ord=max(ordinal))
table_chr<-merge(table_chr_min,table_chr_max,by="chrom")%>%tidyr::drop_na(chrom)


STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$notes<-"GE-2021-PM_96: did_not_converge"
STP_all[STP_all$ID=="GE-2021-PM_96: did_not_converge",]$ID<-"GE-2021-PM_96"
b

STP_all$ID<-gsub("GI_0","GI0",STP_all$ID)
library(stringr)
STP_all<-STP_all%>%mutate(PM=as.numeric(str_split_i(ID,"_",2)),trait=str_split_i(ID,"-",1),year=str_split_i(ID,"-",2))%>%mutate(log10PVal=-log10(p.value))%>%arrange(year,trait,PM)
WinterPerTPGWAS2<-STP_all
###



Chr_breaks=table_chr$max_ord-table_chr$min_ord
ChrTable<-Chr_breaks

chrLabel = c(1:7, 'UN')
winterOrdinalBreaks = c(ChrTable[1]/2)

winterChrLines = ChrTable[1]#for the first position

for (i in 2:8){
 winterOrdinalBreaks[i] = sum(ChrTable[1:i-1])+ChrTable[i]/2
 winterChrLines[i] = sum(ChrTable[1:i])
}


WinterPerTPGWAS2<-STP_all%>%ungroup()%>%mutate(ordinal=plyr::mapvalues(marker,from=ord_list$marker,to=ord_list$ordinal),ordinal=as.numeric(ordinal))

WinterPerTPGWAS2[WinterPerTPGWAS2$ID=="GE-2021-PM_96"&WinterPerTPGWAS2$log10PVal>3.5&!is.na(WinterPerTPGWAS2$log10PVal),]$log10PVal<-0
#this GWAS TP is giving weird assocations, so I just removed it
```

```{r}



#False discovery rate
#all P values are FDR adjusted, to find the p-value of the FDR rate, I found the max p value for which the FDR p value is less than 0.05, code is below
FDR_rate<-max(WinterPerTPGWAS2[WinterPerTPGWAS2$fdr.pvalues<0.051,]$p.value,na.rm=TRUE)

#even the line is slightly more conservative than the true FDR rate since the nearest FDR value below 0.05 is 0.03. This is important because while we do not want false positive(QQ plots should be used for this), we also shouldn't be over stringent with Bonferonni cutoff, we elimate
#ALaAT_ord<-unique(WinterPerTPGWAS2[WinterPerTPGWAS2$marker=="Qsd1",]$ordinal)
ALaAT_ord=WinterPerTPGWAS2[WinterPerTPGWAS2$marker=="Qsd1",]$ordinal%>%unique()

WinterPerTPGWAS2[WinterPerTPGWAS2$trait=="PHS",]$PM<-4
#All chromosomes
ALL_MM<-WinterPerTPGWAS2%>%ungroup()%>%mutate(trait=ifelse(trait=="GI0","GI",trait))%>% ggplot(aes(ordinal, log10PVal, color = as.factor(PM), shape = year))+geom_point()+ggtitle(label="Genome wide association for Germination Energy and Index \n across timepoints and years")+
  geom_vline(xintercept = winterChrLines, color = 'black')+
 geom_vline(xintercept = ALaAT_ord, color = 'red',linetype='longdash')+
  annotate(geom= 'text', x = ALaAT_ord, y = 12, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+
  scale_x_continuous(label = c("1H","2H", "3H", "4H", "5H", "6H", "7H", "UN"),
                     breaks = winterOrdinalBreaks)+
  ylab('-log(p-value)')+xlab('Chromosome')+ geom_hline(yintercept = -log10(FDR_rate),linetype="longdash")+geom_hline(yintercept = -log10(0.05/9658) ,alpha=0.2,color="red")+theme_bw()+
 facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()

CH5_MM<-WinterPerTPGWAS2%>%ungroup()%>%mutate(trait=ifelse(trait=="GI0","GI",trait))%>% ggplot(aes(ordinal, log10PVal, color = as.factor(PM), shape = year))+geom_point()+ggtitle(label="Genome wide association for Germination Energy, Index and PHS on CH 5")+
  geom_vline(xintercept = winterChrLines, color = 'black')+
 geom_vline(xintercept = 5760, color = 'red',linetype='longdash')+
  annotate(geom= 'text', x = 5760, y = 12, label = 'AlaAT1')+
  geom_vline(xintercept = winterChrLines)+

  ylab('-log(p-value)')+xlab('Chromosome 5')+ geom_hline(yintercept = -log10(FDR_rate),linetype="longdash")+geom_hline(yintercept = -log10(0.05/9658) ,alpha=0.2,color="red")+theme_bw()+
 facet_grid(rows = vars(trait), scales = 'free_y')+theme_bw()+scale_x_continuous(limits = c(5000,7000)) 
#Ch 5


#print(paste("FDR p value at alpha 0.05 :",round(FDR_rate,8)))
#print(paste("Bonferroni cutoff at 0.05 :",round(0.05/9658,8)))
```

Genome wide association results

### Model fitting using asrGWAS

Single step genome wide association analysis was conducted using the ASRgwas @ASRgwas2022 package across all trait, year, PM date and combined year combinations(36 analysis). Not all combinations resulted in identified marker trait associations. Fixed effects were identified by wald tests and if models did not converge with reduced inclusion of effects. Within years, significant fixed effects included Location, Replication(Rep) and the spatial dimension of Row. In analysis across years, Year was identified as a signficant term as either a fixed effect or residual term for all GI trait models and all GE trait models except for GE 2020/2021 at PM date 152. Not all models converged, particularly models for the GE trait as variation was low at early and later timepoints. After accounting for model terms, significant marker traits were identified based on false discovery rate(FDR) adjusted p-values using the R package p-adjust within each GWAS analysis. Traits were considered significant if the FDR adjusted p-value was below 0.05(alpha value. The highest unadjusted p-value that was below the FDR threshold identified in this study was 0.00003.

### Genome wide association analysis

Usining a false discovery rate of alpha=0.05, 50 unique marker trait associations were identifed across the genome. Given the highly structured breeding populations, the LD in this population was high and markers in LD were condensed to single blocks if r2 was greater than 0.95 between markers on specific chromosome. A range of the minimum and maxmimum positions of significant markers within each LD group are presented as we do not have the resolution to determined where in LD blocks the casual variant is located. Significant marker trait associations observed at multiple timepoint trait combinations include JHI_Hv50k_2016_64721 and marker LD group SCRI_RS_135633, JHI_Hv50k_2016_90999, JHI_Hv50k_2016_116224 on 2H; marker SCRI_RS_16934 on 3H; marker LD group JHI_Hv50k_2016_275586, JHI_Hv50k_2016_276615, JHI_Hv50k_2016_276209 on the end of 4H; marker JHI_Hv50k_2016_282096, marker group AlaAT JHI_Hv50k_2016_308652, JHI_Hv50k_2016_308798, JHI_Hv50k_2016_310278 and marker group JHI_Hv50k_2016_310358, JHI_Hv50k_2016_312229, JHI_Hv50k_2016_314246 on 5H; and marker group JHI_Hv50k_2016_450348, JHI_Hv50k_2016_450348, JHI_Hv50k_2016_486280 and marker JHI_Hv50k_2016_507218 on 7H.

Percent variance explained(PVE) was highest (for markers in perfect LD with KASP marker AlaAT . Other significant regions include the marker region JHI_Hv50k_2016_310278- JHI_Hv50k_2016_314246 on 5H associated with PHS and GE, that was not associated with AlaAT as the LD was low(r2=0.1) between the 462656156-486764914 region and the 435636564-448089400 region associated with AlaAT.Potential gene candidates in this region include a GRAM containing/ABA-responsive protein(HORVU.MOREX.r3.5HG0487770.1), a 1-4-alpha glucan branching, and a root meristem growth factor, all associated with seed germination and strach degredation associated with barley germination. Other regions of interest include Dimeric alpha-amylase inhibitor, mitogen activated protein kinase, beta amalayse and 1,4- alpha glucan branching enzyme located on 2H, a IAA amino acid hydrolase identified on 3H, and QTL associated with diastatic power(QDp.DiMo-4H) and beta glucan(QBgnm.MT2-4H) identified in @mohmammed et al 2015. After accounting for LD blocks, gene annotation from barley genome reference version 2 and 3 were gathered into a list of all markers 2 Mb upstream and downstream of either single markers or the minimum and maximum of an LD block using modifications of the GALLO package. Lists assocatied with each markers are documented in supplemental table 3.

# Dicussion

## Dicussion Phenotype distributions

Efforts to identify sources of PHS resistance that do not compromise malting quality has been a key target in malting barley breeding, particularly in regions where there is a high chance for rain events at harvest. Numerous barley mapping studies have identified dormancy loci such as SD1 and SD2 on 5H and SD3 on 7H, but few have evaluated dormancy loss with extensive germination assays in a winter malting barley breeding population. We found that the SD1 locus contributed to significant difference in rate dormancy loss over time in winter malting barley from 5 days to 152 days after physiological maturity. In addtion to the SD1, we found numerous other regions of low to moderate effect associated with dormancy loss, including markers associated with alpha-amylase inhibitors and beta amylase glucan branching enzymes on 2H, a marker two regions for Beta glucan and Diastatic power on 4H, and SD3 on 7H @romagosaIndividualLocusEffects1999. Identifying multiple regions associated with dormancy loss provides breeders with a greater ability to select for a desired balance of seed dormancy in target environments

Observed preharvest sprouting was low in the WMB DH population. Preharvest scoring overlaps at the same post PM range as first timepoint observed in dormancy assays as phs samples are harvested at physiological maturity, after ripened for 4 days and scored 3 days after misting, equivalent to PM 4 when given misting conditions and PM 7 when scored. Despite similar physiological states, the phenotypic correlation of PHS and GE and GI at early timepoints was low. This is most likely due to the the high dormancy observed in 2021 both with PHS and seed dormancy as well as sampling error.

Seed dormancy variation was significant in the WMB barley population. Seed dormancy loss is defined when GE values per line reach 95% and mean seed dormancy loss across lines reached 90%. In 2020, mean seed dormancy loss occured between timepoints $PM_{47}$ and $PM_{68}$ in 2020 and between timepoints $PM_{68}$ and $PM_{96}$ in 2021. Genetic variation for GE was initially high in 2020 and signficantly reduced once seed dormancy loss was near complete. Genetic varaition for GE in 2021 increased slightly and decreased once dormancy loss occured, albeit at later timepoints compared to 2020.GI genetic variation for 2020 and 2021 followed similar rates, but mean GI values were higher at earlier timepoints in 2020 to 2021.. Compared to GI of spring malting barley populations, winter malting barleys were observed to have lower 1 day germination values, even after dormancy loss, which is reflected as a penalty in the calculated GI score and high genetic variation at later timepoints. The low phs incidicene, prolonged dormancy and moderate GI values suggests that selection for barley with shorter dormancy times and higher GI should be a priority in winter malting barley

Genetic correlations between GE, GI and PHS were surprisingly high at early and later timepoints. GWAS analysis results of this region suggest some common MTAs associations between GE, GI and PHS but only at a few specific combinations. AlaAT was found to not be associated with this identified region based on statisical analysis of including AlaAT as an effect in the PHS GWAS model and due to the separate observed LD group identified for AlaAT on 5H (Supplementary Figure 5H LD block). PHS and GI are similar measurements as both observe rate of germination over time, where as GE is a binary observation of seed germination. Marker trait association associated with GE at $PM_{152}$ was in low LD compared to surrounding upstream and downstream markers and most likely was associated with a different factor contributing to GE, such as disease.

## Statistical modeling for GWAS

Fitting single step models can reduce the amount of information loss associated when using fixed effect responses without weights, or using random effects models where BLUPS are either not deregressed and shrunk twice or degressed but partial shrinkage occurs regardless. Fitting single step models for each timepoint produced challenges however, as some models did not converge for some timepoints. The lack of convergence issues associated with with GE and early and late timepoiints was most likely due low variation of the trait. Other model terms such as Location were significant across many combinations, indicating the potential of either field effects or sampling error within locations. Seed dormancy measurements were most often(but not always) scored by two individuals by scoring one replication each and significant fixed effects of replication most likely indicate differences of scoring by an individual or scoring error. Fixed effect terms such Location:Row most likely correspond to some form of sampling error. Although not presented, Qsd1 was tested as a fixed effect in the various models to determine potential epistatic effects associated with the Qsd1 locus. No novel QTL were identified and QTL that had reduced PEV in timepoints where the AlaAT locus explained the most variance were detected in different timepoint, trait and year combinations.

## MTA associations

SD1 was initially discovered as a seed dormancy QTL in barley by @oberthurGeneticAnalysisSeed1995 and has since validated by many researchers @romagosaIndividualLocusEffects1999 @han1996, cloned @satoAlanineAminotransferaseControls2016 and sanger sequenced by @vetchMutationsHvMKK3HvAlaAT12020 and @sweeneyInteractionsBarleySD12021. AlaAT has also been a target of CRISPR-Cas9 targeted mutagenesis where targeted mutants showed highly delayed, but not inhibited germination. [@hisano2022]

### Role of alanine amino transferase in population

Alanine amino transferase played a significant role in dormancy break in our winter malting barley population. Figures 3a-c show germination rate and energy for each family of the population grouped by haplotype for AlaAT over years 2020, 2021, and combined analysis 2020/2021. A check "family" was included for comparison to Charles for 2020 and Endeavor for 2021. Parents Scala, SY Tepee and Wintmalt contained the non-dormant AlaAT allele(N) and the common parent DH130910 and Flavia contained the dormant allele(D). In both years, GE showed clear differentiation over time based on AlaAT haplotype, particularly at time point 2(19 days post PM) for 2020 and timepoints 1.5-3.5(19 days post PM through 68 days post PM). GI showed clear differentiation both across timepoints and within timepoints based on the AlaAT haplotype. The differences in GE and GI for both years based on each haplotype demonstrate that AlaAT significantly affects germination rate, energy and subsequently the rate of dormancy breakage. We also observed significant differences between GE and GI depending on the year. In 2020, dormancy break occurred around time point 3(47 days post PM). In 2021, dormancy break was substantially higher and did not occur until timepoints 3.5(68 days post PM) and 4(96 days post PM). The significant effect of the year environment on dormancy demonstrates the need to test seed germination traits across multiple years

### 2H

Three potential candidate genes were identified 160022796-469196911 bp region on 2H potenetially related to seed dormancy. The first identifed dimeric alpha-amylase inhibitor @menaMajorBarleyAllergen1992 (Bdai-1)(HORVU.MOREX.r2.2HG0176960) associated with bakers asthma and with foam retention in beer. From a germination perspective, the inhibition of alpha amylase could delay the breakdown of starch and potentially delay germination. The second potential gene of interest includes a mitogen-activated protein kinase, a similar protein structure of mitogen-activated protein kinase kinase 3(MKK3) associated with germination, malting quality and PHS sucsecptiblity in spring barley @sweeney. The third potential gene of 1,4-alpha-glucan-branching enzyme have been described by @reginaControlStarchBranching2010

#3H

Other potential novel loci include HvVP1 (Viviparous-1) which may be associated with JHI-Hv50k-2016-165725 on chromosome 3H and segregating in the SY Tepee family. HvVP1 is a master transcription factor regulator that controls switching between seed maturation and germination in barley HvVP1 @abrahamDevelopmentalSwitchGene2016

#5H B

The region between 435636564-448089400 on 5H included the KASP marker AlaAT_L214F and 50K snp markers in high ld (r=0.95) with AlaAT.\
An abscisic aldehyde oxidase 3(HORVU.MOREX.r2.5HG0399320) was also identified in this region. \### AlaAT

#5H C

Three potential gene candidates were found in the 5H 462656156-486764914 LD region based on gene annotations and previous literature. The first includes gene HORVU.MOREX.r3.5HG0486380 identified as*ABI5(Abscisic Acid INSENSITIVE 5)* @seilerAbscisicAcidFlux2014 @collinBarleyABI5Abscisic2020 in response to drought tolerance and stress. The second includes a includes a GEM GRAM-containing/ABA-responsive protein(HORVU.MOREX.r3.5HG0487770.1) @mauriGEMMemberGRAM2016 located dowstream of asABI5. They found that that mutant types of the GEM would delay germination . Mauri et al. 2016 also discovered that 24 hour cold treatment eliminated the effects of GEM, which could explain why the loci was detected in PHS assays at a higher prevalence than GE and GI assays, as PHS samples did not undergo any cold treatment. Increases in potential ABA content in the 2021 year could also prolonged the effect of this gene in PM_5 for GI in 2021. Conditions for grainfill in 2021 were dry up until harvest which could of induced higher ABA assocated with this gene. A third potential gene related to seed germination identified by gene annotation includes a 1,4-alpha-glucan branching enzyme(GlgB(HORVU.MOREX.r3.5HG0488090.1) identified in morex version 3, a key regulator in amylopection degredatio @sunTwoGenesEncoding1998, @reginaControlStarchBranching2010. Isoamylase HvISA3 (HORVU.MOREX.r2.5HG0404420) of similar function was identified in the same region for morex version 2 @shuQuantificationAmyloseAmylopectin2014. 1,4-alpha-glucan branching,a key regulator in amylopection degredatio @sunTwoGenesEncoding1998, @reginaControlStarchBranching2010. Isoamylases are importanct for starch degredation as they hydrolyze Î±-(1,6) glycosidic linkages and debranch amylopectin during grain filling @gousReviewAmylopectinSynthesis2017. Starch with higher amylose content is hydrolyzed more slowly by amylolytic enzymes and higher amylose content has been hypothesized to be a contributing factor either grain dormancy or delay in germination rate observed by moderate GI values

##4H

On 4H two marker regions were of potential interest for seed dormancy were found. The first region included another alpha amylase inhibitor as well as a QTL identified for protein content QGpc.StMo-4H @mohammadiGenomewideAssociationStudy2015. The second LD region on 4H located on the far distal end of 4H was identified in numerous trait timepoint combinations. The malting quality QTLs QDp.DiMo-4H(Diastatic Power) and QBgnm.MT2-4H(Beta Glucan) identified in @mohammadiGenomewideAssociationStudy2015. Higher diastatic power could result in faster degredation of starches in the barley grain, resulting in barley grains. Variation in beta glucan content could be an indirect indicator of seed dormancy, as higher beta glucan can indicate that some seed dormancy exists.

#7H

On 7H marker located near the centromeric regions 14595309-387792817 were identified to have a\
beta-amylase(HORVU.MOREX.r2.7HG0614220), ABA responsive binding factor(HORVU.MOREX.r2.7HG0533970), Glucan endo-1,3-beta-glucosidase(HORVU.MOREX.r2.7HG0535140) and protein(HORVU.MOREX.r2.7HG0535020). The region was also identied as SD3 in @romagosaIndividualLocusEffects1999

Genetic and phenotypic correlations between GE and GI were high across all timepoints and traits except for $PM_{152}$. The KASP marker AlaAT and 50K markers located in high LD(r=0.95) in the 435636564-448089400 region on 5H was detected for the most trait, timepoint and year combinations. PVE was highest in this region.

Initial KASPs assay of AlaAT and MKK3 of the winter barley population from @sweeney2021_QTL found variation in AlaAT for the parents of the winter malting barley population but monomorphic for the dormant allele of MKK3, suggesting that AlaAT would likely be the largest genetic factor contributing to the rate of seed dormancy loss. The KASP marker for AlaAT and markers in the same LD region as AlaAT explained the highest PVE.

```{r}
#Table 2
library(dplyr)
library(flextable)
load("ms/Rdata/Significant_marker_table.Rdata")


marker_tableR[marker_tableR$SNP=="JHI_Hv50k_2016_310278, JHI_Hv50k_2016_312229, JHI_Hv50k_2016_314246",]$Gene_candidate<-"GRAM-containing/ABA-responsive protein(HORVU.MOREX.r3.5HG0487770.1), 1,4-alpha-glucan branching enzymeGlgB(HORVU.MOREX.r3.5HG0488090.1),root meristem growth factor(HORVU.MOREX.r3.5HG0486840.1),ABI5(Abscisic Acid INSENSITIVE 5)HORVU.MOREX.r3.5HG048638"
dim(marker_tableR)
sig_markersAll<-flextable(marker_tableR,cwidth = 0.1,cheight = 0.1)%>%theme_zebra()%>%font(fontname = "Times New Roman",part = "header")
sig_markersAll
```

The observed dormancy for most non dormant haplotypes at earlier post physiological maturity (PM) days is encouraging for balancing the selection of high germination rate and maintaining initial seed dormancy for PHS. If we assume that a malster, at the earliest, would start malting barley approximately 47 days after harvest from the field, we will want to select lines that maintain dormancy until 50-60 days post PM. Most non-dormant AlaAT haplotypes fit this profile for the 2020 and 2021 crop year. However, given the substantial dormancy we observed in 2021, we must be careful in considering how it relates to our PHS values observed for that same year. PHS will need to be tested in multiple years where conditions would develop less dormant barley. If we continue to have low PHS mean and variation across multiple years with potentially differing levels of dormancy per year, we can then focus the selection pressure to increase germination rate with less concern of reducing our PHS resistance for the given AlaAT haplotypes.

### Impacts on malting quality(if there is time)

GWAS for malting quality did not identify any signficant marker trait associations, however correlation between observered phenotypic values and GEBV was high. Most malting quality characteristics were responses to dormancy

### Selection of seed dormancy

Barley with reduced dormancy is a pre requiste for malt quality and not a correlated trait. Experimental lines with little dormancy can have poor malting quality while lines with phs resistance and some dormancy can have high malting quality once the dormancy loss is complete. Given the characterization of seed dormancy loss overtime presented there are many lines with high PHS resistance and short dormancy periods that could be suited for malting quality. Typically, malt that is stored after harvest is stored for at least 30-60 days after harvest.

Potential areas of improvement for seed germination include improvement of GI once there is complete dormancy loss. Many lines observed in this population reached peak GI of 5.5 compared to Charles, Endeavor and KWS Scala GI values in the ranges of 7 to 9. Initial germination delay could be caused by a number of previously described genes. ABA

### How can germination assays be improved

Dormancy assays current limited throughput and many considerations were made to balance the statistical power effectiveness and allocation of time to maximize observation of dormancy loss over time. With the advent of imaging technologies, higher throughput methods using image analysis could not only classify germinated kernels to non germination kernels, but also track the vigor of kernels over the course of the germination experiment, as germination vigor is an important factor in malting

# Conclusion

# 
